<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.4.553">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="dcterms.date" content="2024-04-23">

<title>Biologia della Conservazione 23/24 - Biologia della Conservazione</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../">
<script src="../../site_libs/quarto-html/quarto.js"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-text-highlighting-styles">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-text-highlighting-styles">
<script src="../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-bootstrap" data-mode="light">
<link href="../../site_libs/bootstrap/bootstrap-dark.min.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-bootstrap" data-mode="dark">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>

  <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>

<meta property="og:title" content="Biologia della Conservazione 23/24 - Biologia della Conservazione">
<meta property="og:description" content="Class 10 - Inbreeding">
<meta property="og:site_name" content="Biologia della Conservazione 23/24">
<meta property="og:locale" content="en_US">
</head>

<body class="nav-sidebar docked nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="../../index.html">
    <span class="navbar-title">Biologia della Conservazione 23/24</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item">
    <a class="nav-link active" href="../../index.html" aria-current="page"> 
<span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../calendar.html"> 
<span class="menu-text">Calendar</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../contact.html"> 
<span class="menu-text">Contact</span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
          <div class="quarto-navbar-tools">
  <a href="" class="quarto-color-scheme-toggle quarto-navigation-tool  px-1" onclick="window.quartoToggleColorScheme(); return false;" title="Toggle dark mode"><i class="bi"></i></a>
  <a href="" class="quarto-reader-toggle quarto-navigation-tool px-1" onclick="window.quartoToggleReader(); return false;" title="Toggle reader mode">
  <div class="quarto-reader-toggle-btn">
  <i class="bi"></i>
  </div>
</a>
</div>
      </div> <!-- /container-fluid -->
    </nav>
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item">Biologia della Conservazione</li></ol></nav>
        <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation docked overflow-auto">
    <div class="pt-lg-2 mt-2 text-center sidebar-header">
      <a href="../../index.html" class="sidebar-logo-link">
      <img src="../.././figures/avatar.jpg" alt="" class="sidebar-logo py-0 d-lg-inline d-none">
      </a>
      <div class="sidebar-tools-main">
    <a href="mailto:giuliano.colosimo@protonmail.com" title="" class="quarto-navigation-tool px-1" aria-label=""><i class="bi bi-envelope"></i></a>
</div>
      </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true">
 <span class="menu-text">Classes</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../.././classes/1/class_1.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Class 1 - Introduction to conservation biology</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../.././classes/2/class_2.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Class 2 - Biodiversity</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../.././classes/3/class_3.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Class 3 - Biodiversity &amp; Extinctions</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../.././classes/4/class_4.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Class 4 - Populations (1)</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../.././classes/5/class_5.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Class 5 - Populations (2) - Intro to Conservation Genetics</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../.././classes/6/class_6.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Class 6 - Conservation Genetics</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../.././classes/7/class_7.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Class 7 - Genetic Drift/Effective Population Size</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../.././classes/8/class_8.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Class 8 - Manuscript discussion-Genetic Drift/Effective Population Size</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../.././classes/9/class_9.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Class 9 - Manuscript discussion-Genetic Drift/Effective Population Size</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../.././classes/10/class_10.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Class 10 - Inbreeding</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../.././classes/11/class_11.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Class 11 - Arguments review for mid term</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../.././classes/12/class_12.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Class 12 - Mid-term exam revision/Genetic load</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../.././classes/13/class_13.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Class 13 - Population Structure/Hybridization</span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#topics" id="toc-topics" class="nav-link active" data-scroll-target="#topics">Topics</a></li>
  <li><a href="#inbreeding" id="toc-inbreeding" class="nav-link" data-scroll-target="#inbreeding">Inbreeding</a></li>
  <li><a href="#what-is-inbreeding" id="toc-what-is-inbreeding" class="nav-link" data-scroll-target="#what-is-inbreeding">What is inbreeding?</a></li>
  <li><a href="#the-pedigree-inbreeding-coefficient" id="toc-the-pedigree-inbreeding-coefficient" class="nav-link" data-scroll-target="#the-pedigree-inbreeding-coefficient">The pedigree inbreeding coefficient</a></li>
  <li><a href="#how-to-estimate-the-pedigree" id="toc-how-to-estimate-the-pedigree" class="nav-link" data-scroll-target="#how-to-estimate-the-pedigree">How to estimate the pedigree?</a></li>
  <li><a href="#inbreeding-depression" id="toc-inbreeding-depression" class="nav-link" data-scroll-target="#inbreeding-depression">Inbreeding depression</a></li>
  <li><a href="#the-concept-of-genetic-load" id="toc-the-concept-of-genetic-load" class="nav-link" data-scroll-target="#the-concept-of-genetic-load">The concept of genetic load</a></li>
  <li><a href="#references" id="toc-references" class="nav-link" data-scroll-target="#references">References</a></li>
  </ul>
<div class="toc-actions"><ul><li><a href="https://github.com/quarto-dev/quarto-demo/issues/new" class="toc-action"><i class="bi bi-github"></i>Report an issue</a></li></ul></div><div class="quarto-alternate-formats"><h2>Other Formats</h2><ul><li><a href="class_10.html"><i class="bi bi-file-slides"></i>RevealJS</a></li></ul></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Biologia della Conservazione</h1>
<p class="subtitle lead">Class 10 - Inbreeding</p>
</div>



<div class="quarto-title-meta">

    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">April 23, 2024</p>
    </div>
  </div>
  
    
  </div>
  


</header>


<table class="table">
<thead>
<tr class="header">
<th>Student</th>
<th>Title</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>FALVO</td>
<td><a href="https://drive.proton.me/urls/9WVQGE6J5M#GEndO0gEvVJq">Developing gene drive technologies to eradicate invasive rodents from islands</a></td>
</tr>
<tr class="even">
<td>D’ALESSANDRO</td>
<td><a href="https://drive.proton.me/urls/HQMV65ETB0#vB21xDrPrAXN">Population genetic analysis of the recently rediscovered hula painted frog (<em>Latonia nigriventer</em>) reveals high genetic diversity and low inbreeding</a></td>
</tr>
<tr class="odd">
<td>PIROLI</td>
<td><a href="https://drive.proton.me/urls/TD1F4DDC70#hzwxC3gFnM51">First evidence of recruitment in critically endangered Galapagos pink iguanas</a></td>
</tr>
</tbody>
</table>
<hr>
<section id="topics" class="level2">
<h2 class="anchored" data-anchor-id="topics">Topics</h2>
<div class="columns">
<div class="column" style="width:50%;">
<ol type="1">
<li><span style="color: green;">Introduction to conservation biology</span></li>
<li><span style="color: green;">Biodiversity</span></li>
<li><span style="color: green;">Biodiversity and extinctions</span></li>
<li><span style="color: green;">Populations</span></li>
<li><span style="color: green;">Basics of conservation genetics</span></li>
<li><span style="color: green;">Genetic drift</span></li>
<li><span style="color: green;">Effective population size</span></li>
<li><span style="color: orange;">Inbreeding depression</span></li>
<li><span style="color: blue;">Population structure</span></li>
<li><span style="color: blue;">Hybridization</span></li>
<li><span style="color: blue;">Applying genetic information to conservation</span></li>
</ol>
</div><div class="column" style="width:50%;">
<ol start="12" type="1">
<li><span style="color: blue;">Managing genetic diversity in captive populations</span></li>
<li><span style="color: blue;">Managing genetic diversity in the wild</span></li>
<li><span style="color: blue;">Population viability analysis</span></li>
<li><span style="color: blue;">Invasive species</span></li>
<li><span style="color: blue;">CITES and trade regulations of endangered species</span></li>
<li><span style="color: blue;">Conservation and climate change</span></li>
<li><span style="color: blue;">Future climate impacts</span></li>
<li><span style="color: blue;">Conservation economics</span></li>
<li><span style="color: blue;">Conservation technology</span></li>
</ol>
</div>
</div>
</section>
<section id="inbreeding" class="level2">
<h2 class="anchored" data-anchor-id="inbreeding">Inbreeding</h2>
<div class="columns">
<div class="column" style="width:50%;">
<p><img src="./class_10_1.jpg" class="img-fluid"></p>
</div><div class="column" style="width:50%;">
<p><img src="./class_10_2.jpg" class="img-fluid"></p>
<ul>
<li>There cannot be a class on inbreeding without these pictures. I am not sure who should receive credit, but if you type “inbreeding” in your browser and go to image results they immediately pop out.</li>
</ul>
</div>
</div>
<hr>
</section>
<section id="what-is-inbreeding" class="level2">
<h2 class="anchored" data-anchor-id="what-is-inbreeding">What is inbreeding?</h2>
<ul>
<li>In its most simple definition inbreeding is the mating between related individuals.</li>
<li>In large population inbreeding can happen because of the non-random assortment of gametes. In many tree species spatially closer trees are more likely to be related than trees further apart.</li>
</ul>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="./class_10_3.png" class="img-fluid figure-img"></p>
<figcaption>Larger filled circles indicate tree locations; open circles around them delineate seed shadows within which most seed will fall. Small symbols indicate seedlings from each maternal parent. After <span class="citation" data-cites="Allendorf2022">Allendorf et al. (<a href="#ref-Allendorf2022" role="doc-biblioref">2022</a>)</span>, pg. 178</figcaption>
</figure>
</div>
<hr>
<ul>
<li>Can inbreeding happen when there is random mating?</li>
</ul>
<p>. . .</p>
<ul>
<li>Imagine the case of small populations. In the case of a population with 10 breeders, after a few generations the most distantly related individuals will be cousins, and even if the population mates at random, individuals will be highly inbred.</li>
<li>This has been called <strong>inbreeding effect of small populations</strong> <span class="citation" data-cites="Crow1970">(<a href="#ref-Crow1970" role="doc-biblioref">Crow &amp; Kimura, 1970</a>, pg. 101)</span>.</li>
</ul>
<p>. . .</p>
<ul>
<li>So, we can consider an individual inbred if its parents share a common ancestor and its genes (one, two, or big chunks of the genome) are identical by descent (IBD).</li>
<li>What we know from studying evolution and from the examples on the colaescence theory is that eventually all parents of an individual share a common ancestor.</li>
<li>We must, therefore, define inbreeding relative to some “base” population in which we assume all individuals are unrelated to one another <span class="citation" data-cites="Allendorf2022">(<a href="#ref-Allendorf2022" role="doc-biblioref">Allendorf et al., 2022</a>, pg. 371)</span>.</li>
</ul>
<hr>
<ul>
<li>The concept of inbreeding is relatively easy to understand, but can also be confusing as the same term is associated with different phenomena (gentic drift, non random mating within populations, increase in genome wide homozygosity).</li>
<li>All these concepts are measured by the same quantity, inbreeding coefficients <span class="citation" data-cites="Templeton1994">(<a href="#ref-Templeton1994" role="doc-biblioref">Templeton &amp; Read, 1994</a>)</span>, and all these concepts can, to certain degree, be used interchangeably as they all refer to a situation in which relatives mate with each other <span class="citation" data-cites="Keller2002a">(<a href="#ref-Keller2002a" role="doc-biblioref">Keller &amp; Waller, 2002</a>)</span>.</li>
<li>As inbred individuals receive two copies of the same allele from its parents, they will present higher homozygosity and lower heterozygosity throughout the entire genome.</li>
<li>We can, then, define the inbreeding coefficient (<span class="math inline">\(F\)</span>) as that proportion of the genome in an individual that is IBD (sometime an alternative word is used, <em>i.e.</em>, autozygous).</li>
<li>The alternative to autozygous is allozygous, which means that an individual possesses two alleles that are descendant from different ancestral alleles in the base population.</li>
</ul>
<hr>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="./class_10_4.png" class="img-fluid figure-img"></p>
<figcaption>Figure from <span class="citation" data-cites="Allendorf2022">Allendorf et al. (<a href="#ref-Allendorf2022" role="doc-biblioref">2022</a>)</span>, pg. 374</figcaption>
</figure>
</div>
<hr>
</section>
<section id="the-pedigree-inbreeding-coefficient" class="level2">
<h2 class="anchored" data-anchor-id="the-pedigree-inbreeding-coefficient">The pedigree inbreeding coefficient</h2>
<ul>
<li>The pedigree inbreeding coefficient (<span class="math inline">\(F_P\)</span>) refers to the expected increase in homozigosity of the inbred individual relative to the population founders or founder.</li>
<li>The value ranges from <span class="math inline">\(0\)</span> (indicating no inbreeding) to <span class="math inline">\(1\)</span> (indicating completely inbred).</li>
</ul>
<div class="columns">
<div class="column" style="width:50%;">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="./class_10_5.png" class="img-fluid figure-img"></p>
<figcaption>Classical representation of a pedigree. Squares are males and rounds are females <span class="citation" data-cites="Allendorf2022">(<a href="#ref-Allendorf2022" role="doc-biblioref">Allendorf et al., 2022</a>)</span>.</figcaption>
</figure>
</div>
</div><div class="column" style="width:50%;">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="./class_10_6.png" class="img-fluid figure-img"></p>
<figcaption>Path diagram representing the pedigree. Approach used to calculate the inbreeding coefficient <span class="math inline">\(F_p\)</span> <span class="citation" data-cites="Allendorf2022">(<a href="#ref-Allendorf2022" role="doc-biblioref">Allendorf et al., 2022</a>)</span>.</figcaption>
</figure>
</div>
</div>
</div>
<hr>
</section>
<section id="how-to-estimate-the-pedigree" class="level2">
<h2 class="anchored" data-anchor-id="how-to-estimate-the-pedigree">How to estimate the pedigree?</h2>
<ol type="1">
<li>Draw the pedigree as shown in previous figure where each individual appears only one.</li>
<li>Identify the ancestors of the male and female that generated X. If the father and the mother share no ancestors <span class="math inline">\(F_p = 0\)</span>.</li>
<li>Trace all of the paths that lead from one of X’s parents, through the common ancestor, and then back again to the other parent of X (in our example there is only one path: from D to A and from A to E. The notation to indicate such path is D<u>A</u>E).</li>
<li>Determine <span class="math inline">\(N\)</span>, the number of individuals in the path, excluding our target individual X. In our case <span class="math inline">\(N = 3\)</span>.</li>
<li>With a single path we can use</li>
</ol>
<p><span class="math display">\[
\label{fp}
\tag{10.1}
F_P = (\frac{1}{2})^N(1+F_{P(CA)})
\]</span></p>
<p>where <span class="math inline">\(F_{P(CA)}\)</span> is the pedigree inbreeding coefficient of the common ancestor.</p>
<hr>
<ol start="6" type="1">
<li>If there is more than just one path, <span class="math inline">\(F_P\)</span> becomes the sum of all <span class="math inline">\(F_P\)</span> values obtained from each single paths.</li>
</ol>
<p><span class="math display">\[
\label{fp2}
\tag{10.2}
F_P = \sum[(\frac{1}{2})^N(1+F_{P(CA)})]
\]</span></p>
<p>So, in our previous example, taken from <span class="citation" data-cites="Allendorf2022">Allendorf et al. (<a href="#ref-Allendorf2022" role="doc-biblioref">2022</a>)</span>, we can use equation <span class="math inline">\(\ref{fp}\)</span> and the estimated inbreeding coefficient is</p>
<p><span class="math display">\[
F_{P(X)}= (\frac{1}{2})^3(1+0) =0.125
\]</span></p>
<p>We can read this result in two complementary ways:</p>
<ol type="1">
<li>Individual X is expected to have 12.5% of his genes IBD;</li>
<li>X is expected to have its heterozygosity reduced by 12.5% compared to the ancestral population.</li>
</ol>
<hr>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="./class_10_7.png" class="img-fluid figure-img"></p>
<figcaption>What is <span class="math inline">\(F_{p(K)}\)</span> based on the above path diagram representation? After <span class="citation" data-cites="Allendorf2022">Allendorf et al. (<a href="#ref-Allendorf2022" role="doc-biblioref">2022</a>)</span>.</figcaption>
</figure>
</div>
<hr>
<ul>
<li>There is one loop that contains a common ancestor of both parents of K (I<u>G</u>J). Therefore, using equation <span class="math inline">\(\ref{fp}\)</span>, <span class="math inline">\(F_{P(K)} = (1/2)^3 (1 + F_{P(G)})\)</span>.</li>
<li>Individual G is also inbred; there is one loop with three individuals through a common ancestor for individual G (D<u>B</u>E). Therefore, <span class="math inline">\(F_{P(G)} = (1/2)^3 (1 + F_{P(B)})\)</span>. Individual B is not inbred (<span class="math inline">\(F_{P(B)} = 0\)</span>) since she is a founder in this pedigree.</li>
<li>Solving the equations: <span class="math inline">\(F_{P(G)} = 0.125\)</span>, and <span class="math inline">\(F_{P(K)} = (1/2)^3 (1.125) = 0.141\)</span>.</li>
</ul>
<hr>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="./class10_pedigreeBonus.png" class="img-fluid figure-img"></p>
<figcaption>After <span class="citation" data-cites="Allendorf2022">Allendorf et al. (<a href="#ref-Allendorf2022" role="doc-biblioref">2022</a>)</span></figcaption>
</figure>
</div>
<hr>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="./class10_pedigreeBonus2.png" class="img-fluid figure-img"></p>
<figcaption>Calculation of the pedigree inbreeding coefficient for the previous slide. After <span class="citation" data-cites="Allendorf2022">Allendorf et al. (<a href="#ref-Allendorf2022" role="doc-biblioref">2022</a>)</span></figcaption>
</figure>
</div>
<hr>
<ul>
<li><p>Two major short comings while using <span class="math inline">\(F_P\)</span>:</p>
<ol type="1">
<li>It does not account for the inbreeding level of individuals that are not included in the reference pedigree.</li>
<li>This metric is sub-optimal to obtain an accurate estimate of the proportion of the genome that is IBD, because there is a much variation in the proportion of genes that are IBD imputable to chance alone (“They all have the same”Coefficient of Inbreeding” but differ in the extent of homozygosity” <span class="citation" data-cites="Fisher1965">(<a href="#ref-Fisher1965" role="doc-biblioref">Fisher, 1965</a>)</span>).</li>
</ol></li>
<li><p>In general, it is often very complicated to obtain accurate pedigrees for natural populations, and while this approach is still massively used in all species for which we can maintain appropriate pedigree databases, coefficient of inbreeding in wild animals is often obtained through the analysis of molecular markers.</p></li>
<li><p>Molecular marker-based measures of <span class="math inline">\(F\)</span> range from simple estimates of individual heterozygosity to more advanced methods that use mapped loci to estimate <span class="math inline">\(F\)</span> via identification of IBD chromosome segments as stretches of homozygous genotypes at mapped single nucleotide polymorphisms <span class="citation" data-cites="Allendorf2022">(<a href="#ref-Allendorf2022" role="doc-biblioref">Allendorf et al., 2022</a>, pg. 381)</span>.</p></li>
</ul>
<hr>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="./class_10_8.png" class="img-fluid figure-img"></p>
<figcaption>Heterozygosity across chromosomes 10-14 of scandinavian Wolves as analyzed by <span class="citation" data-cites="Kardos2017">Kardos et al. (<a href="#ref-Kardos2017" role="doc-biblioref">2017</a>)</span>.</figcaption>
</figure>
</div>
<hr>
</section>
<section id="inbreeding-depression" class="level2">
<h2 class="anchored" data-anchor-id="inbreeding-depression">Inbreeding depression</h2>
<ul>
<li>Inbred individuals generally have reduced fitness in comparison to noninbred individuals from the same population because of their increased homozygosity.</li>
<li>Inbreeding depression is the reduction in fitness (or phenotype value) of progeny from matings between related individuals relative to the fitness of progeny between unrelated individuals</li>
<li>Inbreeding depression in natural populations contributes to the extinction of populations under some circumstances <span class="citation" data-cites="Allendorf2022">(<a href="#ref-Allendorf2022" role="doc-biblioref">Allendorf et al., 2022</a>)</span>.</li>
<li>The importance of inbreeding depression has been debated since the time of Darwin. Even today, the importance of inbreeding depression is sometimes questioned by some conservationists and policy-makers.</li>
</ul>
<hr>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="./class_10_11.png" class="img-fluid figure-img"></p>
<figcaption>Did you know that Charles Darwin and his wife were cousins?</figcaption>
</figure>
</div>
<hr>
<div class="columns">
<div class="column" style="width:50%;">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="./class_10_9.jpg" class="img-fluid figure-img"></p>
<figcaption>The concomitant contribution of different factors in determining a vortex effect leading the populations to extinction. Image downloaded <a href="https://bio1152.nicerweb.com/Locked/media/ch56/vortex.html">here</a>.</figcaption>
</figure>
</div>
</div><div class="column" style="width:50%;">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="./class_10_10.png" class="img-fluid figure-img"></p>
<figcaption>A more recent take on extinction vortex <span class="citation" data-cites="Allendorf2022">(<a href="#ref-Allendorf2022" role="doc-biblioref">Allendorf et al., 2022</a>)</span>.</figcaption>
</figure>
</div>
</div>
</div>
<hr>
<ul>
<li>Inbreeding depression can result from either increased homozygosity or reduced heterozygosity.</li>
<li>Aren’t increased homozygosity or reduced heterozygosity two sides of the same coin?</li>
</ul>
<p>. . .</p>
<ul>
<li>Yes, but the mechanism related to a fitness reduction is distinct! Increased homozygosity leads to expression of a greater number of deleterious recessive alleles in inbred individuals, thereby lowering their fitness. Reduced heterozygosity reduces fitness of inbred individuals at loci where the heterozygotes have a selective advantage over homozygotes (heterozygous advantage or overdominance).<br>
</li>
<li>Most evidence show support for the primary problem of inbreeding depression is increased homozygosity <span class="citation" data-cites="Allendorf2022">(<a href="#ref-Allendorf2022" role="doc-biblioref">Allendorf et al., 2022</a>)</span>.</li>
</ul>
<hr>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="./class10_callimico.jpg" class="img-fluid figure-img"></p>
<figcaption>Callimico monkeys from the <a href="https://www.beardsleyzoo.org">Connecticut Zoo</a></figcaption>
</figure>
</div>
<hr>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="./class_10_12.png" class="img-fluid figure-img"></p>
<figcaption>Relationship between inbreeding (<span class="math inline">\(F_P\)</span>) and infant survival in captive callimico monkeys <span class="citation" data-cites="Lacy1997">(<a href="#ref-Lacy1997" role="doc-biblioref">Lacy, 1997</a>)</span>.</figcaption>
</figure>
</div>
<hr>
<ul>
<li>It is crucial to know the mechanisms causing inbreeding depression because it affects the ability of a population to “adapt” to inbreeding.</li>
<li>A population could adapt to inbreeding if inbreeding depression is caused by deleterious recessive alleles that potentially could be removed (purged by selection).</li>
<li>However, inbreeding depression caused by heterozygous advantage cannot be purged because overdominant loci will always suffer reduced fitness as homozygosity increases due to increased inbreeding <span class="citation" data-cites="Allendorf2022">(<a href="#ref-Allendorf2022" role="doc-biblioref">Allendorf et al., 2022</a>)</span>.</li>
<li>Deleterious recessive alleles are present in the genome of all species because they are continually introduced by mutation, and natural selection is inefficient at removing them because most copies are “hidden” phenotypically in heterozygotes that do not have reduced fitness <span class="citation" data-cites="Allendorf2022">(<a href="#ref-Allendorf2022" role="doc-biblioref">Allendorf et al., 2022</a>)</span>.</li>
<li>We, therefore, expect all species to show some inbreeding depression due to the increase in homozygosity of recessive deleterious alleles.</li>
</ul>
<hr>
<ul>
<li>The effects of inbreeding depression on survival are often measured by the mean number of “lethal equivalents” (LEs) per diploid genome.</li>
<li>A lethal equivalent is a set of deleterious alleles that would cause death if homozygous.</li>
<li>One lethal equivalent may be a single allele that is lethal when homozygous, two alleles each with a probability of 0.5 of causing death when homozygous, or 10 alleles each with a probability of 0.10 of causing death when homozygous.</li>
<li>The number of LEs present in a species or population is generally estimated by regressing survival on the inbreeding coefficient.</li>
<li>The effects of inbreeding on the probability of survival, S, can be expressed as a function of <span class="math inline">\(F_P\)</span> following <span class="citation" data-cites="Morton1956">Morton, Crow, &amp; Muller (<a href="#ref-Morton1956" role="doc-biblioref">1956</a>)</span>.</li>
</ul>
<p><span class="math display">\[
\label{le}
\tag{10.3}
S = e^{-(A+BF_P)} \\ ln \ S= -A-BF_P
\]</span></p>
<p>where <span class="math inline">\(e^{-A}\)</span> is survival in an outbred population, and B is the inbreeding load.</p>
<hr>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="./class_10_13.png" class="img-fluid figure-img"></p>
<figcaption>Relationship between inbreeding coefficient (F) and reduction in fitness. Inbreeding depression is the reduction in fitness of inbred individuals and measured by the number of LEs per gamete (B). The bracket shows the reduction in fitness (inbreeding depression) predicted for individuals with an F of 0.5. After <span class="citation" data-cites="Keller2002a">Keller &amp; Waller (<a href="#ref-Keller2002a" role="doc-biblioref">2002</a>)</span> and <span class="citation" data-cites="Allendorf2022">Allendorf et al. (<a href="#ref-Allendorf2022" role="doc-biblioref">2022</a>)</span>.</figcaption>
</figure>
</div>
<hr>
</section>
<section id="the-concept-of-genetic-load" class="level2">
<h2 class="anchored" data-anchor-id="the-concept-of-genetic-load">The concept of genetic load</h2>
<ul>
<li><p>Genetic load is the relative difference in fitness between the theoretically fittest genotype within a population and the average genotype in that population.</p></li>
<li><p>The term “load” was first used by <span class="citation" data-cites="Muller1950">Muller (<a href="#ref-Muller1950" role="doc-biblioref">1950</a>)</span> in his consideration of the possible effects of increased mutation rates because of nuclear weapons.</p>
<ol type="1">
<li>Mutation load: The decrease in fitness caused by the accumulation of deleterious mutations.</li>
<li>Segregation load: The decrease in fitness caused by heterozygous advantage.</li>
<li>Drift load: The decrease in fitness caused by the increase in frequency of deleterious alleles resulting from genetic drift.</li>
<li>Migration load: The reduction in fitness caused by the migration into a population of individuals that are less adapted to the local environment than native individuals.</li>
</ol></li>
</ul>
</section>
<section id="references" class="level2">




</section>

<div id="quarto-appendix" class="default"><section class="quarto-appendix-contents" role="doc-bibliography" id="quarto-bibliography"><h2 class="anchored quarto-appendix-heading">References</h2><div id="refs" class="references csl-bib-body hanging-indent" data-entry-spacing="0" data-line-spacing="2" role="list">
<div id="ref-Allendorf2022" class="csl-entry" role="listitem">
Allendorf, F. W., Funk, W. C., Aitken, S. N., Byrne, M., Luikart, G., &amp; Antunes, A. (2022). <em><span class="nocase">Conservation and the Genomics of Populations</span></em>. Oxford University Press. Retrieved from <a href="https://doi.org/10.1093/oso/9780198856566.001.0001">https://doi.org/10.1093/oso/9780198856566.001.0001</a>
</div>
<div id="ref-Crow1970" class="csl-entry" role="listitem">
Crow, J. F., &amp; Kimura, M. (1970). <em><span class="nocase">An Introduction to Population Genetics Theory</span></em> (p. 591). New York. Retrieved from <a href="http://www.amazon.de/Introduction-Population-Genetics-Theory-James/dp/0060414383/ref=sr{\_}1{\_}3?ie=UTF8{\&amp;}qid=1321741251{\&amp;}sr=8-3">http://www.amazon.de/Introduction-Population-Genetics-Theory-James/dp/0060414383/ref=sr{\_}1{\_}3?ie=UTF8{\&amp;}qid=1321741251{\&amp;}sr=8-3</a>
</div>
<div id="ref-Fisher1965" class="csl-entry" role="listitem">
Fisher, R. A. (1965). <em>The theory of inbreeding.</em> (Second.). Academic Press.
</div>
<div id="ref-Kardos2017" class="csl-entry" role="listitem">
Kardos, M., Åkesson, M., Fountain, T., Flagstad, Øystein, Liberg, O., Olason, P., Sand, H., et al. (2017). Genomic consequences of intensive inbreeding in an isolated wolf population. <em>Nature Ecology <span>&amp;</span> Evolution</em>, <em>2</em>(1), 124–131. Springer Science; Business Media <span>LLC</span>. Retrieved from <a href="https://doi.org/10.1038%2Fs41559-017-0375-4">https://doi.org/10.1038%2Fs41559-017-0375-4</a>
</div>
<div id="ref-Keller2002a" class="csl-entry" role="listitem">
Keller, L. F., &amp; Waller, D. M. (2002). <a href="https://doi.org/10.1016/S0169-5347(02)02489-8"><span class="nocase">Inbreeding effects in wild populations</span></a>. <em>Trends in Ecology and Evolution</em>, <em>17</em>(5), 230–241.
</div>
<div id="ref-Lacy1997" class="csl-entry" role="listitem">
Lacy, R. C. (1997). <span class="nocase">Importance of Genetic Variation to the Viability of Mammalian Populations</span>. <em>Journal of Mammalogy</em>, <em>78</em>(2), 320–335. Retrieved from <a href="http://www.jstor.org/stable/1382885?origin=crossref">http://www.jstor.org/stable/1382885?origin=crossref</a>
</div>
<div id="ref-Morton1956" class="csl-entry" role="listitem">
Morton, N. E., Crow, J. F., &amp; Muller, H. J. (1956). <a href="https://doi.org/10.1073/pnas.42.11.855"><span class="nocase">An Estimate of the Mutational Damage in Man From Data on Consanguineous Marriages.</span></a> <em>Proceedings of the National Academy of Sciences of the United States of America</em>, <em>42</em>(11), 855–863.
</div>
<div id="ref-Muller1950" class="csl-entry" role="listitem">
Muller, H. J. (1950). <a href="https://www.ncbi.nlm.nih.gov/pubmed/14771033">Our load of mutations.</a> <em>Am J Hum Genet</em>, <em>2</em>(2), 111–176.
</div>
<div id="ref-Templeton1994" class="csl-entry" role="listitem">
Templeton, A. R., &amp; Read, B. (1994). Inbreeding: One word, several meanings, much confusion. <em>Conservation genetics</em> (pp. 91–105). Birkh<span>ä</span>user Basel. Retrieved from <a href="https://doi.org/10.1007%2F978-3-0348-8510-2_9">https://doi.org/10.1007%2F978-3-0348-8510-2_9</a>
</div>
</div></section></div></main> <!-- /main -->

<script>
  // htmlwidgets need to know to resize themselves when slides are shown/hidden.
  // Fire the "slideenter" event (handled by htmlwidgets.js) when the current
  // slide changes (different for each slide format).
  (function () {
    // dispatch for htmlwidgets
    function fireSlideEnter() {
      const event = window.document.createEvent("Event");
      event.initEvent("slideenter", true, true);
      window.document.dispatchEvent(event);
    }

    function fireSlideChanged(previousSlide, currentSlide) {
      fireSlideEnter();

      // dispatch for shiny
      if (window.jQuery) {
        if (previousSlide) {
          window.jQuery(previousSlide).trigger("hidden");
        }
        if (currentSlide) {
          window.jQuery(currentSlide).trigger("shown");
        }
      }
    }

    // hookup for slidy
    if (window.w3c_slidy) {
      window.w3c_slidy.add_observer(function (slide_num) {
        // slide_num starts at position 1
        fireSlideChanged(null, w3c_slidy.slides[slide_num - 1]);
      });
    }

  })();
</script>

<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const disableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'prefetch';
    }
  }
  const enableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'stylesheet';
    }
  }
  const manageTransitions = (selector, allowTransitions) => {
    const els = window.document.querySelectorAll(selector);
    for (let i=0; i < els.length; i++) {
      const el = els[i];
      if (allowTransitions) {
        el.classList.remove('notransition');
      } else {
        el.classList.add('notransition');
      }
    }
  }
  const toggleGiscusIfUsed = (isAlternate, darkModeDefault) => {
    const baseTheme = document.querySelector('#giscus-base-theme')?.value ?? 'light';
    const alternateTheme = document.querySelector('#giscus-alt-theme')?.value ?? 'dark';
    let newTheme = '';
    if(darkModeDefault) {
      newTheme = isAlternate ? baseTheme : alternateTheme;
    } else {
      newTheme = isAlternate ? alternateTheme : baseTheme;
    }
    const changeGiscusTheme = () => {
      // From: https://github.com/giscus/giscus/issues/336
      const sendMessage = (message) => {
        const iframe = document.querySelector('iframe.giscus-frame');
        if (!iframe) return;
        iframe.contentWindow.postMessage({ giscus: message }, 'https://giscus.app');
      }
      sendMessage({
        setConfig: {
          theme: newTheme
        }
      });
    }
    const isGiscussLoaded = window.document.querySelector('iframe.giscus-frame') !== null;
    if (isGiscussLoaded) {
      changeGiscusTheme();
    }
  }
  const toggleColorMode = (alternate) => {
    // Switch the stylesheets
    const alternateStylesheets = window.document.querySelectorAll('link.quarto-color-scheme.quarto-color-alternate');
    manageTransitions('#quarto-margin-sidebar .nav-link', false);
    if (alternate) {
      enableStylesheet(alternateStylesheets);
      for (const sheetNode of alternateStylesheets) {
        if (sheetNode.id === "quarto-bootstrap") {
          toggleBodyColorMode(sheetNode);
        }
      }
    } else {
      disableStylesheet(alternateStylesheets);
      toggleBodyColorPrimary();
    }
    manageTransitions('#quarto-margin-sidebar .nav-link', true);
    // Switch the toggles
    const toggles = window.document.querySelectorAll('.quarto-color-scheme-toggle');
    for (let i=0; i < toggles.length; i++) {
      const toggle = toggles[i];
      if (toggle) {
        if (alternate) {
          toggle.classList.add("alternate");     
        } else {
          toggle.classList.remove("alternate");
        }
      }
    }
    // Hack to workaround the fact that safari doesn't
    // properly recolor the scrollbar when toggling (#1455)
    if (navigator.userAgent.indexOf('Safari') > 0 && navigator.userAgent.indexOf('Chrome') == -1) {
      manageTransitions("body", false);
      window.scrollTo(0, 1);
      setTimeout(() => {
        window.scrollTo(0, 0);
        manageTransitions("body", true);
      }, 40);  
    }
  }
  const isFileUrl = () => { 
    return window.location.protocol === 'file:';
  }
  const hasAlternateSentinel = () => {  
    let styleSentinel = getColorSchemeSentinel();
    if (styleSentinel !== null) {
      return styleSentinel === "alternate";
    } else {
      return false;
    }
  }
  const setStyleSentinel = (alternate) => {
    const value = alternate ? "alternate" : "default";
    if (!isFileUrl()) {
      window.localStorage.setItem("quarto-color-scheme", value);
    } else {
      localAlternateSentinel = value;
    }
  }
  const getColorSchemeSentinel = () => {
    if (!isFileUrl()) {
      const storageValue = window.localStorage.getItem("quarto-color-scheme");
      return storageValue != null ? storageValue : localAlternateSentinel;
    } else {
      return localAlternateSentinel;
    }
  }
  const darkModeDefault = false;
  let localAlternateSentinel = darkModeDefault ? 'alternate' : 'default';
  // Dark / light mode switch
  window.quartoToggleColorScheme = () => {
    // Read the current dark / light value 
    let toAlternate = !hasAlternateSentinel();
    toggleColorMode(toAlternate);
    setStyleSentinel(toAlternate);
    toggleGiscusIfUsed(toAlternate, darkModeDefault);
  };
  // Ensure there is a toggle, if there isn't float one in the top right
  if (window.document.querySelector('.quarto-color-scheme-toggle') === null) {
    const a = window.document.createElement('a');
    a.classList.add('top-right');
    a.classList.add('quarto-color-scheme-toggle');
    a.href = "";
    a.onclick = function() { try { window.quartoToggleColorScheme(); } catch {} return false; };
    const i = window.document.createElement("i");
    i.classList.add('bi');
    a.appendChild(i);
    window.document.body.appendChild(a);
  }
  // Switch to dark mode if need be
  if (hasAlternateSentinel()) {
    toggleColorMode(true);
  } else {
    toggleColorMode(false);
  }
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp("https:\/\/giulianocolosimo\.github\.com");
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-left">
<p>© Copyright 2024 Giuliano Colosimo</p>
</div>   
    <div class="nav-footer-center">
      &nbsp;
    <div class="toc-actions d-sm-block d-md-none"><ul><li><a href="https://github.com/quarto-dev/quarto-demo/issues/new" class="toc-action"><i class="bi bi-github"></i>Report an issue</a></li></ul></div></div>
    <div class="nav-footer-right">
<p>This page is built with ❤️ and <a href="https://quarto.org/">Quarto</a>. Inspiration for this site comes from <a href="https://www.mm218.dev">here</a>.</p>
</div>
  </div>
</footer>




</body></html>