<!DOCTYPE html>
<html lang="en"><head>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-html/tabby.min.js"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/light-border.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-html.min.css" rel="stylesheet" data-mode="light">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles"><meta charset="utf-8">
  <meta name="generator" content="quarto-1.4.553">

  <meta name="dcterms.date" content="2024-05-07">
  <title>Biologia della Conservazione 23/24 - Biologia della Conservazione</title>
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no, minimal-ui">
  <link rel="stylesheet" href="../../site_libs/revealjs/dist/reset.css">
  <link rel="stylesheet" href="../../site_libs/revealjs/dist/reveal.css">
  <style>
    code{white-space: pre-wrap;}
    span.smallcaps{font-variant: small-caps;}
    div.columns{display: flex; gap: min(4vw, 1.5em);}
    div.column{flex: auto; overflow-x: auto;}
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
    ul.task-list{list-style: none;}
    ul.task-list li input[type="checkbox"] {
      width: 0.8em;
      margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
      vertical-align: middle;
    }
    /* CSS for citations */
    div.csl-bib-body { }
    div.csl-entry {
      clear: both;
      margin-bottom: 0em;
    }
    .hanging-indent div.csl-entry {
      margin-left:2em;
      text-indent:-2em;
    }
    div.csl-left-margin {
      min-width:2em;
      float:left;
    }
    div.csl-right-inline {
      margin-left:2em;
      padding-left:1em;
    }
    div.csl-indent {
      margin-left: 2em;
    }  </style>
  <link rel="stylesheet" href="../../site_libs/revealjs/dist/theme/quarto.css">
  <link href="../../site_libs/revealjs/plugin/quarto-line-highlight/line-highlight.css" rel="stylesheet">
  <link href="../../site_libs/revealjs/plugin/reveal-menu/menu.css" rel="stylesheet">
  <link href="../../site_libs/revealjs/plugin/reveal-menu/quarto-menu.css" rel="stylesheet">
  <link href="../../site_libs/revealjs/plugin/reveal-chalkboard/font-awesome/css/all.css" rel="stylesheet">
  <link href="../../site_libs/revealjs/plugin/reveal-chalkboard/style.css" rel="stylesheet">
  <link href="../../site_libs/revealjs/plugin/quarto-support/footer.css" rel="stylesheet">
  <style type="text/css">

  .callout {
    margin-top: 1em;
    margin-bottom: 1em;  
    border-radius: .25rem;
  }

  .callout.callout-style-simple { 
    padding: 0em 0.5em;
    border-left: solid #acacac .3rem;
    border-right: solid 1px silver;
    border-top: solid 1px silver;
    border-bottom: solid 1px silver;
    display: flex;
  }

  .callout.callout-style-default {
    border-left: solid #acacac .3rem;
    border-right: solid 1px silver;
    border-top: solid 1px silver;
    border-bottom: solid 1px silver;
  }

  .callout .callout-body-container {
    flex-grow: 1;
  }

  .callout.callout-style-simple .callout-body {
    font-size: 1rem;
    font-weight: 400;
  }

  .callout.callout-style-default .callout-body {
    font-size: 0.9rem;
    font-weight: 400;
  }

  .callout.callout-titled.callout-style-simple .callout-body {
    margin-top: 0.2em;
  }

  .callout:not(.callout-titled) .callout-body {
      display: flex;
  }

  .callout:not(.no-icon).callout-titled.callout-style-simple .callout-content {
    padding-left: 1.6em;
  }

  .callout.callout-titled .callout-header {
    padding-top: 0.2em;
    margin-bottom: -0.2em;
  }

  .callout.callout-titled .callout-title  p {
    margin-top: 0.5em;
    margin-bottom: 0.5em;
  }
    
  .callout.callout-titled.callout-style-simple .callout-content  p {
    margin-top: 0;
  }

  .callout.callout-titled.callout-style-default .callout-content  p {
    margin-top: 0.7em;
  }

  .callout.callout-style-simple div.callout-title {
    border-bottom: none;
    font-size: .9rem;
    font-weight: 600;
    opacity: 75%;
  }

  .callout.callout-style-default  div.callout-title {
    border-bottom: none;
    font-weight: 600;
    opacity: 85%;
    font-size: 0.9rem;
    padding-left: 0.5em;
    padding-right: 0.5em;
  }

  .callout.callout-style-default div.callout-content {
    padding-left: 0.5em;
    padding-right: 0.5em;
  }

  .callout.callout-style-simple .callout-icon::before {
    height: 1rem;
    width: 1rem;
    display: inline-block;
    content: "";
    background-repeat: no-repeat;
    background-size: 1rem 1rem;
  }

  .callout.callout-style-default .callout-icon::before {
    height: 0.9rem;
    width: 0.9rem;
    display: inline-block;
    content: "";
    background-repeat: no-repeat;
    background-size: 0.9rem 0.9rem;
  }

  .callout-title {
    display: flex
  }
    
  .callout-icon::before {
    margin-top: 1rem;
    padding-right: .5rem;
  }

  .callout.no-icon::before {
    display: none !important;
  }

  .callout.callout-titled .callout-body > .callout-content > :last-child {
    padding-bottom: 0.5rem;
    margin-bottom: 0;
  }

  .callout.callout-titled .callout-icon::before {
    margin-top: .5rem;
    padding-right: .5rem;
  }

  .callout:not(.callout-titled) .callout-icon::before {
    margin-top: 1rem;
    padding-right: .5rem;
  }

  /* Callout Types */

  div.callout-note {
    border-left-color: #4582ec !important;
  }

  div.callout-note .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAAEU0lEQVRYCcVXTWhcVRQ+586kSUMMxkyaElstCto2SIhitS5Ek8xUKV2poatCcVHtUlFQk8mbaaziwpWgglJwVaquitBOfhQXFlqlzSJpFSpIYyXNjBNiTCck7x2/8/LeNDOZxDuEkgOXe++553zfefee+/OYLOXFk3+1LLrRdiO81yNqZ6K9cG0P3MeFaMIQjXssE8Z1JzLO9ls20MBZX7oG8w9GxB0goaPrW5aNMp1yOZIa7Wv6o2ykpLtmAPs/vrG14Z+6d4jpbSKuhdcSyq9wGMPXjonwmESXrriLzFGOdDBLB8Y6MNYBu0dRokSygMA/mrun8MGFN3behm6VVAwg4WR3i6FvYK1T7MHo9BK7ydH+1uurECoouk5MPRyVSBrBHMYwVobG2aOXM07sWrn5qgB60rc6mcwIDJtQrnrEr44kmy+UO9r0u9O5/YbkS9juQckLed3DyW2XV/qWBBB3ptvI8EUY3I9p/67OW+g967TNr3Sotn3IuVlfMLVnsBwH4fsnebJvyGm5GeIUA3jljERmrv49SizPYuq+z7c2H/jlGC+Ghhupn/hcapqmcudB9jwJ/3jvnvu6vu5lVzF1fXyZuZZ7U8nRmVzytvT+H3kilYvH09mLWrQdwFSsFEsxFVs5fK7A0g8gMZjbif4ACpKbjv7gNGaD8bUrlk8x+KRflttr22JEMRUbTUwwDQScyzPgedQHZT0xnx7ujw2jfVfExwYHwOsDTjLdJ2ebmeQIlJ7neo41s/DrsL3kl+W2lWvAga0tR3zueGr6GL78M3ifH0rGXrBC2aAR8uYcIA5gwV8zIE8onoh8u0Fca/ciF7j1uOzEnqcIm59sEXoGc0+z6+H45V1CvAvHcD7THztu669cnp+L0okAeIc6zjbM/24LgGM1gZk7jnRu1aQWoU9sfUOuhrmtaPIO3YY1KLLWZaEO5TKUbMY5zx8W9UJ6elpLwKXbsaZ4EFl7B4bMtDv0iRipKoDQT2sNQI9b1utXFdYisi+wzZ/ri/1m7QfDgEuvgUUEIJPq3DhX/5DWNqIXDOweC2wvIR90Oq3lDpdMIgD2r0dXvGdsEW5H6x6HLRJYU7C69VefO1x8Gde1ZFSJLfWS1jbCnhtOPxmpfv2LXOA2Xk2tvnwKKPFuZ/oRmwBwqRQDcKNeVQkYcOjtWVBuM/JuYw5b6isojIkYxyYAFn5K7ZBF10fea52y8QltAg6jnMqNHFBmGkQ1j+U43HMi2xMar1Nv0zGsf1s8nUsmUtPOOrbFIR8bHFDMB5zL13Gmr/kGlCkUzedTzzmzsaJXhYawnA3UmARpiYj5ooJZiUoxFRtK3X6pgNPv+IZVPcnwbOl6f+aBaO1CNvPW9n9LmCp01nuSaTRF2YxHqZ8DYQT6WsXT+RD6eUztwYLZ8rM+rcPxamv1VQzFUkzFXvkiVrySGQgJNvXHJAxiU3/NwiC03rSf05VBaPtu/Z7/B8Yn/w7eguloAAAAAElFTkSuQmCC');
  }

  div.callout-note.callout-style-default .callout-title {
    background-color: #dae6fb
  }

  div.callout-important {
    border-left-color: #d9534f !important;
  }

  div.callout-important .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAAEKklEQVRYCcVXTWhcVRS+575MJym48A+hSRFr00ySRQhURRfd2HYjk2SSTokuBCkU2o0LoSKKraKIBTcuFCoidGFD08nkBzdREbpQ1EDNIv8qSGMFUboImMSZd4/f9zJv8ibJMC8xJQfO3HPPPef7zrvvvnvviIkpC9nsw0UttFunbUhpFzFtarSd6WJkStVMw5xyVqYTvkwfzuf/5FgtkVoB0729j1rjXwThS7Vio+Mo6DNnvLfahoZ+i/o32lULuJ3NNiz7q6+pyAUkJaFF6JwaM2lUJlV0MlnQn5aTRbEu0SEqHUa0A4AdiGuB1kFXRfVyg5d87+Dg4DL6m2TLAub60ilj7A1Ec4odSAc8X95sHh7+ZRPCFo6Fnp7HfU/fBng/hi10CjCnWnJjsxvDNxWw0NfV6Rv5GgP3I3jGWXumdTD/3cbEOP2ZbOZp69yniG3FQ9z1jD7bnBu9Fc2tKGC2q+uAJOQHBDRiZX1x36o7fWBs7J9ownbtO+n0/qWkvW7UPIfc37WgT6ZGR++EOJyeQDSb9UB+DZ1G6DdLDzyS+b/kBCYGsYgJbSQHuThGKRcw5xdeQf8YdNHsc6ePXrlSYMBuSIAFTGAtQo+VuALo4BX83N190NWZWbynBjhOHsmNfFWLeL6v+ynsA58zDvvAC8j5PkbOcXCMg2PZFk3q8MjI7WAG/Dp9AwP7jdGBOOQkAvlFUB+irtm16I1Zw9YBcpGTGXYmk3kQIC/Cds55l+iMI3jqhjAuaoe+am2Jw5GT3Nbz3CkE12NavmzN5+erJW7046n/CH1RO/RVa8lBLozXk9uqykkGAyRXLWlLv5jyp4RFsG5vGVzpDLnIjTWgnRy2Rr+tDKvRc7Y8AyZq10jj8DqXdnIRNtFZb+t/ZRtXcDiVnzpqx8mPcDWxgARUqx0W1QB9MeUZiNrV4qP+Ehc+BpNgATsTX8ozYKL2NtFYAHc84fG7ndxUPr+AR/iQSns7uSUufAymwDOb2+NjK27lEFocm/EE2WpyIy/Hi66MWuMKJn8RvxIcj87IM5Vh9663ziW36kR0HNenXuxmfaD8JC7tfKbrhFr7LiZCrMjrzTeGx+PmkosrkNzW94ObzwocJ7A1HokLolY+AvkTiD/q1H0cN48c5EL8Crkttsa/AXQVDmutfyku0E7jShx49XqV3MFK8IryDhYVbj7Sj2P2eBxwcXoe8T8idsKKPRcnZw1b+slFTubwUwhktrfnAt7J++jwQtLZcm3sr9LQrjRzz6cfMv9aLvgmnAGvpoaGLxM4mAEaLV7iAzQ3oU0IvD5x9ix3yF2RAAuYAOO2f7PEFWCXZ4C9Pb2UsgDeVnFSpbFK7/IWu7TPTvBqzbGdCHOJQSxiEjt6IyZmxQyEJHv6xyQsYk//moVFsN2zP6fRImjfq7/n/wFDguUQFNEwugAAAABJRU5ErkJggg==');
  }

  div.callout-important.callout-style-default .callout-title {
    background-color: #f7dddc
  }

  div.callout-warning {
    border-left-color: #f0ad4e !important;
  }

  div.callout-warning .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAAETklEQVRYCeVWW2gcVRg+58yaTUnizqbipZeX4uWhBEniBaoUX1Ioze52t7sRq6APio9V9MEaoWlVsFasRq0gltaAPuxms8lu0gcviE/FFOstVbSIxgcv6SU7EZqmdc7v9+9mJtNks51NTUH84ed889/PP+cmxP+d5FIbMJmNbpREu4WUkiTtCicKny0l1pIKmBzovF2S+hIJHX8iEu3hZJ5lNZGqyRrGSIQpq15AzF28jgpeY6yk6GVdrfFqdrD6Iw+QlB8g0YS2g7dyQmXM/IDhBhT0UCiRf59lfqmmDvzRt6kByV/m4JjtzuaujMUM2c5Z2d6JdKrRb3K2q6mA+oYVz8JnDdKPmmNthzkAk/lN63sYPgevrguc72aZX/L9C6x09GYyxBgCX4NlvyGUHOKELlm5rXeR1kchuChJt4SSwyddZRXgvwMGvYo4QSlk3/zkHD8UHxwVJA6zjZZqP8v8kK8OWLnIZtLyCAJagYC4rTGW/9Pqj92N/c+LUaAj27movwbi19tk/whRCIE7Q9vyI6yvRpftAKVTdUjOW40X3h5OXsKCdmFcx0xlLJoSuQngnrJe7Kcjm4OMq9FlC7CMmScQANuNvjfP3PjGXDBaUQmbp296S5L4DrpbrHN1T87ZVEZVCzg1FF0Ft+dKrlLukI+/c9ENo+TvlTDbYFvuKPtQ9+l052rXrgKoWkDAFnvh0wTOmYn8R5f4k/jN/fZiCM1tQx9jQQ4ANhqG4hiL0qIFTGViG9DKB7GYzgubnpofgYRwO+DFjh0Zin2m4b/97EDkXkc+f6xYAPX0KK2I/7fUQuwzuwo/L3AkcjugPNixC8cHf0FyPjWlItmLxWw4Ou9YsQCr5fijMGoD/zpdRy95HRysyXA74MWOnscpO4j2y3HAVisw85hX5+AFBRSHt4ShfLFkIMXTqyKFc46xdzQM6XbAi702a7sy04J0+feReMFKp5q9esYLCqAZYw/k14E/xcLLsFElaornTuJB0svMuJINy8xkIYuL+xPAlWRceH6+HX7THJ0djLUom46zREu7tTkxwmf/FdOZ/sh6Q8qvEAiHpm4PJ4a/doJe0gH1t+aHRgCzOvBvJedEK5OFE5jpm4AGP2a8Dxe3gGJ/pAutug9Gp6he92CsSsWBaEcxGx0FHytmIpuqGkOpldqNYQK8cSoXvd+xLxXADw0kf6UkJNFtdo5MOgaLjiQOQHcn+A6h5NuL2s0qsC2LOM75PcF3yr5STuBSAcGG+meA14K/CI21HcS4LBT6tv0QAh8Dr5l93AhZzG5ZJ4VxAqdZUEl9z7WJ4aN+svMvwHHL21UKTd1mqvChH7/Za5xzXBBKrUcB0TQ+Ulgkfbi/H/YT5EptrGzsEK7tR1B7ln9BBwckYfMiuSqklSznIuoIIOM42MQO+QnduCoFCI0bpkzjCjddHPN/F+2Yu+sd9bKNpVwHhbS3LluK/0zgfwD0xYI5dXuzlQAAAABJRU5ErkJggg==');
  }

  div.callout-warning.callout-style-default .callout-title {
    background-color: #fcefdc
  }

  div.callout-tip {
    border-left-color: #02b875 !important;
  }

  div.callout-tip .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAADr0lEQVRYCe1XTWgTQRj9ZjZV8a9SPIkKgj8I1bMHsUWrqYLVg4Ue6v9BwZOxSYsIerFao7UiUryIqJcqgtpimhbBXoSCVxUFe9CTiogUrUp2Pt+3aUI2u5vdNh4dmMzOzHvvezuz8xNFM0mjnbXaNu1MvFWRXkXEyE6aYOYJpdW4IXuA4r0fo8qqSMDBU0v1HJUgVieAXxzCsdE/YJTdFcVIZQNMyhruOMJKXYFoLfIfIvVIMWdsrd+Rpd86ZmyzzjJmLStqRn0v8lzkb4rVIXvnpScOJuAn2ACC65FkPzEdEy4TPWRLJ2h7z4cArXzzaOdKlbOvKKX25Wl00jSnrwVxAg3o4dRxhO13RBSdNvH0xSARv3adTXbBdTf64IWO2vH0LT+cv4GR1DJt+DUItaQogeBX/chhbTBxEiZ6gftlDNXTrvT7co4ub5A6gp9HIcHvzTa46OS5fBeP87Qm0fQkr4FsYgVQ7Qg+ZayaDg9jhg1GkWj8RG6lkeSacrrHgDaxdoBiZPg+NXV/KifMuB6//JmYH4CntVEHy/keA6x4h4CU5oFy8GzrBS18cLJMXcljAKB6INjWsRcuZBWVaS3GDrqB7rdapVIeA+isQ57Eev9eCqzqOa81CY05VLd6SamW2wA2H3SiTbnbSxmzfp7WtKZkqy4mdyAlGx7ennghYf8voqp9cLSgKdqNfa6RdRsAAkPwRuJZNbpByn+RrJi1RXTwdi8RQF6ymDwGMAtZ6TVE+4uoKh+MYkcLsT0Hk8eAienbiGdjJHZTpmNjlbFJNKDVAp2fJlYju6IreQxQ08UJDNYdoLSl6AadO+fFuCQqVMB1NJwPm69T04Wv5WhfcWyfXQB+wXRs1pt+nCknRa0LVzSA/2B+a9+zQJadb7IyyV24YAxKp2Jqs3emZTuNnKxsah+uabKbMk7CbTgJx/zIgQYErIeTKRQ9yD9wxVof5YolPHqaWo7TD6tJlh7jQnK5z2n3+fGdggIOx2kaa2YI9QWarc5Ce1ipNWMKeSG4DysFF52KBmTNMmn5HqCFkwy34rDg05gDwgH3bBi+sgFhN/e8QvRn8kbamCOhgrZ9GJhFDgfcMHzFb6BAtjKpFhzTjwv1KCVuxHvCbsSiEz4CANnj84cwHdFXAbAOJ4LTSAawGWFn5tDhLMYz6nWeU2wJfIhmIJBefcd/A5FWQWGgrWzyORZ3Q6HuV+Jf0Bj+BTX69fm1zWgK7By1YTXchFDORywnfQ7GpzOo6S+qECrsx2ifVQAAAABJRU5ErkJggg==');
  }

  div.callout-tip.callout-style-default .callout-title {
    background-color: #ccf1e3
  }

  div.callout-caution {
    border-left-color: #fd7e14 !important;
  }

  div.callout-caution .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAACV0lEQVRYCdVWzWoUQRCuqp2ICBLJXgITZL1EfQDBW/bkzUMUD7klD+ATSHBEfAIfQO+iXsWDxJsHL96EHAwhgzlkg8nBg25XWb0zIb0zs9muYYWkoKeru+vn664fBqElyZNuyh167NXJ8Ut8McjbmEraKHkd7uAnAFku+VWdb3reSmRV8PKSLfZ0Gjn3a6Xlcq9YGb6tADjn+lUfTXtVmaZ1KwBIvFI11rRXlWlatwIAAv2asaa9mlB9wwygiDX26qaw1yYPzFXg2N1GgG0FMF8Oj+VIx7E/03lHx8UhvYyNZLN7BwSPgekXXLribw7w5/c8EF+DBK5idvDVYtEEwMeYefjjLAdEyQ3M9nfOkgnPTEkYU+sxMq0BxNR6jExrAI31H1rzvLEfRIdgcv1XEdj6QTQAS2wtstEALLG1yEZ3QhH6oDX7ExBSFEkFINXH98NTrme5IOaaA7kIfiu2L8A3qhH9zRbukdCqdsA98TdElyeMe5BI8Rs2xHRIsoTSSVFfCFCWGPn9XHb4cdobRIWABNf0add9jakDjQJpJ1bTXOJXnnRXHRf+dNL1ZV1MBRCXhMbaHqGI1JkKIL7+i8uffuP6wVQAzO7+qVEbF6NbS0LJureYcWXUUhH66nLR5rYmva+2tjRFtojkM2aD76HEGAD3tPtKM309FJg5j/K682ywcWJ3PASCcycH/22u+Bh7Aa0ehM2Fu4z0SAE81HF9RkB21c5bEn4Dzw+/qNOyXr3DCTQDMBOdhi4nAgiFDGCinIa2owCEChUwD8qzd03PG+qdW/4fDzjUMcE1ZpIAAAAASUVORK5CYII=');
  }

  div.callout-caution.callout-style-default .callout-title {
    background-color: #ffe5d0
  }

  </style>
  <style type="text/css">
    .reveal div.sourceCode {
      margin: 0;
      overflow: auto;
    }
    .reveal div.hanging-indent {
      margin-left: 1em;
      text-indent: -1em;
    }
    .reveal .slide:not(.center) {
      height: 100%;
    }
    .reveal .slide.scrollable {
      overflow-y: auto;
    }
    .reveal .footnotes {
      height: 100%;
      overflow-y: auto;
    }
    .reveal .slide .absolute {
      position: absolute;
      display: block;
    }
    .reveal .footnotes ol {
      counter-reset: ol;
      list-style-type: none; 
      margin-left: 0;
    }
    .reveal .footnotes ol li:before {
      counter-increment: ol;
      content: counter(ol) ". "; 
    }
    .reveal .footnotes ol li > p:first-child {
      display: inline-block;
    }
    .reveal .slide ul,
    .reveal .slide ol {
      margin-bottom: 0.5em;
    }
    .reveal .slide ul li,
    .reveal .slide ol li {
      margin-top: 0.4em;
      margin-bottom: 0.2em;
    }
    .reveal .slide ul[role="tablist"] li {
      margin-bottom: 0;
    }
    .reveal .slide ul li > *:first-child,
    .reveal .slide ol li > *:first-child {
      margin-block-start: 0;
    }
    .reveal .slide ul li > *:last-child,
    .reveal .slide ol li > *:last-child {
      margin-block-end: 0;
    }
    .reveal .slide .columns:nth-child(3) {
      margin-block-start: 0.8em;
    }
    .reveal blockquote {
      box-shadow: none;
    }
    .reveal .tippy-content>* {
      margin-top: 0.2em;
      margin-bottom: 0.7em;
    }
    .reveal .tippy-content>*:last-child {
      margin-bottom: 0.2em;
    }
    .reveal .slide > img.stretch.quarto-figure-center,
    .reveal .slide > img.r-stretch.quarto-figure-center {
      display: block;
      margin-left: auto;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-left,
    .reveal .slide > img.r-stretch.quarto-figure-left  {
      display: block;
      margin-left: 0;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-right,
    .reveal .slide > img.r-stretch.quarto-figure-right  {
      display: block;
      margin-left: auto;
      margin-right: 0; 
    }
  </style>
<meta property="og:title" content="Biologia della Conservazione 23/24 - Biologia della Conservazione">
<meta property="og:description" content="Class 13 - Population Structure/Hybridization">
<meta property="og:site_name" content="Biologia della Conservazione 23/24">
<meta property="og:locale" content="en_US">
</head>
<body class="quarto-light">
  <div class="reveal">
    <div class="slides">

<section id="title-slide" class="quarto-title-block center">
  <h1 class="title">Biologia della Conservazione</h1>
  <p class="subtitle">Class 13 - Population Structure/Hybridization</p>

<div class="quarto-title-authors">
</div>

  <p class="date">2024-05-07</p>
</section>
<section class="slide level2">

<table>
<thead>
<tr class="header">
<th>Student</th>
<th>Title</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>FALVO</td>
<td><a href="https://drive.proton.me/urls/9WVQGE6J5M#GEndO0gEvVJq">Developing gene drive technologies to eradicate invasive rodents from islands</a></td>
</tr>
<tr class="even">
<td>D’ALESSANDRO</td>
<td><a href="https://drive.proton.me/urls/HQMV65ETB0#vB21xDrPrAXN">Population genetic analysis of the recently rediscovered hula painted frog (<em>Latonia nigriventer</em>) reveals high genetic diversity and low inbreeding</a></td>
</tr>
<tr class="odd">
<td>PIROLI</td>
<td><a href="https://drive.proton.me/urls/TD1F4DDC70#hzwxC3gFnM51">First evidence of recruitment in critically endangered Galapagos pink iguanas</a></td>
</tr>
<tr class="even">
<td>TRONCONE</td>
<td><a href="https://drive.proton.me/urls/SW4YPN2ZGC#RWYgHuXfrvK4">Genetic characterization and implications for conservation of the last autochthonous Mouflon population in Europe</a></td>
</tr>
<tr class="odd">
<td>GARCIA</td>
<td><a href="https://drive.proton.me/urls/FJ5FMEQ8EC#QhbaqVfIyFkE">The critically endangered vaquita is not doomed to extinction by inbreeding depression</a></td>
</tr>
</tbody>
</table>
</section>
<section id="topics" class="slide level2">
<h2>Topics</h2>
<div class="columns">
<div class="column" style="width:50%;">
<ol type="1">
<li><span style="color: green;">Introduction to conservation biology</span></li>
<li><span style="color: green;">Biodiversity</span></li>
<li><span style="color: green;">Biodiversity and extinctions</span></li>
<li><span style="color: green;">Populations</span></li>
<li><span style="color: green;">Basics of conservation genetics</span></li>
<li><span style="color: green;">Genetic drift</span></li>
<li><span style="color: green;">Effective population size</span></li>
<li><span style="color: green;">Inbreeding depression</span></li>
<li><span style="color: orange;">Population structure</span></li>
<li><span style="color: orange;">Hybridization</span></li>
<li><span style="color: blue;">Applying genetic information to conservation</span></li>
</ol>
</div><div class="column" style="width:50%;">
<ol start="12" type="1">
<li><span style="color: blue;">Managing genetic diversity in captive populations</span></li>
<li><span style="color: blue;">Managing genetic diversity in the wild</span></li>
<li><span style="color: blue;">Population viability analysis</span></li>
<li><span style="color: blue;">Invasive species</span></li>
<li><span style="color: blue;">CITES and trade regulations of endangered species</span></li>
<li><span style="color: blue;">Conservation and climate change</span></li>
<li><span style="color: blue;">Future climate impacts</span></li>
<li><span style="color: blue;">Conservation economics</span></li>
<li><span style="color: blue;">Conservation technology</span></li>
</ol>
</div>
</div>
</section>
<section id="population-structure" class="slide level2">
<h2>Population structure</h2>
<ul>
<li>In most species individuals are scattered in local conglomerates that are called <em>demes</em> (<em>i.e.</em>, local populations or subpopulations)</li>
</ul>
<div class="fragment">
<div class="columns">
<div class="column" style="width:50%;">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="./class_13_1.png"></p>
<figcaption>After <span class="citation" data-cites="Akakaya2007">Akçakaya, Mills, &amp; Doncaster (<a href="#/references" role="doc-biblioref" onclick="">2007</a>)</span></figcaption>
</figure>
</div>
</div><div class="column" style="width:50%;">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="./class_13_2.jpg"></p>
<figcaption>After <span class="citation" data-cites="Frankham2010">Frankham, Ballou, &amp; Briscoe (<a href="#/references" role="doc-biblioref" onclick="">2010</a>)</span></figcaption>
</figure>
</div>
</div>
</div>
</div>
</section>
<section id="population-structure-1" class="slide level2">
<h2>Population structure</h2>
<ul>
<li><p>A metapopulation is defined as a set of discrete populations (demes) of the same species, in the same general geographical area, potentially able to exchange individuals through migration, dispersal, or human-mediated movement <span class="citation" data-cites="Akakaya2007 Hanski1997">(<a href="#/references" role="doc-biblioref" onclick="">Akçakaya et al., 2007</a>; <a href="#/references" role="doc-biblioref" onclick="">Hanski &amp; Simberloff, 1997</a>)</span></p></li>
<li><p>This subdivision (or structure) of populations can be natural and/or human driven</p></li>
<li><p>Regardless of the cause, the existence of population structure in many species forces us to think at the genetic variation within a species at two primary levels:</p>
<ol type="1">
<li>Genetic variation within local demes.</li>
<li>Genetic variation bewteen local demes.</li>
</ol></li>
<li><p>Not all species are scattered!</p></li>
</ul>
</section>
<section id="population-structure-2" class="slide level2">
<h2>Population structure</h2>

<img data-src="./class_13_3.png" class="r-stretch quarto-figure-center"><p class="caption">After <span class="citation" data-cites="Allendorf2022">Allendorf et al. (<a href="#/references" role="doc-biblioref" onclick="">2022</a>)</span>, p.&nbsp;184</p></section>
<section id="population-structure-3" class="slide level2">
<h2>Population structure</h2>

<img data-src="./class_13_4.png" class="r-stretch quarto-figure-center"><p class="caption">Boreal forest distibution in North America. Available <a href="https://canadiangeographic.ca/articles/new-framework-identifies-climate-change-refugia-in-boreal-forest/">here</a></p></section>
<section id="population-structure-4" class="slide level2">
<h2>Population structure</h2>
<ul>
<li>Even in the absence of barriers to gene flow, genetic differences can accrue so that individuals that are very distant from each other may present significant differences at the genetic level: isolation by distance</li>
</ul>

<img data-src="./class_13_5.png" class="r-stretch quarto-figure-center"><p class="caption">Variability in the correlation between species genetic diversity and geographic distance. ranges. Modified from <span class="citation" data-cites="Forbes1999">Forbes &amp; Hogg (<a href="#/references" role="doc-biblioref" onclick="">1999</a>)</span> with additional unpublished data from M.K. Schwartz <span class="citation" data-cites="Allendorf2022">(<a href="#/references" role="doc-biblioref" onclick="">Allendorf et al., 2022</a>)</span>.</p></section>
<section id="population-structure-5" class="slide level2">
<h2>Population structure</h2>
<ul>
<li>Describing the patterns and amount of differentiation among demes is of the uttermost importance to develop effective conservation strategies <span class="citation" data-cites="Thatte2018">(see for example <a href="#/references" role="doc-biblioref" onclick="">Thatte, Joshi, Vaidyanathan, Landguth, &amp; Ramakrishnan, 2018</a>)</span>.</li>
</ul>

<img data-src="./class_13_6.png" class="r-stretch quarto-figure-center"><p class="caption">Study area in <span class="citation" data-cites="Thatte2018">Thatte et al. (<a href="#/references" role="doc-biblioref" onclick="">2018</a>)</span>. Genetically identified tigers as red dots</p></section>
<section id="population-structure-6" class="slide level2">
<h2>Population structure</h2>
<ul>
<li><span class="citation" data-cites="Frankham2011">Frankham et al. (<a href="#/references" role="doc-biblioref" onclick="">2011</a>)</span> describe the risks of outbreeding depression when translocating individuals from genetically distinct populations to supplement imperiled ones.</li>
</ul>
<div class="fragment">
<div class="columns">
<div class="column" style="width:50%;">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="./class_13_7.png" style="width:85.0%"></p>
<figcaption>After <span class="citation" data-cites="Welch2017">Welch et al. (<a href="#/references" role="doc-biblioref" onclick="">2017</a>)</span>.</figcaption>
</figure>
</div>
</div><div class="column" style="width:50%;">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="./class_13_8.png" style="width:83.0%"></p>
<figcaption>After <span class="citation" data-cites="Welch2017">Welch et al. (<a href="#/references" role="doc-biblioref" onclick="">2017</a>)</span>.</figcaption>
</figure>
</div>
</div>
</div>
</div>
</section>
<section id="population-structure-7" class="slide level2">
<h2>Population structure</h2>

<img data-src="./class_13_9.png" class="r-stretch quarto-figure-center"><p class="caption">After <span class="citation" data-cites="Colosimo2021b">Colosimo et al. (<a href="#/references" role="doc-biblioref" onclick="">2021</a>)</span></p></section>
<section id="population-structure-8" class="slide level2">
<h2>Population structure</h2>
<ul>
<li>Developing priorities for the conservation of a species often requires an understanding of adaptive genetic differentiation among populations <span class="citation" data-cites="Allendorf2022">(<a href="#/references" role="doc-biblioref" onclick="">Allendorf et al., 2022</a>)</span>.</li>
<li>To understand how genetic variation is distributed at different loci within subdivided populations it is necessary to look at different interacting forces like gene flow, genetic drift and natural selection.</li>
<li><span class="math inline">\(F\)</span> statistics <span class="citation" data-cites="Wright1931 Wright1951 Malecot1948">(<a href="#/references" role="doc-biblioref" onclick="">Malécot, 1948</a>; <a href="#/references" role="doc-biblioref" onclick="">Wright, 1931</a>, <a href="#/references" role="doc-biblioref" onclick="">1951</a>)</span> are undoubtedly the most used indexes to describe genetic differentiation in and between demes.</li>
<li><span class="math inline">\(F\)</span> statistics are, in short, inbreeding coefficients measuring different types of inbreeding at different levels.</li>
</ul>
</section>
<section id="f-statistics" class="slide level2">
<h2><span class="math inline">\(F\)</span> statistics</h2>
<ul>
<li>We recognize three primary coefficients: <span class="math inline">\(F_{IS}\)</span>, <span class="math inline">\(F_{IT}\)</span>, and <span class="math inline">\(F_{ST}\)</span>.</li>
<li><span class="math inline">\(F_{IS}\)</span> measures Hardy–Weinberg departures within local demes.</li>
<li><span class="math inline">\(F_{ST}\)</span> measures divergence in allele frequencies among demes.</li>
<li><span class="math inline">\(F_{IT}\)</span> measures Hardy–Weinberg departures in the metapopulation due to <span class="math inline">\(F_{IS}\)</span> (nonrandom mating within local demes) and <span class="math inline">\(F_{ST}\)</span> (allele frequency divergence among demes).</li>
<li>The original <span class="math inline">\(F\)</span> statistics developed by wright focused on loci with only two alleles.</li>
<li><span class="citation" data-cites="Nei1977">Nei (<a href="#/references" role="doc-biblioref" onclick="">1977</a>)</span> extended these measures to multiple alleles and used a different notation to name what he called <em>analysis of gene diversity</em>: <span class="math inline">\(G_{IS}\)</span>, <span class="math inline">\(G_{IT}\)</span>, and <span class="math inline">\(G_{ST}\)</span>. Often the two groups of indexes are used interchangeably.</li>
</ul>
</section>
<section id="f_is" class="slide level2">
<h2><span class="math inline">\(F_{IS}\)</span></h2>
<p>This is a measure of departure from HW proportions within local demes.</p>
<p><span class="math display">\[
\label{fis}
\tag{13.1}
F_{IS} = 1-\frac{H_O}{H_S}
\]</span></p>
<p>where <span class="math inline">\(H_O\)</span> is the observed heterozygosity averaged over all demes while <span class="math inline">\(H_S\)</span> is the expected heterozygosity averaged over all demes. <span class="math inline">\(F_{IS}\)</span> values can range from <span class="math inline">\(-1 \leq 0 \leq 1\)</span>, with positive values indicating an excess of homozygotes and negative values indicating an excess of heterozygotes <span class="citation" data-cites="Allendorf2022">(<a href="#/references" role="doc-biblioref" onclick="">Allendorf et al., 2022</a>)</span>.</p>
</section>
<section id="f_st" class="slide level2">
<h2><span class="math inline">\(F_{ST}\)</span></h2>
<p>Often called <em>fixation index</em>, it is a measure of genetic divergence among demes.</p>
<p><span class="math display">\[
\label{fst}
\tag{13.2}
F_{ST} = 1-\frac{H_S}{H_T}
\]</span></p>
<p>with <span class="math inline">\(H_T\)</span> as the expected HW heterozygosity if the metapopulation were panmictic, and <span class="math inline">\(H_S\)</span> is the HW proportion of heterozygotes in each separate deme and then averaged over all demes. This measure varies from <span class="math inline">\(0\)</span> (demes all have equal allele frequencies) to <span class="math inline">\(1\)</span> (demes are all fixed for different alleles).</p>
</section>
<section id="f_it" class="slide level2">
<h2><span class="math inline">\(F_{IT}\)</span></h2>
<p>Measures the departure from HW proportions due to departures from HW proportions within local demes and divergence among demes <span class="citation" data-cites="Allendorf2022">(<a href="#/references" role="doc-biblioref" onclick="">Allendorf et al., 2022</a>)</span>.</p>
<p><span class="math display">\[
\label{fit}
\tag{13.3}
F_{IT} = 1-\frac{H_O}{H_T}
\]</span></p>
<p>where <span class="math inline">\(H_O\)</span> is the observed heterozygosity averaged over all demes while <span class="math inline">\(H_T\)</span> is the expected HW heterozygosity if the metapopulation were panmictic.</p>
<div class="fragment">
<ul>
<li>The fixation indexes described in the previous slides are related by the following expression:</li>
</ul>
<p><span class="math display">\[
\label{findexrelation}
\tag{13.4}
F_{IT}= F_{IS}+F_{ST}-(F_{IS}*F_{ST})
\]</span></p>
</div>
</section>
<section id="the-wahlund-effect" class="slide level2">
<h2>The Wahlund effect</h2>
<ul>
<li>A deficit of heterozygosity due to the presence of differential distribution of genetic diversity among demes is often called Wahlund effect, for the the Swedish geneticist who described it <span class="citation" data-cites="Wahlund1928">(<a href="#/references" role="doc-biblioref" onclick="">Wahlund, 1928</a>)</span>.</li>
<li>He analyzed the excess of homozygotes and deficit of heterozygotes in terms of the variance of allele frequencies among S deme <span class="citation" data-cites="Allendorf2022">(<a href="#/references" role="doc-biblioref" onclick="">Allendorf et al., 2022</a>)</span>.</li>
</ul>
<p><span class="math display">\[
\label{wahlund}
\tag{13.5}
Var(q)=\frac{1}{S}\sum(q_i-\bar{q})^2 \\ \\ F_{ST} = \frac{Var(q)}{pq}
\]</span></p>
</section>
<section id="population-structure-and-habitat-fragmentation" class="slide level2">
<h2>Population structure and habitat fragmentation</h2>
<ul>
<li>Habitat fragmentation is a combination of two processes: reduction in total area, and the creation of separate islands.</li>
<li>Human induced fragmentation and habitat loss is recognized as the main cause of biodiversity loss <span class="citation" data-cites="deBaan2013">(<a href="#/references" role="doc-biblioref" onclick="">Baan, Alkemade, &amp; Koellner, 2013</a>)</span>.</li>
<li>Habitat fragmentation causes an overall reduction in population size.</li>
<li>Habitat fragmentation also causes an overall reduction in gene flow among the patches it creates.</li>
<li>Inhospitable, cleared habitat among fragments typically inhibits gene flow. The same goes for roads rivers, water bodies, mountains, or any other artificial or natural barrier <span class="citation" data-cites="Colosimo2014">Riley et al. (<a href="#/references" role="doc-biblioref" onclick="">2006</a>)</span>.</li>
<li>With restricted gene flow fragmentation is highly deleterious in the long term.</li>
<li>Single Large or Several Small <span class="citation" data-cites="Frankham2010">(<a href="#/references" role="doc-biblioref" onclick="">Frankham et al., 2010</a>)</span>.</li>
</ul>
</section>
<section id="population-structure-and-habitat-fragmentation-1" class="slide level2">
<h2>Population structure and habitat fragmentation</h2>

<img data-src="./class_13_10.png" class="r-stretch quarto-figure-center"><p class="caption">Fragmentation of Atlantic forest. After <span class="citation" data-cites="Luci2011">Nunes (<a href="#/references" role="doc-biblioref" onclick="">2011</a>)</span></p></section>
<section id="population-structure-and-habitat-fragmentation-2" class="slide level2">
<h2>Population structure and habitat fragmentation</h2>

<img data-src="./class_13_11.jpg" class="r-stretch quarto-figure-center"><p class="caption">Single large (SL) VS several small (SS). After <span class="citation" data-cites="Frankham2010">Frankham et al. (<a href="#/references" role="doc-biblioref" onclick="">2010</a>)</span></p></section>
<section id="population-structure-and-habitat-fragmentation-3" class="slide level2">
<h2>Population structure and habitat fragmentation</h2>
<p>Fragmentation happens in two steps:</p>
<ol type="1">
<li>Initial genetic subdivision of the population.</li>
<li>Cumulative diversification through genetic drift, selection, inbreeding etc. etc. etc.</li>
</ol>
<p>After an initial fragmentation, different fragments will have different initial allele frequencies. This diversification can be described by the binomial sampling variance (<span class="math inline">\(\sigma_p^2 = \frac{pq}{2N}\)</span>) and the smaller the fragment, the higher the variance.</p>
<div class="fragment">
<p>The degree of differentiation among fragments can be described by partitioning the overall inbreeding into components within and among populations (<span class="math inline">\(F\)</span> statistics).</p>
</div>
<div class="fragment">
<p>We will look at some examples on estimating <span class="math inline">\(F\)</span> statistics from genotype data.</p>
</div>
</section>
<section id="population-structure-and-habitat-fragmentation-4" class="slide level2">
<h2>Population structure and habitat fragmentation</h2>
<h3 id="gene-flow">Gene flow</h3>
<p>What is gene flow?</p>
<ul>
<li>Movement of alleles between population due to migration.</li>
</ul>
<p>The effects of fragmentation on gene flow depend on:</p>
<ul>
<li>Number of population fragments.</li>
<li>Distribution of population sizes in the fragments.</li>
<li>Distance between fragments.</li>
<li>Dispersal ability of the species.</li>
<li>Migration rates among fragments.</li>
<li>Environment of the matrix among fragments.</li>
<li>Times since fragmentation and extinction and recolonization rates across fragments.</li>
</ul>
</section>
<section id="the-black-footed-rock-wallaby" class="slide level2">
<h2>The black footed rock wallaby</h2>
<div class="columns">
<div class="column" style="width:50%;">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="./class_13_12.jpg" class="quarto-figure quarto-figure-center" style="width:20.0%"></p>
</figure>
</div>
</div><div class="column" style="width:50%;">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="./class_13_14.jpg" class="quarto-figure quarto-figure-center" style="width:50.0%"></p>
</figure>
</div>
</div>
</div>
<table>
<colgroup>
<col style="width: 24%">
<col style="width: 26%">
<col style="width: 24%">
<col style="width: 24%">
</colgroup>
<thead>
<tr class="header">
<th>Location</th>
<th>Prop. of polymorphic loci</th>
<th>Mean # of alleles</th>
<th>Average heterozygosity</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Barrow Island (1)</td>
<td>0.1</td>
<td>1.2</td>
<td>0.05</td>
</tr>
<tr class="even">
<td>Exmouth (2)</td>
<td>1.0</td>
<td>3.4</td>
<td>0.62</td>
</tr>
<tr class="odd">
<td>Wheatbelt (3)</td>
<td>1.0</td>
<td>4.4</td>
<td>0.56</td>
</tr>
</tbody>
</table>
<ul>
<li>Data from <span class="citation" data-cites="Eldridge1999">Eldridge et al. (<a href="#/references" role="doc-biblioref" onclick="">1999</a>)</span>, images after <span class="citation" data-cites="Frankham2010">Frankham et al. (<a href="#/references" role="doc-biblioref" onclick="">2010</a>)</span>.</li>
</ul>
</section>
<section id="example---1" class="slide level2">
<h2>Example - 1</h2>
<ul>
<li>Populations with same allele frequencies. One with random mating the other one with inbreeding.</li>
</ul>
<table>
<colgroup>
<col style="width: 18%">
<col style="width: 15%">
<col style="width: 15%">
<col style="width: 15%">
<col style="width: 20%">
<col style="width: 7%">
<col style="width: 7%">
</colgroup>
<thead>
<tr class="header">
<th>Population</th>
<th><span class="math inline">\(A_1A_1\)</span></th>
<th><span class="math inline">\(A_1A_2\)</span></th>
<th><span class="math inline">\(A_2A_2\)</span></th>
<th>Allele Freq</th>
<th>F</th>
<th>He</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>1</td>
<td>0.25</td>
<td>0.5</td>
<td>0.25</td>
<td><span class="math inline">\(p=0.5\)</span></td>
<td>0</td>
<td>0.5</td>
</tr>
<tr class="even">
<td></td>
<td></td>
<td></td>
<td></td>
<td><span class="math inline">\(q=0.5\)</span></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td>2</td>
<td>0.4</td>
<td>0.2</td>
<td>0.4</td>
<td><span class="math inline">\(p=0.5\)</span></td>
<td>0.6</td>
<td>0.5</td>
</tr>
<tr class="even">
<td></td>
<td></td>
<td></td>
<td></td>
<td><span class="math inline">\(q=0.5\)</span></td>
<td></td>
<td></td>
</tr>
</tbody>
</table>
<ul>
<li><span class="math inline">\(H_O\)</span> observed heterozygosity averaged over all demes.</li>
<li><span class="math inline">\(H_S\)</span> expected heterozygosity averaged over all demes.</li>
<li><span class="math inline">\(H_T\)</span> expected heterozygosity of the metapopulation.</li>
<li><span class="math inline">\(F_{IS} = 1-\frac{H_O}{H_S}\)</span>; <span class="math inline">\(F_{IT} = 1-\frac{H_O}{H_T}\)</span>; <span class="math inline">\(F_{ST} = 1-\frac{H_S}{H_T}\)</span>.</li>
</ul>
</section>
<section id="example---2" class="slide level2">
<h2>Example - 2</h2>
<ul>
<li>Random mating populations with different allele frequencies.</li>
</ul>
<table>
<colgroup>
<col style="width: 18%">
<col style="width: 15%">
<col style="width: 15%">
<col style="width: 15%">
<col style="width: 19%">
<col style="width: 7%">
<col style="width: 9%">
</colgroup>
<thead>
<tr class="header">
<th>Population</th>
<th><span class="math inline">\(A_1A_1\)</span></th>
<th><span class="math inline">\(A_1A_2\)</span></th>
<th><span class="math inline">\(A_2A_2\)</span></th>
<th>Allele Freq</th>
<th>F</th>
<th>He</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>1</td>
<td>0.25</td>
<td>0.5</td>
<td>0.25</td>
<td><span class="math inline">\(p=0.5\)</span></td>
<td>0</td>
<td>0.5</td>
</tr>
<tr class="even">
<td></td>
<td></td>
<td></td>
<td></td>
<td><span class="math inline">\(q=0.5\)</span></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td>2</td>
<td>0.04</td>
<td>0.32</td>
<td>0.64</td>
<td><span class="math inline">\(p=0.2\)</span></td>
<td>0</td>
<td>0.32</td>
</tr>
<tr class="even">
<td></td>
<td></td>
<td></td>
<td></td>
<td><span class="math inline">\(q=0.8\)</span></td>
<td></td>
<td></td>
</tr>
</tbody>
</table>
<ul>
<li><span class="math inline">\(H_O\)</span> observed heterozygosity averaged over all demes.</li>
<li><span class="math inline">\(H_S\)</span> expected heterozygosity averaged over all demes.</li>
<li><span class="math inline">\(H_T\)</span> expected heterozygosity of the metapopulation.</li>
<li><span class="math inline">\(F_{IS} = 1-\frac{H_O}{H_S}\)</span>; <span class="math inline">\(F_{IT} = 1-\frac{H_O}{H_T}\)</span>; <span class="math inline">\(F_{ST} = 1-\frac{H_S}{H_T}\)</span>.</li>
</ul>
</section>
<section id="example---3" class="slide level2">
<h2>Example - 3</h2>
<ul>
<li>Populations with different allele frequencies, one random mating and one with inbreeding.</li>
</ul>
<table>
<colgroup>
<col style="width: 18%">
<col style="width: 15%">
<col style="width: 15%">
<col style="width: 15%">
<col style="width: 19%">
<col style="width: 7%">
<col style="width: 9%">
</colgroup>
<thead>
<tr class="header">
<th>Population</th>
<th><span class="math inline">\(A_1A_1\)</span></th>
<th><span class="math inline">\(A_1A_2\)</span></th>
<th><span class="math inline">\(A_2A_2\)</span></th>
<th>Allele Freq</th>
<th>F</th>
<th>He</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>1</td>
<td>0.25</td>
<td>0.5</td>
<td>0.25</td>
<td><span class="math inline">\(p=0.5\)</span></td>
<td>0</td>
<td>0.5</td>
</tr>
<tr class="even">
<td></td>
<td></td>
<td></td>
<td></td>
<td><span class="math inline">\(q=0.5\)</span></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td>2</td>
<td>0.14</td>
<td>0.13</td>
<td>0.74</td>
<td><span class="math inline">\(p=0.2\)</span></td>
<td>0.6</td>
<td>0.32</td>
</tr>
<tr class="even">
<td></td>
<td></td>
<td></td>
<td></td>
<td><span class="math inline">\(q=0.8\)</span></td>
<td></td>
<td></td>
</tr>
</tbody>
</table>
<ul>
<li><span class="math inline">\(H_O\)</span> observed heterozygosity averaged over all demes.</li>
<li><span class="math inline">\(H_S\)</span> expected heterozygosity averaged over all demes.</li>
<li><span class="math inline">\(H_T\)</span> expected heterozygosity of the metapopulation.</li>
<li><span class="math inline">\(F_{IS} = 1-\frac{H_O}{H_S}\)</span>; <span class="math inline">\(F_{IT} = 1-\frac{H_O}{H_T}\)</span>; <span class="math inline">\(F_{ST} = 1-\frac{H_S}{H_T}\)</span>.</li>
</ul>
</section>
<section id="example---4" class="slide level2">
<h2>Example - 4</h2>

<img data-src="./class_13_15.jpg" class="r-stretch quarto-figure-center"><p class="caption"><span class="math inline">\(F_{ST}\)</span> increases over generations in fragmented populations at a rate inversely dependent on population size. Values that are <span class="math inline">\(\ge 0.15\)</span> indicate significant differentiation between fragments <span class="citation" data-cites="Frankham2010">(<a href="#/references" role="doc-biblioref" onclick="">Frankham et al., 2010</a>)</span>.</p></section>
<section id="migration-and-gene-flow" class="slide level2">
<h2>Migration and gene flow</h2>
<ul>
<li>Migration reduces the impact of fragmentation in a way that is dependent on the rate of gene flow.</li>
<li>When is gene flow sufficiently high to prevent the genetic impact of fragmentation?</li>
</ul>
<div class="fragment">
<div class="columns">
<div class="column" style="width:50%;">
<ul>
<li>A single migrant per generation is considered sufficient to prevent differentiation between demes.</li>
<li>The migration rate <span class="math inline">\(m\)</span> is the proportion of genes in a population derived from migrants per generation.</li>
<li><span class="math inline">\(Nm\)</span> is the number of migrants per generation <span class="citation" data-cites="Frankham2010">(<a href="#/references" role="doc-biblioref" onclick="">Frankham et al., 2010</a>)</span>.</li>
</ul>
</div><div class="column" style="width:50%;">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="./class_13_16.jpg" class="quarto-figure quarto-figure-center" style="width:65.0%"></p>
</figure>
</div>
</div>
</div>
</div>
</section>
<section id="migration-and-gene-flow-1" class="slide level2">
<h2>Migration and gene flow</h2>
<ul>
<li>In reality, more than one migrant per generation is needed as not all migrants are successful at reproduction.</li>
<li>Different values have been proposed over time: 5, 1-10, <span class="math inline">\(\ge\)</span> 10 <span class="citation" data-cites="Mills1996 Lacy1987 Vucetich2000">(<a href="#/references" role="doc-biblioref" onclick="">Lacy, 1987</a>; <a href="#/references" role="doc-biblioref" onclick="">Mills &amp; Allendorf, 1996</a>; <a href="#/references" role="doc-biblioref" onclick="">Vucetich &amp; Waite, 2000</a>)</span>.</li>
<li>One way to overcome the issue of non idealized populations is to look at <span class="math inline">\(N_em\)</span>.</li>
<li>When population size and migration rates are constant inbreeding and migration reach an equilibrium, measured by <span class="math inline">\(F_{ST}\)</span></li>
</ul>
<p><span class="math display">\[
\label{fst2}
\tag{13.6}
F_{ST} = \frac{1}{4N_em+1}
\]</span></p>
<ul>
<li>Equation <span class="math inline">\(\ref{fst2}\)</span> applies when <span class="math inline">\(m\)</span> is small and all population sizes and migration rates among them are the same.</li>
</ul>
</section>
<section id="migration-and-gene-flow-2" class="slide level2">
<h2>Migration and gene flow</h2>
<ul>
<li>Measuring migration rates is complicated!</li>
<li>It is possible to rearrange equation <span class="math inline">\(\ref{fst}\)</span> to estimate <span class="math inline">\(Nm\)</span>, and this is an approach widely used. Unfortunately this approach will lead to biased results especially when different models of migration than the island model are considered.</li>
</ul>

<img data-src="./class_13_17.jpg" class="r-stretch quarto-figure-center"><p class="caption">Migration models, after <span class="citation" data-cites="Frankham2010">Frankham et al. (<a href="#/references" role="doc-biblioref" onclick="">2010</a>)</span></p></section>
<section id="limitations-of-f_st" class="slide level2">
<h2>Limitations of <span class="math inline">\(F_{ST}\)</span></h2>
<ul>
<li>This measure of differentiation between demes does not take in to account genealogical information (i.e., there is no information about the identity of the alleles).</li>
<li>A measure related to <span class="math inline">\(F_{ST}\)</span>, called <span class="math inline">\(R_{ST}\)</span>, uses information on the length of alleles at microsatellite, and assumes that each mutation changes an allele’s length by only one repeat unit <span class="citation" data-cites="Slatkin1995 Allendorf2022">(<a href="#/references" role="doc-biblioref" onclick="">Allendorf et al., 2022</a>; <a href="#/references" role="doc-biblioref" onclick="">Slatkin, 1995</a>)</span>.</li>
<li>Another measure of differentiation that uses information on allele genealogical relationships is <span class="math inline">\(\Phi_{ST}\)</span> <span class="citation" data-cites="Excoffier1992 Allendorf2022">(<a href="#/references" role="doc-biblioref" onclick="">Allendorf et al., 2022</a>; <a href="#/references" role="doc-biblioref" onclick="">Excoffier, Smouse, &amp; Quattro, 1992</a>)</span>.</li>
<li>One more alternative is offered by <span class="citation" data-cites="Gaggiotti2010">Gaggiotti &amp; Foll (<a href="#/references" role="doc-biblioref" onclick="">2010</a>)</span>. They have presented a method to estimate population-specific <span class="math inline">\(F_{ST}\)</span> values (<span class="math inline">\(F_{ST}^′\)</span>) rather than global or pairwise <span class="math inline">\(F_{ST}\)</span> values. They define <span class="math inline">\(F_{ST}^′\)</span> s as the probability that two genes chosen at random from the population share a common ancestor within that population <span class="citation" data-cites="Allendorf2022">(<a href="#/references" role="doc-biblioref" onclick="">Allendorf et al., 2022</a>)</span>.</li>
</ul>
</section>
<section id="limitations-of-f_st-1" class="slide level2">
<h2>Limitations of <span class="math inline">\(F_{ST}\)</span></h2>
<ul>
<li><p><span class="math inline">\(F_{ST}\)</span> (and its analog <span class="math inline">\(G_{ST}\)</span>) has limitations when using loci with high mutation rates and high heterozygosities, such as microsatellites. <span class="math inline">\(F_{ST}\)</span> is biased downward when variation within subpopulations (<span class="math inline">\(H_S\)</span>) is high. The source of this bias is obvious: when variation within populations is high, the proportion of the total variation distributed between populations can never be very high <span class="citation" data-cites="Allendorf2022">(<a href="#/references" role="doc-biblioref" onclick="">Allendorf et al., 2022</a>)</span>.</p>
<ul>
<li><span class="citation" data-cites="Hedrick2005">Hedrick (<a href="#/references" role="doc-biblioref" onclick="">2005</a>)</span> introduced <span class="math inline">\(G_{ST}^′\)</span>, which is <span class="math inline">\(G_{ST}\)</span> divided by its maximum possible values with the same overall allele frequencies. Thus, <span class="math inline">\(G_{ST}^′\)</span> has a range from 0 to 1, and was designed to be independent of <span class="math inline">\(H_S\)</span>. If <span class="math inline">\(H_S\)</span> is high, then <span class="math inline">\(G_{ST}^′\)</span> can be much greater than <span class="math inline">\(G_{ST}\)</span>. <span class="math inline">\(G_{ST}^′\)</span> is designed to be a standardized measure of <span class="math inline">\(G_{ST}\)</span>, which accounts for different levels of total genetic variation at different loci <span class="citation" data-cites="Allendorf2022">(<a href="#/references" role="doc-biblioref" onclick="">Allendorf et al., 2022</a>)</span>.</li>
</ul></li>
</ul>
</section>
<section id="hierarchical-population-structure" class="slide level2">
<h2>Hierarchical population structure</h2>
<ul>
<li>Populations are often structured at multiple hierarchical levels; for example, locally and regionally.</li>
</ul>

<img data-src="./class_13_18.png" class="r-stretch quarto-figure-center"><p class="caption">Organization of hierarchical population structure with two levels of subdivision: subpopulations within regions (<span class="math inline">\(F_{SR}\)</span>) and regions within the total species (<span class="math inline">\(F_{RT}\)</span>). Each region has two subpopulations. <span class="math inline">\(F_{SR}\)</span> is the proportion of the total diversity due to differences between subpopulations within regions. <span class="math inline">\(F_{RT}\)</span> is the proportion of the total diversity due to differences between regions <span class="citation" data-cites="Allendorf2022">(<a href="#/references" role="doc-biblioref" onclick="">Allendorf et al., 2022</a>)</span>.</p></section>
<section id="alternative-approach-to-estimating-nm" class="slide level2">
<h2>Alternative approach to estimating <span class="math inline">\(Nm\)</span></h2>
<ul>
<li>Another indirect estimator of <span class="math inline">\(Nm\)</span> is the private allele method <span class="citation" data-cites="Slatkin1987">(<a href="#/references" role="doc-biblioref" onclick="">Slatkin, 1987</a>)</span>.</li>
<li>A private allele is one found in only one population. Slatkin showed that a linear relationship exists between <span class="math inline">\(Nm\)</span> and the average frequency of private alleles. This method works because if gene flow (<span class="math inline">\(Nm\)</span>) is low, populations will have numerous private alleles that arise through mutation, for example.</li>
<li>The time during which a new allele remains private depends only on migration rates, such that the proportion of alleles that are private decreases as migration rate increases. If gene flow is high, private alleles will be uncommon <span class="citation" data-cites="Allendorf2022">(<a href="#/references" role="doc-biblioref" onclick="">Allendorf et al., 2022</a>)</span>.</li>
<li>This method could be less biased than the <span class="math inline">\(F_{ST}\)</span>-island model method, when using highly polymorphic markers, because it apparently is less sensitive to problems of homoplasy created by back mutations than is the <span class="math inline">\(F_{ST}\)</span> method <span class="citation" data-cites="Allen1995">(<a href="#/references" role="doc-biblioref" onclick="">Allen, Amos, Pomeroy, &amp; Twiss, 1995</a>)</span>.</li>
</ul>
</section>
<section id="assignment-tests" class="slide level2">
<h2>Assignment tests</h2>
<div class="columns">
<div class="column" style="width:50%;">
<ul>
<li><p>Assignment tests offer an attractive alternative to the traditional capture–mark–recapture approach to estimating <span class="math inline">\(Nm\)</span> directly.</p></li>
<li><p>For example, we can genotype many individuals in a single population sample, and then determine the proportion of “immigrant” individuals; that is, individuals with a foreign genotype that are unlikely to have originated locally.</p></li>
</ul>
</div><div class="column" style="width:50%;">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="./class_13_19.png"></p>
<figcaption>After <span class="citation" data-cites="Allendorf2022">Allendorf et al. (<a href="#/references" role="doc-biblioref" onclick="">2022</a>)</span></figcaption>
</figure>
</div>
</div>
</div>
</section>
<section id="hybridization" class="slide level2">
<h2>Hybridization</h2>
<ul>
<li><p>Rates of hybridization and introgression have increased dramatically worldwide because of widespread intentional and incidental translocations of organisms and habitat modifications by humans <span class="citation" data-cites="Allendorf2022">(<a href="#/references" role="doc-biblioref" onclick="">Allendorf et al., 2022</a>)</span>.</p></li>
<li><p>Rates of hybridization and introgression have increased dramatically worldwide because of widespread intentional and incidental translocations of organisms and habitat modifications by humans <span class="citation" data-cites="Allendorf2022">(<a href="#/references" role="doc-biblioref" onclick="">Allendorf et al., 2022</a>)</span>.</p></li>
<li><p>The increasing pace of the three interacting human activities that contribute most to increased rates of hybridization (introductions of plants and animals, fragmentation, and habitat modification) suggests that this problem will become even more serious in the future <span class="citation" data-cites="Allendorf2022">(<a href="#/references" role="doc-biblioref" onclick="">Allendorf et al., 2022</a>)</span></p></li>
<li><p>For example, increased turbidity in Lake Victoria, Africa, has reduced color perception of cichlid fishes and has interfered with the mate choice that produced reproductive isolation among species <span class="citation" data-cites="Seehausen1997 Allendorf2022">(<a href="#/references" role="doc-biblioref" onclick="">Allendorf et al., 2022</a>; <a href="#/references" role="doc-biblioref" onclick="">Seehausen, Alphen, &amp; Witte, 1997</a>)</span></p></li>
</ul>
</section>
<section id="hybridization-1" class="slide level2">
<h2>Hybridization</h2>
<ul>
<li><p>On the other hand, hybridization is a natural part of the evolutionary process <span class="citation" data-cites="Allendorf2022">(<a href="#/references" role="doc-biblioref" onclick="">Allendorf et al., 2022</a>)</span>.</p></li>
<li><p>Hybridization has long been recognized as playing an important role in the evolution of plants.</p></li>
<li><p>In addition, recent studies have found that hybridization has also played an important role in the evolution of animals</p></li>
<li><p>Many early conservation policies generally did not allow protection of hybrids. However, increased appreciation of the important role of hybridization as an evolutionary process has caused a re-evaluation of these policies. Determining whether hybridization is natural or anthropogenic is crucial for conservation, but is often difficult <span class="citation" data-cites="Allendorf2022">(<a href="#/references" role="doc-biblioref" onclick="">Allendorf et al., 2022</a>)</span>.</p></li>
</ul>
</section>
<section id="hybridization-2" class="slide level2">
<h2>Hybridization</h2>

<img data-src="./class_13_20.png" class="r-stretch quarto-figure-center"><p class="caption">Microsatellite gel image showing genotype profiles (locus Lc106) for 10 lynx, 10 bobcat, and three putative hybrids. After <span class="citation" data-cites="Schwartz2004">Schwartz et al. (<a href="#/references" role="doc-biblioref" onclick="">2004</a>)</span></p></section>
<section id="hybridization-3" class="slide level2">
<h2>Hybridization</h2>

<img data-src="./class_13_21.png" class="r-stretch quarto-figure-center"><p class="caption">Which is which? Photo credit: G. Colosimo.</p></section>
<section id="hybridization-4" class="slide level2">
<h2>Hybridization</h2>

<img data-src="./class_13_22.png" class="r-stretch quarto-figure-center"><p class="caption">Demographic (a) and genetic (b) swamping. After <span class="citation" data-cites="Allendorf2022">Allendorf et al. (<a href="#/references" role="doc-biblioref" onclick="">2022</a>)</span></p></section>
<section class="slide level2">

<p>How can we use all the genetic data we have learnt to gather and analyse?</p>
<p><br></p>
<div class="fragment">
<ol type="1">
<li>Clarification of relatedness, taxonomy and phylogeny among populations.</li>
</ol>
</div>
<div class="fragment">
<ol start="2" type="1">
<li>Determination of population management units based on genetic criteria.</li>
</ol>
</div>
<div class="fragment">
<ol start="3" type="1">
<li>Estimation of gene flow and dispersal among populations.</li>
</ol>
</div>
<div class="fragment">
<ol start="4" type="1">
<li>Determination of the time since past genetic bottlenecks in a population or the time since significant differentiation between populations.</li>
</ol>
</div>
<div class="fragment">
<ol start="5" type="1">
<li>Understanding patterns of reproductive ecology.</li>
</ol>
</div>
<div class="fragment">
<ol start="6" type="1">
<li>Locating original sources of wildlife products.</li>
</ol>
</div>
</section>
<section id="clarification-of-relatedness-taxonomy-and-phylogeny-among-populations." class="slide level2">
<h2>1. Clarification of relatedness, taxonomy and phylogeny among populations.</h2>
<ul>
<li>The most basic question in conservation biology: how many species are there?</li>
<li>Assessing the taxonomy of one population or species still remains one of the most important applications of conservation genetics.</li>
<li>Until 1970 various taxonomic units were classified using morphological features.</li>
<li>In 1973 the Endangered Species Act (ESA, US Gov.) extended legal protection to species and subspecies.</li>
</ul>
<blockquote>
<p>When taxonomic distinction became the basis for legal protection…the innocence was lost forever <span class="citation" data-cites="Obrien1996">(<a href="#/references" role="doc-biblioref" onclick="">O’Brien, 1996</a>)</span>.</p>
</blockquote>
</section>
<section id="determination-of-population-management-units-based-on-genetic-criteria." class="slide level2">
<h2>2. Determination of population management units based on genetic criteria.</h2>
<ul>
<li><p>On what basis we delimit meaningful management units in the wild?</p></li>
<li><p>The genetic differences as measured by the <span class="math inline">\(F_{ST}\)</span> statistics provide an objective criterion for defining population management units.</p></li>
<li><p>Combining nuclear and mtDNA markers it was possible to identify 4 demographically autonomous population units, a northeastern India unit, a central India unit, and 2 genetically distinct unit in the south of India to be administrered separately because of their genetic uniqueness. These units were radically different from what had been proposed solely on the political considerations <span class="citation" data-cites="Vidya2005">(<a href="#/references" role="doc-biblioref" onclick="">Vidya, Fernando, Melnick, &amp; Sukumar, 2005</a>)</span>.</p></li>
</ul>
</section>
<section id="estimation-of-gene-flow-and-dispersal-among-populations." class="slide level2">
<h2>3. Estimation of gene flow and dispersal among populations.</h2>
<ul>
<li><p>Managing gene flow between populations, through a detailed analysis of <span class="math inline">\(F_{ST}\)</span> patterns, can help to better understand and define the individuals that could be used in translocation projects, when to do so, and the duration of the whole translocation process <span class="citation" data-cites="Allendorf2022">(<a href="#/references" role="doc-biblioref" onclick="">Allendorf et al., 2022</a>)</span>.</p></li>
<li><p>Consideration about gene flow rates are extremely important in the management of wild plant and animal populations.</p></li>
<li><p><span class="citation" data-cites="Dolan1999">Dolan, Yahr, Menges, &amp; Halfhill (<a href="#/references" role="doc-biblioref" onclick="">1999</a>)</span> indirectly estimated gene flow measuring the genetic differentiation of three rare species of perennial scrubs in Florida. In one of them, <em>Hypericum cumulicola</em>, gene flow was absent. The limited genetic variation detected in this species limits the adaptive response of it to future environmental changes and makes it more prone to extinction.</p></li>
</ul>
</section>
<section id="determination-of-the-time-since-past-genetic-bottlenecks." class="slide level2">
<h2>4. Determination of the time since past genetic bottlenecks.</h2>

<img data-src="./class_13_23.png" class="r-stretch quarto-figure-center"><p class="caption">Each point shown represents a population and the vertical segment represents the age of its associated founder event where the segment length is proportional to the founder age <span class="citation" data-cites="Tournebize2022">(<a href="#/references" role="doc-biblioref" onclick="">Tournebize, Chu, &amp; Moorjani, 2022</a>)</span>.</p></section>
<section id="understanding-patterns-of-reproductive-ecology." class="slide level2">
<h2>5. Understanding patterns of reproductive ecology.</h2>

<img data-src="./class_13_24.png" class="r-stretch quarto-figure-center"><p class="caption">Structure graphical output for two syntopic iguana species from the Galápagos islands <span class="citation" data-cites="DiGiambattista2018">(<a href="#/references" role="doc-biblioref" onclick="">Di Giambattista et al., 2018</a>)</span>.</p></section>
<section class="slide level2">


<img data-src="./class_13_25.png" class="r-stretch quarto-figure-center"><p class="caption">Adapted from <span class="citation" data-cites="MacLeod2015">MacLeod et al. (<a href="#/references" role="doc-biblioref" onclick="">2015</a>)</span>.</p></section>
<section id="locating-original-sources-of-wildlife-products." class="slide level2">
<h2>6. Locating original sources of wildlife products.</h2>
<ul>
<li><p>The International Whaling Commission (IWC) imposed a moratorium on commercial whaling in 1982 <span class="citation" data-cites="Allendorf2022">(<a href="#/references" role="doc-biblioref" onclick="">Allendorf et al., 2022</a>)</span>.</p></li>
<li><p>A small quota of very specific species of whales was still available to countries like Japan and South Korea.</p></li>
</ul>
<blockquote>
<p>The source of these products was assumed to be undocumented `incidental takes’ from fisheries’ by-catch, although we cannot exclude the possibility of illegal hunting or smuggling. The demographic impact of this undocumented exploitation was evaluated using the model of population dynamics adopted by the Scienti¢c Committee of the International Whaling Commission. For the range of exploitation consistent with the market sample, this protected stock was predicted to decline towards extinction over the next few decades <span class="citation" data-cites="Baker2000">(<a href="#/references" role="doc-biblioref" onclick="">Baker, Lento, Cipriano, &amp; Palumbi, 2000</a>)</span>.</p>
</blockquote>
</section>
<section id="managing-genetic-diversity-in-wild-populations" class="slide level2">
<h2>Managing genetic diversity in wild populations</h2>
<ul>
<li><p>The IUCN lists the common adder, <em>Vipera berus</em>, as in the category <a href="https://www.iucnredlist.org/species/47756146/743903">Least Concern</a>. It has a very widespread distribution, although populations are fragmented in small and often isolated demes.</p></li>
<li><p>One isolated population of this species, in Sweden, declined in 1960s and then began to exhibit the a high proportion of deformed and stillborn offspring coupled with low genetic variability, i.e., signs of inbreeding depression <span class="citation" data-cites="Madsen1999">(<a href="#/references" role="doc-biblioref" onclick="">Madsen, Shine, Olsson, &amp; Wittzell, 1999</a>)</span>.</p></li>
<li><p><span class="citation" data-cites="Madsen1999">Madsen et al. (<a href="#/references" role="doc-biblioref" onclick="">1999</a>)</span> introduced 20 youg adder individuals collected from larger and more genetically diverse population in this declining population. After only 4 years, the time needed for an adder to become reproductively active, started to document a reduction in still born babies and a reduction in deformities and an increase in genetic diversity.</p></li>
</ul>
</section>
<section class="slide level2">


<img data-src="./class_13_26.png" class="r-stretch quarto-figure-center"><p class="caption">Numbers of adult male Adders (Vipera berus) captured from 1981 to 2010 along the seashore at an isolated population just west of Smygehuk Village in southern Sweden. After <span class="citation" data-cites="Madsen2011">Madsen &amp; Ujvari (<a href="#/references" role="doc-biblioref" onclick="">2011</a>)</span></p></section>
<section class="slide level2">

<h3 id="hybridization-and-introgression-are-two-important-factors-to-monitor-in-wild-populations.">Hybridization and introgression are two important factors to monitor in wild populations.</h3>
<ul>
<li><p>When two closely related species coexist and one of them is more rare, the rare one risks to be genetically swamped due to hybridization and/or introgression (the incorporation of genetic material from a species/population/subpopulation into the genome of another) <span class="citation" data-cites="Allendorf2022">(<a href="#/references" role="doc-biblioref" onclick="">Allendorf et al., 2022</a>)</span>.</p></li>
<li><p>A very well documented example of managing genetic problems in wildlife population comes from the red wolf (<em>Canis rufus</em>) from south-central United States.</p></li>
<li><p>In 1975 red wolf became extinct in the wild <span class="citation" data-cites="Avise1996">(<a href="#/references" role="doc-biblioref" onclick="">Avise &amp; Hamrick, 1996</a>)</span>. The united States Fish and Wildlife Services (USFWS) initiated a campaign to re-establish a population of this species using captive-bred individuals.</p></li>
<li><p>Released individuals became established and that area was designed as the Red Wolf Experimental Population Area (RWEP-A).</p></li>
</ul>
</section>
<section class="slide level2">

<ul>
<li>The biggest threat was not much the hybridization between red wolves and coyotes (<em>Canis latrans</em>), but the backcrossing between <span class="math inline">\(F_1\)</span> hybrids and adult red wolf individuals (i.e., introgression).</li>
</ul>

<img data-src="./class_13_27.png" class="r-stretch quarto-figure-center"><p class="caption">After <span class="citation" data-cites="Fredrickson2006">Fredrickson &amp; Hedrick (<a href="#/references" role="doc-biblioref" onclick="">2006</a>)</span></p></section>
<section id="managing-genetic-diversity-in-captive-populations" class="slide level2">
<h2>Managing genetic diversity in captive populations</h2>
<ul>
<li><p>Attempts at turkey reintroduction using captive bred individuals were a failure in 95% of cases in the united States <span class="citation" data-cites="Allendorf2022">(<a href="#/references" role="doc-biblioref" onclick="">Allendorf et al., 2022</a>)</span>.</p></li>
<li><p>As <span class="citation" data-cites="Leopold1944">Leopold (<a href="#/references" role="doc-biblioref" onclick="">1944</a>)</span> said:</p></li>
</ul>
<blockquote>
<p>Wild turkey are wary and shy, which are advantageous characteristics in eluding natural and human enemies. The hens and young ones automatically react to threats in ways that are self protective…Birds of the domestic strain, on the other hand, are differently adapted. Many of their physiological reaction and psychological characteristics are favorable to existence in the barnyard but may perclude existence in the wild.</p>
</blockquote>
<ul>
<li>The problem stated above is present in every single population and it is known as <em>adaptation to captivity</em> <span class="citation" data-cites="Allendorf2022">(<a href="#/references" role="doc-biblioref" onclick="">Allendorf et al., 2022</a>)</span>.</li>
</ul>
</section>
<section class="slide level2">

<ul>
<li><p>The more accustomed wild animals become to their artificial environment the less likely they are to survive when reintroduced in the wild.</p></li>
<li><p>Lifetime reproductive success of and overall fitness of stealhead trout (<em>Oncorhynchus mykiss</em>) declined precipitously in just two generations of captivity <span class="citation" data-cites="Araki2007">(<a href="#/references" role="doc-biblioref" onclick="">Araki, Cooper, &amp; Blouin, 2007</a>)</span>.</p></li>
<li><p>The data provided in <span class="citation" data-cites="Araki2007">Araki et al. (<a href="#/references" role="doc-biblioref" onclick="">2007</a>)</span> suggest that 37.5% fitness decline per captive reared generation demonstrating that genetic adaptation (GA) to captivity is something dangerous that should not be underestimated.</p></li>
<li><p>GA can be mathematically described:</p></li>
</ul>
<p><span class="math display">\[
\label{ga}
\tag{13.7}
GA = \frac{Sh^2}{L}\sum_{i=1}^{\frac{y}{L}}[1-\frac{1}{2N_e}]^{\frac{y}{L}}(1-m_i)
\]</span></p>
</section>
<section class="slide level2">

<ul>
<li>Where <span class="math inline">\(S\)</span> is the selection differential in captivity (the strength in selection for captive traits).</li>
<li><span class="math inline">\(h^2\)</span> is heritability (additive genetic variation for reproductive fitness).</li>
</ul>
</section>
<section class="slide level2">

<h3 id="heritability">Heritability</h3>
<ul>
<li><p>The heritability of a character is probably the most important determinant of individuals to evolve in response to selection!</p></li>
<li><p>Heritability is the proportion of phenotypic variance in a population due to genetic differences that parents can pass through offspring in the next generation.</p></li>
<li><p>Heritability is symbolized <span class="math inline">\(h^2\)</span>, but the superscript has no mathematical meaning!</p></li>
<li><p>Molecular measures of genetic diversity do not often answer some of the questions we try to address in conservation genetics as correlation between molecular and quantitative measures of genetic diversity are low.</p></li>
<li><p>The heritability associated with quantitative characters that are related to reproductive fitness are of great interests to conservation biologists.</p></li>
</ul>
</section>
<section class="slide level2">


<img data-src="./class_13_28.jpg" class="r-stretch quarto-figure-center"><p class="caption">Distribution of phenotypes for 5 quantitative characters that are very important for reproductive fitness. After <span class="citation" data-cites="Frankham2010">Frankham et al. (<a href="#/references" role="doc-biblioref" onclick="">2010</a>)</span></p></section>
<section class="slide level2">

<ul>
<li><p>Quantitative traits are influenced by mani loci plus environmental influences (feeding, exposure to sunlight, water availability et. etc. etc.).</p></li>
<li><p>It is not possible to directly infer genotype from phenotype only when we refer to quantitative characters, and individuals with the same genotype may have different genotypes, or individuals with the same genotypes may have different phenotype.</p></li>
<li><p>Genetic diversity for quantitative characters in outbred populations is due to the segregation of multiple mendelian loci referred to as Quantitative Trait Loci (QTL).</p></li>
<li><p>QTLs possess alleles that add to or detract from the magnitude of the quantitative character.</p></li>
<li><p>How do we measure what portion of the phenotypic variance for quantitative traits derives from genetic variation among individuals?</p></li>
</ul>
</section>
<section class="slide level2">

<ol type="1">
<li>Variation within and among populations.</li>
<li>Comparison of inbred with outbred populations.</li>
<li>Resemblance among relatives.</li>
</ol>
<ul>
<li>Because of the great impact that the environment can have on the expression of quantitative traits characters it is extremely important that all comparisons are carried out in a controlled situation and synchronously (i.e., at the same time): common garden experiment.</li>
</ul>
</section>
<section class="slide level2">


<img data-src="./class_13_29.jpg" class="r-stretch quarto-figure-center"><p class="caption">Quantitative genetic variation for corolla length in tobacco. After <span class="citation" data-cites="Strickberger1985">Strickberger (<a href="#/references" role="doc-biblioref" onclick="">1985</a>)</span>.</p></section>
<section class="slide level2">

<p><span class="math display">\[
\label{ga2}
\tag{13.8}
GA = \frac{Sh^2}{L}\sum_{i=1}^{\frac{y}{L}}[1-\frac{1}{2N_e}]^{\frac{y}{L}}(1-m_i)
\]</span></p>
<ul>
<li>Where <span class="math inline">\(S\)</span> is the selection differential in captivity (the strength in selection for captive traits).</li>
<li><span class="math inline">\(h^2\)</span> is heritability (additive genetic variation for reproductive fitness).</li>
<li><span class="math inline">\(y/L\)</span> number of generations in captivity (<span class="math inline">\(y\)</span> = years, <span class="math inline">\(L\)</span> = generation length).</li>
<li><span class="math inline">\(N_e\)</span>, you should recognize this one….</li>
<li><span class="math inline">\(m_i\)</span> proportion of genetic material coming from immigrants.</li>
</ul>
</section>
<section class="slide level2">

<h3 id="what-could-we-do-to-minimize-ga">What could we do to minimize GA?</h3>
<ul>
<li>Reduce the amount of time in captivity (i.e., reduce the number of iterations <span class="math inline">\(\sum_{i=1}^{y/L}\)</span>).</li>
<li>Reduce the strength of selection, <span class="math inline">\(S\)</span>, in captivity. Very complicated, as it entails making the captive environment as similar as possible to the natural environment, both at the biotic and the abiotic level.</li>
<li>Add more immigrants (unrelated individuals) from wild or captive populations to slow the rate of genetic adaptation.</li>
</ul>
<p>Understanding the intentions, methods and scope of the management plans is essential if we want to grasp the way in which captive populations are integrated into worldwide conservation strategies today <span class="citation" data-cites="Allendorf2022">(<a href="#/references" role="doc-biblioref" onclick="">Allendorf et al., 2022</a>)</span>.</p>
</section>
<section id="references" class="slide level2 smaller scrollable">
<h2>References</h2>

<div class="quarto-auto-generated-content">
<p><img src="../../figures/Tor-Vergata_logo_cmyk.png" class="slide-logo"></p>
<div class="footer footer-default">
<p><a href="https://giulianocolosimo.github.io/biologia_della_conservazione_23_24/">Biologia della conservazione 23/24 - Giuliano Colosimo, PhD</a></p>
</div>
</div>
<div id="refs" class="references csl-bib-body hanging-indent" data-entry-spacing="0" data-line-spacing="2" role="list">
<div id="ref-Akakaya2007" class="csl-entry" role="listitem">
Akçakaya, H. R., Mills, G., &amp; Doncaster, C. P. (2007). The role of metapopulations in conservation.
</div>
<div id="ref-Allen1995" class="csl-entry" role="listitem">
Allen, P. J., Amos, W., Pomeroy, P. P., &amp; Twiss, S. D. (1995). Microsatellite variation in grey seals (<em>halichoerus grypus</em>) shows evidence of genetic differentiation between two british breeding colonies. <em>Molecular Ecology</em>, <em>4</em>(6), 653–662. Wiley. Retrieved from <a href="https://doi.org/10.1111%2Fj.1365-294x.1995.tb00266.x">https://doi.org/10.1111%2Fj.1365-294x.1995.tb00266.x</a>
</div>
<div id="ref-Allendorf2022" class="csl-entry" role="listitem">
Allendorf, F. W., Funk, W. C., Aitken, S. N., Byrne, M., Luikart, G., &amp; Antunes, A. (2022). <em><span class="nocase">Conservation and the Genomics of Populations</span></em>. Oxford University Press. Retrieved from <a href="https://doi.org/10.1093/oso/9780198856566.001.0001">https://doi.org/10.1093/oso/9780198856566.001.0001</a>
</div>
<div id="ref-Araki2007" class="csl-entry" role="listitem">
Araki, H., Cooper, B., &amp; Blouin, M. S. (2007). Genetic effects of captive breeding cause a rapid, cumulative fitness decline in the wild. <em>Science</em>, <em>318</em>(5847), 100–103. Retrieved from <a href="https://www.science.org/doi/abs/10.1126/science.1145621">https://www.science.org/doi/abs/10.1126/science.1145621</a>
</div>
<div id="ref-Avise1996" class="csl-entry" role="listitem">
Avise, J. C., &amp; Hamrick, J. L. (1996). <em>Conservation genetics : Case histories from nature</em>. (J. C. Avise &amp; J. L. Hamrick, Eds.)<em>Conservation genetics : case histories from nature /</em>. New York: Chapman &amp; Hall.
</div>
<div id="ref-deBaan2013" class="csl-entry" role="listitem">
Baan, L. de, Alkemade, R., &amp; Koellner, T. (2013). Land use impacts on biodiversity in LCA: A global approach. <em>The International Journal of Life Cycle Assessment</em>, <em>18</em>(6), 1216–1230. Retrieved from <a href="https://link.springer.com/content/pdf/10.1007/s11367-012-0412-0.pdf">https://link.springer.com/content/pdf/10.1007/s11367-012-0412-0.pdf</a>
</div>
<div id="ref-Baker2000" class="csl-entry" role="listitem">
Baker, C. S., Lento, G. M., Cipriano, F., &amp; Palumbi, S. R. (2000). Predicted decline of protected whales based on molecular genetic monitoring of japanese and korean markets. <em>Proceedings of the Royal Society of London. Series B: Biological Sciences</em>, <em>267</em>(1449), 1191–1199. The Royal Society. Retrieved from <a href="https://doi.org/10.1098%2Frspb.2000.1128">https://doi.org/10.1098%2Frspb.2000.1128</a>
</div>
<div id="ref-Colosimo2021b" class="csl-entry" role="listitem">
Colosimo, G., Jackson, A. C., Benton, A., Varela-Stokes, A., Iverson, J., Knapp, C. R., &amp; Welch, M. (2021). <span class="nocase">Correlated Population Genetic Structure in a Three-Tiered Host-Parasite System: The Potential for Coevolution and Adaptive Divergence</span>. <em>Journal of Heredity</em>, <em>112</em>(7), 590–601. Retrieved from <a href="https://doi.org/10.1093/jhered/esab058">https://doi.org/10.1093/jhered/esab058</a>
</div>
<div id="ref-Colosimo2014" class="csl-entry" role="listitem">
Colosimo, G., Knapp, C. R., Wallace, L. E., &amp; Welch, M. E. (2014). <span class="nocase">Inferred vs realized patterns of gene flow: an analysis of population structure in the Andros Island Rock Iguana</span>. (V. Ketmaier, Ed.)<em>PLoS ONE</em>, <em>9</em>(9), e106963. Retrieved from <a href="http://dx.plos.org/10.1371/journal.pone.0106963">http://dx.plos.org/10.1371/journal.pone.0106963</a>
</div>
<div id="ref-DiGiambattista2018" class="csl-entry" role="listitem">
Di Giambattista, L., Fulvo, A., Fabiani, A., Bonanni, J., Carrión, J. E., &amp; Gentile, G. (2018). <span class="nocase">Molecular data exclude current hybridization between iguanas <em>Conolophus marthae</em> and <em>C. subcristatus</em> on Wolf Volcano (Gal<span class="nocase">á</span>pagos Islands)</span>. <em>Conservation Genetics</em>, <em>19</em>(6), 1461–1469. Springer Netherlands. Retrieved from <a href="http://link.springer.com/10.1007/s10592-018-1114-3">http://link.springer.com/10.1007/s10592-018-1114-3</a>
</div>
<div id="ref-Dolan1999" class="csl-entry" role="listitem">
Dolan, R. W., Yahr, R., Menges, E. S., &amp; Halfhill, M. D. (1999). Conservation implications of genetic variation in three rare species endemic to florida rosemary scrub. <em>American Journal of Botany</em>, <em>86</em>(11), 1556–1562. Botanical Society of America. Retrieved May 8, 2023, from <a href="http://www.jstor.org/stable/2656793">http://www.jstor.org/stable/2656793</a>
</div>
<div id="ref-Eldridge1999" class="csl-entry" role="listitem">
Eldridge, M. D. B., King, J. M., Loupis, A. K., Spencer, P. B. S., Taylor, A. C., Pope, L. C., &amp; Hall, G. P. (1999). Unprecedented low levels of genetic variation and inbreeding depression in an island population of the black-footed rock-wallaby. <em>Conservation Biology</em>, <em>13</em>(3), 531–541. Retrieved from <a href="https://conbio.onlinelibrary.wiley.com/doi/abs/10.1046/j.1523-1739.1999.98115.x">https://conbio.onlinelibrary.wiley.com/doi/abs/10.1046/j.1523-1739.1999.98115.x</a>
</div>
<div id="ref-Excoffier1992" class="csl-entry" role="listitem">
Excoffier, L., Smouse, P. E., &amp; Quattro, J. M. (1992). <a href="https://doi.org/10.1007/s00424-009-0730-7"><span class="nocase">Analysis of molecular variance inferred from metric distances among DNA haplotypes: Application to human mitochondrial DNA restriction data</span></a>. <em>Genetics</em>, <em>131</em>(2), 479–491.
</div>
<div id="ref-Forbes1999" class="csl-entry" role="listitem">
Forbes, S. H., &amp; Hogg, J. T. (1999). Assessing population structure at high levels of differentiation: Microsatellite comparisons of bighorn sheep and large carnivores. <em>Animal Conservation</em>, <em>2</em>(3), 223–233. Retrieved from <a href="https://zslpublications.onlinelibrary.wiley.com/doi/abs/10.1111/j.1469-1795.1999.tb00068.x">https://zslpublications.onlinelibrary.wiley.com/doi/abs/10.1111/j.1469-1795.1999.tb00068.x</a>
</div>
<div id="ref-Frankham2010" class="csl-entry" role="listitem">
Frankham, R., Ballou, J. D., &amp; Briscoe, D. A. (2010). <em>Introduction to conservation genetics</em>. Cambridge University Press. Retrieved from <a href="https://doi.org/10.1017%2Fcbo9780511809002">https://doi.org/10.1017%2Fcbo9780511809002</a>
</div>
<div id="ref-Frankham2011" class="csl-entry" role="listitem">
Frankham, R., Ballou, J. D., Eldridge, M. D. B., Lacy, R. C., Ralls, K., Dudash, M. R., &amp; Fenster, C. B. (2011). <a href="https://doi.org/10.1111/j.1523-1739.2011.01662.x"><span class="nocase">Predicting the probability of outbreeding depression</span></a>. <em>Conservation Biology</em>, <em>25</em>(3), 465–475.
</div>
<div id="ref-Fredrickson2006" class="csl-entry" role="listitem">
Fredrickson, R. J., &amp; Hedrick, P. W. (2006). Dynamics of hybridization and introgression in red wolves and coyotes. <em>Conservation Biology</em>, <em>20</em>(4), 1272–1283. Retrieved from <a href="https://conbio.onlinelibrary.wiley.com/doi/abs/10.1111/j.1523-1739.2006.00401.x">https://conbio.onlinelibrary.wiley.com/doi/abs/10.1111/j.1523-1739.2006.00401.x</a>
</div>
<div id="ref-Gaggiotti2010" class="csl-entry" role="listitem">
Gaggiotti, O. E., &amp; Foll, M. (2010). Quantifying population structure using the f-model. <em>Molecular Ecology Resources</em>, <em>10</em>(5), 821–830. Wiley. Retrieved from <a href="https://doi.org/10.1111%2Fj.1755-0998.2010.02873.x">https://doi.org/10.1111%2Fj.1755-0998.2010.02873.x</a>
</div>
<div id="ref-Hanski1997" class="csl-entry" role="listitem">
Hanski, I., &amp; Simberloff, D. (1997). The metapopulation approach, its history, conceptual domain, and application to conservation. In I. Hanski &amp; M. E. Gilpin (Eds.), <em>Metapopulation biology</em> (pp. 5–26). San Diego: Academic Press. Retrieved from <a href="https://www.sciencedirect.com/science/article/pii/B9780123234452500031">https://www.sciencedirect.com/science/article/pii/B9780123234452500031</a>
</div>
<div id="ref-Hedrick2005" class="csl-entry" role="listitem">
Hedrick, P. W. (2005). <em><span class="nocase">Genetics of Populations</span></em> (Third., pp. 1–737). Sudbury, MA: Jones; Bartlett.
</div>
<div id="ref-Lacy1987" class="csl-entry" role="listitem">
Lacy, R. C. (1987). Loss of genetic diversity from managed populations: Interacting effects of drift, mutation, immigration, selection, and population subdivision. <em>Conservation Biology</em>, <em>1</em>(2), 143–158. Retrieved from <a href="https://conbio.onlinelibrary.wiley.com/doi/abs/10.1111/j.1523-1739.1987.tb00023.x">https://conbio.onlinelibrary.wiley.com/doi/abs/10.1111/j.1523-1739.1987.tb00023.x</a>
</div>
<div id="ref-Leopold1944" class="csl-entry" role="listitem">
Leopold, A. S. (1944). <span class="nocase">The Nature of Heritable Wildness in Turkeys</span>. <em>The Condor</em>, <em>46</em>(4), 133–197. Retrieved from <a href="https://doi.org/10.2307/1364213">https://doi.org/10.2307/1364213</a>
</div>
<div id="ref-MacLeod2015" class="csl-entry" role="listitem">
MacLeod, A., Rodríguez, A., Vences, M., Orozco-terWengel, P., García, C., Trillmich, F., Gentile, G., et al. (2015). <span class="nocase">Hybridization masks speciation in the evolutionary history of the Gal<span class="nocase">á</span>pagos marine iguana</span>. <em>Proceedings of the Royal Society B: Biological Sciences</em>, <em>282</em>(1809), 20150425. Retrieved from <a href="http://rspb.royalsocietypublishing.org/lookup/doi/10.1098/rspb.2015.0425">http://rspb.royalsocietypublishing.org/lookup/doi/10.1098/rspb.2015.0425</a>
</div>
<div id="ref-Madsen1999" class="csl-entry" role="listitem">
Madsen, T., Shine, R., Olsson, M., &amp; Wittzell, H. (1999). <span class="nocase">Conservation biology: Restoration of an inbred adder population</span>. <em>Nature</em>, <em>402</em>(6757), 34–35. Retrieved from <a href="http://www.nature.com/nature/journal/v402/n6757/pdf/402034a0.pdf">http://www.nature.com/nature/journal/v402/n6757/pdf/402034a0.pdf</a>
</div>
<div id="ref-Madsen2011" class="csl-entry" role="listitem">
Madsen, T., &amp; Ujvari, B. (2011). <span class="nocase">The potential demise of a population of adders (<em>Vipera berus</em>) in Smygehuk, Sweden</span>. <em>Herpetological Conservation and Biology</em>, <em>6</em>(1), 72–74.
</div>
<div id="ref-Malecot1948" class="csl-entry" role="listitem">
Malécot, G. (1948). <em>Les math<span>é</span>matiques de l’h<span>é</span>r<span>é</span>dit<span>é</span></em>. Masson. Retrieved from <a href="https://books.google.fi/books?id=g7FjAQAACAAJ">https://books.google.fi/books?id=g7FjAQAACAAJ</a>
</div>
<div id="ref-Mills1996" class="csl-entry" role="listitem">
Mills, L. S., &amp; Allendorf, F. W. (1996). <span class="nocase">The one migrant per generation rule in conservation and management</span>. <em>Conservation Biology</em>, <em>10</em>(6), 1509–1518.
</div>
<div id="ref-Nei1977" class="csl-entry" role="listitem">
Nei, M. (1977). F-statistics and analysis of gene diversity in subdivided populations. <em>Annals of Human Genetics</em>, <em>41</em>(2), 225–233. Wiley. Retrieved from <a href="https://doi.org/10.1111%2Fj.1469-1809.1977.tb01918.x">https://doi.org/10.1111%2Fj.1469-1809.1977.tb01918.x</a>
</div>
<div id="ref-Luci2011" class="csl-entry" role="listitem">
Nunes, L. (2011). <a href="https://doi.org/10.22498/pages.19.2.60">Landslides in s<span>ã</span>o paulo, brazil: An integrated historical perspective</a>. <em>Pages News</em>, <em>19</em>, 60–63.
</div>
<div id="ref-Obrien1996" class="csl-entry" role="listitem">
O’Brien, S. J. (1996). Conservation genetics of the felidae. In J. C. Avise &amp; J. L. Hamrick (Eds.), <em>Conservation genetics: Case histories from nature.</em> (pp. 50–74). New York: Chapman; Hall.
</div>
<div id="ref-Riley2006" class="csl-entry" role="listitem">
Riley, S. P. D., Pollinger, J. P., Sauvajot, R. M., York, E. C., BROMLEY, C., FULLER, T. K., &amp; WAYNE, R. K. (2006). A southern california freeway is a physical and social barrier to gene flow in carnivores. <em>Molecular Ecology</em>, <em>15</em>(7), 1733–1741. Wiley. Retrieved from <a href="https://doi.org/10.1111%2Fj.1365-294x.2006.02907.x">https://doi.org/10.1111%2Fj.1365-294x.2006.02907.x</a>
</div>
<div id="ref-Schwartz2004" class="csl-entry" role="listitem">
Schwartz, M. K., Pilgrim, K. L., McKelvey, K. S., Lindquist, E. L., Claar, J. J., Loch, S., &amp; Ruggiero, L. F. (2004). Hybridization between canada lynx and bobcats: Genetic results and management implications. <em>Conservation Genetics</em>, <em>5</em>(3), 349–355. Retrieved from <a href="https://link.springer.com/content/pdf/10.1023/B:COGE.0000031141.47148.8b.pdf">https://link.springer.com/content/pdf/10.1023/B:COGE.0000031141.47148.8b.pdf</a>
</div>
<div id="ref-Seehausen1997" class="csl-entry" role="listitem">
Seehausen, O., Alphen, J. J. M. van, &amp; Witte, F. (1997). Cichlid fish diversity threatened by eutrophication that curbs sexual selection. <em>Science</em>, <em>277</em>(5333), 1808–1811. Retrieved from <a href="https://www.science.org/doi/abs/10.1126/science.277.5333.1808">https://www.science.org/doi/abs/10.1126/science.277.5333.1808</a>
</div>
<div id="ref-Slatkin1987" class="csl-entry" role="listitem">
Slatkin, M. (1987). <a href="https://doi.org/10.1126/science.3576198"><span class="nocase">Gene flow and the geographic structure of natural populations</span></a>. <em>Science</em>, <em>236</em>(4803), 787–792.
</div>
<div id="ref-Slatkin1995" class="csl-entry" role="listitem">
Slatkin, M. (1995). A measure of population subdivision based on microsatellite allele frequencies. <em>Genetics</em>, <em>139</em>(1), 457–462. Oxford University Press (<span>OUP</span>). Retrieved from <a href="https://doi.org/10.1093%2Fgenetics%2F139.1.457">https://doi.org/10.1093%2Fgenetics%2F139.1.457</a>
</div>
<div id="ref-Strickberger1985" class="csl-entry" role="listitem">
Strickberger, M. W. (1985). <em>Genetics</em>. Macmillan international editions in science. Macmillan. Retrieved from <a href="https://books.google.co.uk/books?id=IWrwAAAAMAAJ">https://books.google.co.uk/books?id=IWrwAAAAMAAJ</a>
</div>
<div id="ref-Thatte2018" class="csl-entry" role="listitem">
Thatte, P., Joshi, A., Vaidyanathan, S., Landguth, E., &amp; Ramakrishnan, U. (2018). Maintaining tiger connectivity and minimizing extinction into the next century: Insights from landscape genetics and spatially-explicit simulations. <em>Biological Conservation</em>, <em>218</em>, 181–191. Retrieved from <a href="https://www.sciencedirect.com/science/article/pii/S0006320717307346">https://www.sciencedirect.com/science/article/pii/S0006320717307346</a>
</div>
<div id="ref-Tournebize2022" class="csl-entry" role="listitem">
Tournebize, R., Chu, G., &amp; Moorjani, P. (2022). Reconstructing the history of founder events using genome-wide patterns of allele sharing across individuals. (K. E. Lohmueller, Ed.)<em><span>PLOS</span> Genetics</em>, <em>18</em>(6), e1010243. Public Library of Science (<span>PLoS</span>). Retrieved from <a href="https://doi.org/10.1371%2Fjournal.pgen.1010243">https://doi.org/10.1371%2Fjournal.pgen.1010243</a>
</div>
<div id="ref-Vidya2005" class="csl-entry" role="listitem">
Vidya, T. N. C., Fernando, P., Melnick, D. J., &amp; Sukumar, R. (2005). <span class="nocase">Population genetic structure and conservation of Asian elephants (<em>Elephas maximus</em>) across India</span>. <em>Animal Conservation</em>, <em>8</em>(4), 377–388. Retrieved from <a href="https://zslpublications.onlinelibrary.wiley.com/doi/abs/10.1017/S1367943005002428">https://zslpublications.onlinelibrary.wiley.com/doi/abs/10.1017/S1367943005002428</a>
</div>
<div id="ref-Vucetich2000" class="csl-entry" role="listitem">
Vucetich, J. A., &amp; Waite, T. A. (2000). <span class="nocase">Is one migrant per generation sufficient for the genetic management of fluctuating populations?</span> <em>Animal Conservation</em>, <em>3</em>, 261–266.
</div>
<div id="ref-Wahlund1928" class="csl-entry" role="listitem">
Wahlund, S. (1928). ZUSAMMENSETZUNG VON POPULATIONEN UND KORRELATIONSERSCHEINUNGEN VOM STANDPUNKT DER VERERBUNGSLEHRE AUS BETRACHTET. <em>Hereditas</em>, <em>11</em>(1), 65–106. Retrieved from <a href="https://onlinelibrary.wiley.com/doi/abs/10.1111/j.1601-5223.1928.tb02483.x">https://onlinelibrary.wiley.com/doi/abs/10.1111/j.1601-5223.1928.tb02483.x</a>
</div>
<div id="ref-Welch2017" class="csl-entry" role="listitem">
Welch, M. E., Colosimo, G., Pasachnik, S. A., Malone, C. L., Hilton, J., Long, J., Getz, A. H., et al. (2017). <span class="nocase">Molecular variation and population structure in critically endangered Turks and Caicos Rock Iguanas: identifying intraspecific conservation units and revising subspecific taxonomy</span>. <em>Conservation Genetics</em>, <em>18</em>(2), 479–493. Retrieved from <a href="http://link.springer.com/10.1007/s10592-016-0922-6">http://link.springer.com/10.1007/s10592-016-0922-6</a>
</div>
<div id="ref-Wright1931" class="csl-entry" role="listitem">
Wright, S. (1931). <a href="https://doi.org/10.1016/S0092-8240(05)80011-4"><span class="nocase">Evolution in Mendelian populations</span></a>. <em>Bulletin of Mathematical Biology</em>, <em>52</em>(1-2), 241–295.
</div>
<div id="ref-Wright1951" class="csl-entry" role="listitem">
Wright, S. (1951). <a href="https://doi.org/10.2307/2407273"><span class="nocase">The genetical structure of populations</span></a>. <em>Annals of eugenics</em>, <em>15</em>, 323–354.
</div>
</div>
</section>
    </div>
  </div>

  <script>window.backupDefine = window.define; window.define = undefined;</script>
  <script src="../../site_libs/revealjs/dist/reveal.js"></script>
  <!-- reveal.js plugins -->
  <script src="../../site_libs/revealjs/plugin/quarto-line-highlight/line-highlight.js"></script>
  <script src="../../site_libs/revealjs/plugin/pdf-export/pdfexport.js"></script>
  <script src="../../site_libs/revealjs/plugin/reveal-menu/menu.js"></script>
  <script src="../../site_libs/revealjs/plugin/reveal-menu/quarto-menu.js"></script>
  <script src="../../site_libs/revealjs/plugin/reveal-chalkboard/plugin.js"></script>
  <script src="../../site_libs/revealjs/plugin/multiplex/socket.io.js"></script>
  <script src="../../site_libs/revealjs/plugin/multiplex/multiplex.js"></script>
  <script src="../../site_libs/revealjs/plugin/quarto-support/support.js"></script>
  

  <script src="../../site_libs/revealjs/plugin/notes/notes.js"></script>
  <script src="../../site_libs/revealjs/plugin/search/search.js"></script>
  <script src="../../site_libs/revealjs/plugin/zoom/zoom.js"></script>
  <script src="../../site_libs/revealjs/plugin/math/math.js"></script>
  <script>window.define = window.backupDefine; window.backupDefine = undefined;</script>

  <script>

      // Full list of configuration options available at:
      // https://revealjs.com/config/
      Reveal.initialize({
'controlsAuto': true,
'previewLinksAuto': false,
'pdfSeparateFragments': false,
'autoAnimateEasing': "ease",
'autoAnimateDuration': 1,
'autoAnimateUnmatched': true,
'menu': {"side":"left","useTextContentForMissingTitles":true,"markers":false,"loadIcons":false,"custom":[{"title":"Tools","icon":"<i class=\"fas fa-gear\"></i>","content":"<ul class=\"slide-menu-items\">\n<li class=\"slide-tool-item active\" data-item=\"0\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.fullscreen(event)\"><kbd>f</kbd> Fullscreen</a></li>\n<li class=\"slide-tool-item\" data-item=\"1\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.speakerMode(event)\"><kbd>s</kbd> Speaker View</a></li>\n<li class=\"slide-tool-item\" data-item=\"2\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.overview(event)\"><kbd>o</kbd> Slide Overview</a></li>\n<li class=\"slide-tool-item\" data-item=\"3\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.togglePdfExport(event)\"><kbd>e</kbd> PDF Export Mode</a></li>\n<li class=\"slide-tool-item\" data-item=\"4\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.toggleChalkboard(event)\"><kbd>b</kbd> Toggle Chalkboard</a></li>\n<li class=\"slide-tool-item\" data-item=\"5\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.toggleNotesCanvas(event)\"><kbd>c</kbd> Toggle Notes Canvas</a></li>\n<li class=\"slide-tool-item\" data-item=\"6\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.downloadDrawings(event)\"><kbd>d</kbd> Download Drawings</a></li>\n<li class=\"slide-tool-item\" data-item=\"7\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.keyboardHelp(event)\"><kbd>?</kbd> Keyboard Help</a></li>\n</ul>"}],"openButton":true},
'chalkboard': {"buttons":true,"boardmarkerWidth":5},
'multiplex': {"secret":"17147123756176200911","id":"497cfb7eb2f19256","url":"https://reveal-multiplex.glitch.me/"},
'smaller': false,
 
        // Display controls in the bottom right corner
        controls: false,

        // Help the user learn the controls by providing hints, for example by
        // bouncing the down arrow when they first encounter a vertical slide
        controlsTutorial: false,

        // Determines where controls appear, "edges" or "bottom-right"
        controlsLayout: 'edges',

        // Visibility rule for backwards navigation arrows; "faded", "hidden"
        // or "visible"
        controlsBackArrows: 'faded',

        // Display a presentation progress bar
        progress: true,

        // Display the page number of the current slide
        slideNumber: 'c/t',

        // 'all', 'print', or 'speaker'
        showSlideNumber: 'all',

        // Add the current slide number to the URL hash so that reloading the
        // page/copying the URL will return you to the same slide
        hash: true,

        // Start with 1 for the hash rather than 0
        hashOneBasedIndex: false,

        // Flags if we should monitor the hash and change slides accordingly
        respondToHashChanges: true,

        // Push each slide change to the browser history
        history: true,

        // Enable keyboard shortcuts for navigation
        keyboard: true,

        // Enable the slide overview mode
        overview: true,

        // Disables the default reveal.js slide layout (scaling and centering)
        // so that you can use custom CSS layout
        disableLayout: false,

        // Vertical centering of slides
        center: false,

        // Enables touch navigation on devices with touch input
        touch: true,

        // Loop the presentation
        loop: false,

        // Change the presentation direction to be RTL
        rtl: false,

        // see https://revealjs.com/vertical-slides/#navigation-mode
        navigationMode: 'linear',

        // Randomizes the order of slides each time the presentation loads
        shuffle: false,

        // Turns fragments on and off globally
        fragments: true,

        // Flags whether to include the current fragment in the URL,
        // so that reloading brings you to the same fragment position
        fragmentInURL: false,

        // Flags if the presentation is running in an embedded mode,
        // i.e. contained within a limited portion of the screen
        embedded: false,

        // Flags if we should show a help overlay when the questionmark
        // key is pressed
        help: true,

        // Flags if it should be possible to pause the presentation (blackout)
        pause: true,

        // Flags if speaker notes should be visible to all viewers
        showNotes: false,

        // Global override for autoplaying embedded media (null/true/false)
        autoPlayMedia: null,

        // Global override for preloading lazy-loaded iframes (null/true/false)
        preloadIframes: null,

        // Number of milliseconds between automatically proceeding to the
        // next slide, disabled when set to 0, this value can be overwritten
        // by using a data-autoslide attribute on your slides
        autoSlide: 0,

        // Stop auto-sliding after user input
        autoSlideStoppable: true,

        // Use this method for navigation when auto-sliding
        autoSlideMethod: null,

        // Specify the average time in seconds that you think you will spend
        // presenting each slide. This is used to show a pacing timer in the
        // speaker view
        defaultTiming: null,

        // Enable slide navigation via mouse wheel
        mouseWheel: false,

        // The display mode that will be used to show slides
        display: 'block',

        // Hide cursor if inactive
        hideInactiveCursor: true,

        // Time before the cursor is hidden (in ms)
        hideCursorTime: 5000,

        // Opens links in an iframe preview overlay
        previewLinks: true,

        // Transition style (none/fade/slide/convex/concave/zoom)
        transition: 'fade',

        // Transition speed (default/fast/slow)
        transitionSpeed: 'default',

        // Transition style for full page slide backgrounds
        // (none/fade/slide/convex/concave/zoom)
        backgroundTransition: 'none',

        // Number of slides away from the current that are visible
        viewDistance: 3,

        // Number of slides away from the current that are visible on mobile
        // devices. It is advisable to set this to a lower number than
        // viewDistance in order to save resources.
        mobileViewDistance: 2,

        // The "normal" size of the presentation, aspect ratio will be preserved
        // when the presentation is scaled to fit different resolutions. Can be
        // specified using percentage units.
        width: 1600,

        height: 900,

        // Factor of the display size that should remain empty around the content
        margin: 0.1,

        math: {
          mathjax: 'https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.0/MathJax.js',
          config: 'TeX-AMS_HTML-full',
          tex2jax: {
            inlineMath: [['\\(','\\)']],
            displayMath: [['\\[','\\]']],
            balanceBraces: true,
            processEscapes: false,
            processRefs: true,
            processEnvironments: true,
            preview: 'TeX',
            skipTags: ['script','noscript','style','textarea','pre','code'],
            ignoreClass: 'tex2jax_ignore',
            processClass: 'tex2jax_process'
          },
        },

        // reveal.js plugins
        plugins: [QuartoLineHighlight, PdfExport, RevealMenu, RevealChalkboard, QuartoSupport,

          RevealMath,
          RevealNotes,
          RevealSearch,
          RevealZoom
        ]
      });
    </script>
    
    <script>
      // htmlwidgets need to know to resize themselves when slides are shown/hidden.
      // Fire the "slideenter" event (handled by htmlwidgets.js) when the current
      // slide changes (different for each slide format).
      (function () {
        // dispatch for htmlwidgets
        function fireSlideEnter() {
          const event = window.document.createEvent("Event");
          event.initEvent("slideenter", true, true);
          window.document.dispatchEvent(event);
        }

        function fireSlideChanged(previousSlide, currentSlide) {
          fireSlideEnter();

          // dispatch for shiny
          if (window.jQuery) {
            if (previousSlide) {
              window.jQuery(previousSlide).trigger("hidden");
            }
            if (currentSlide) {
              window.jQuery(currentSlide).trigger("shown");
            }
          }
        }

        // hookup for slidy
        if (window.w3c_slidy) {
          window.w3c_slidy.add_observer(function (slide_num) {
            // slide_num starts at position 1
            fireSlideChanged(null, w3c_slidy.slides[slide_num - 1]);
          });
        }

      })();
    </script>

    <script id="quarto-html-after-body" type="application/javascript">
    window.document.addEventListener("DOMContentLoaded", function (event) {
      const toggleBodyColorMode = (bsSheetEl) => {
        const mode = bsSheetEl.getAttribute("data-mode");
        const bodyEl = window.document.querySelector("body");
        if (mode === "dark") {
          bodyEl.classList.add("quarto-dark");
          bodyEl.classList.remove("quarto-light");
        } else {
          bodyEl.classList.add("quarto-light");
          bodyEl.classList.remove("quarto-dark");
        }
      }
      const toggleBodyColorPrimary = () => {
        const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
        if (bsSheetEl) {
          toggleBodyColorMode(bsSheetEl);
        }
      }
      toggleBodyColorPrimary();  
      const tabsets =  window.document.querySelectorAll(".panel-tabset-tabby")
      tabsets.forEach(function(tabset) {
        const tabby = new Tabby('#' + tabset.id);
      });
      const isCodeAnnotation = (el) => {
        for (const clz of el.classList) {
          if (clz.startsWith('code-annotation-')) {                     
            return true;
          }
        }
        return false;
      }
      const clipboard = new window.ClipboardJS('.code-copy-button', {
        text: function(trigger) {
          const codeEl = trigger.previousElementSibling.cloneNode(true);
          for (const childEl of codeEl.children) {
            if (isCodeAnnotation(childEl)) {
              childEl.remove();
            }
          }
          return codeEl.innerText;
        }
      });
      clipboard.on('success', function(e) {
        // button target
        const button = e.trigger;
        // don't keep focus
        button.blur();
        // flash "checked"
        button.classList.add('code-copy-button-checked');
        var currentTitle = button.getAttribute("title");
        button.setAttribute("title", "Copied!");
        let tooltip;
        if (window.bootstrap) {
          button.setAttribute("data-bs-toggle", "tooltip");
          button.setAttribute("data-bs-placement", "left");
          button.setAttribute("data-bs-title", "Copied!");
          tooltip = new bootstrap.Tooltip(button, 
            { trigger: "manual", 
              customClass: "code-copy-button-tooltip",
              offset: [0, -8]});
          tooltip.show();    
        }
        setTimeout(function() {
          if (tooltip) {
            tooltip.hide();
            button.removeAttribute("data-bs-title");
            button.removeAttribute("data-bs-toggle");
            button.removeAttribute("data-bs-placement");
          }
          button.setAttribute("title", currentTitle);
          button.classList.remove('code-copy-button-checked');
        }, 1000);
        // clear code selection
        e.clearSelection();
      });
        var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
        var mailtoRegex = new RegExp(/^mailto:/);
          var filterRegex = new RegExp("https:\/\/giulianocolosimo\.github\.com");
        var isInternal = (href) => {
            return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
        }
        // Inspect non-navigation links and adorn them if external
     	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool)');
        for (var i=0; i<links.length; i++) {
          const link = links[i];
          if (!isInternal(link.href)) {
            // undo the damage that might have been done by quarto-nav.js in the case of
            // links that we want to consider external
            if (link.dataset.originalHref !== undefined) {
              link.href = link.dataset.originalHref;
            }
          }
        }
      function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
        const config = {
          allowHTML: true,
          maxWidth: 500,
          delay: 100,
          arrow: false,
          appendTo: function(el) {
              return el.closest('section.slide') || el.parentElement;
          },
          interactive: true,
          interactiveBorder: 10,
          theme: 'light-border',
          placement: 'bottom-start',
        };
        if (contentFn) {
          config.content = contentFn;
        }
        if (onTriggerFn) {
          config.onTrigger = onTriggerFn;
        }
        if (onUntriggerFn) {
          config.onUntrigger = onUntriggerFn;
        }
          config['offset'] = [0,0];
          config['maxWidth'] = 700;
        window.tippy(el, config); 
      }
      const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
      for (var i=0; i<noterefs.length; i++) {
        const ref = noterefs[i];
        tippyHover(ref, function() {
          // use id or data attribute instead here
          let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
          try { href = new URL(href).hash; } catch {}
          const id = href.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note) {
            return note.innerHTML;
          } else {
            return "";
          }
        });
      }
      const findCites = (el) => {
        const parentEl = el.parentElement;
        if (parentEl) {
          const cites = parentEl.dataset.cites;
          if (cites) {
            return {
              el,
              cites: cites.split(' ')
            };
          } else {
            return findCites(el.parentElement)
          }
        } else {
          return undefined;
        }
      };
      var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
      for (var i=0; i<bibliorefs.length; i++) {
        const ref = bibliorefs[i];
        const citeInfo = findCites(ref);
        if (citeInfo) {
          tippyHover(citeInfo.el, function() {
            var popup = window.document.createElement('div');
            citeInfo.cites.forEach(function(cite) {
              var citeDiv = window.document.createElement('div');
              citeDiv.classList.add('hanging-indent');
              citeDiv.classList.add('csl-entry');
              var biblioDiv = window.document.getElementById('ref-' + cite);
              if (biblioDiv) {
                citeDiv.innerHTML = biblioDiv.innerHTML;
              }
              popup.appendChild(citeDiv);
            });
            return popup.innerHTML;
          });
        }
      }
    });
    </script>
    

</body></html>