<!DOCTYPE html>
<html lang="en"><head>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-html/tabby.min.js"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/light-border.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-html.min.css" rel="stylesheet" data-mode="light">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles"><meta charset="utf-8">
  <meta name="generator" content="quarto-1.3.450">

  <meta name="dcterms.date" content="2024-04-11">
  <title>Biologia della Conservazione 23/24 - Biologia della Conservazione</title>
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no, minimal-ui">
  <link rel="stylesheet" href="../../site_libs/revealjs/dist/reset.css">
  <link rel="stylesheet" href="../../site_libs/revealjs/dist/reveal.css">
  <style>
    code{white-space: pre-wrap;}
    span.smallcaps{font-variant: small-caps;}
    div.columns{display: flex; gap: min(4vw, 1.5em);}
    div.column{flex: auto; overflow-x: auto;}
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
    ul.task-list{list-style: none;}
    ul.task-list li input[type="checkbox"] {
      width: 0.8em;
      margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
      vertical-align: middle;
    }
    /* CSS for syntax highlighting */
    pre > code.sourceCode { white-space: pre; position: relative; }
    pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
    pre > code.sourceCode > span:empty { height: 1.2em; }
    .sourceCode { overflow: visible; }
    code.sourceCode > span { color: inherit; text-decoration: inherit; }
    div.sourceCode { margin: 1em 0; }
    pre.sourceCode { margin: 0; }
    @media screen {
    div.sourceCode { overflow: auto; }
    }
    @media print {
    pre > code.sourceCode { white-space: pre-wrap; }
    pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
    }
    pre.numberSource code
      { counter-reset: source-line 0; }
    pre.numberSource code > span
      { position: relative; left: -4em; counter-increment: source-line; }
    pre.numberSource code > span > a:first-child::before
      { content: counter(source-line);
        position: relative; left: -1em; text-align: right; vertical-align: baseline;
        border: none; display: inline-block;
        -webkit-touch-callout: none; -webkit-user-select: none;
        -khtml-user-select: none; -moz-user-select: none;
        -ms-user-select: none; user-select: none;
        padding: 0 4px; width: 4em;
        color: #aaaaaa;
      }
    pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
    div.sourceCode
      { color: #003b4f; background-color: #f1f3f5; }
    @media screen {
    pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
    }
    code span { color: #003b4f; } /* Normal */
    code span.al { color: #ad0000; } /* Alert */
    code span.an { color: #5e5e5e; } /* Annotation */
    code span.at { color: #657422; } /* Attribute */
    code span.bn { color: #ad0000; } /* BaseN */
    code span.bu { } /* BuiltIn */
    code span.cf { color: #003b4f; } /* ControlFlow */
    code span.ch { color: #20794d; } /* Char */
    code span.cn { color: #8f5902; } /* Constant */
    code span.co { color: #5e5e5e; } /* Comment */
    code span.cv { color: #5e5e5e; font-style: italic; } /* CommentVar */
    code span.do { color: #5e5e5e; font-style: italic; } /* Documentation */
    code span.dt { color: #ad0000; } /* DataType */
    code span.dv { color: #ad0000; } /* DecVal */
    code span.er { color: #ad0000; } /* Error */
    code span.ex { } /* Extension */
    code span.fl { color: #ad0000; } /* Float */
    code span.fu { color: #4758ab; } /* Function */
    code span.im { color: #00769e; } /* Import */
    code span.in { color: #5e5e5e; } /* Information */
    code span.kw { color: #003b4f; } /* Keyword */
    code span.op { color: #5e5e5e; } /* Operator */
    code span.ot { color: #003b4f; } /* Other */
    code span.pp { color: #ad0000; } /* Preprocessor */
    code span.sc { color: #5e5e5e; } /* SpecialChar */
    code span.ss { color: #20794d; } /* SpecialString */
    code span.st { color: #20794d; } /* String */
    code span.va { color: #111111; } /* Variable */
    code span.vs { color: #20794d; } /* VerbatimString */
    code span.wa { color: #5e5e5e; font-style: italic; } /* Warning */
    /* CSS for citations */
    div.csl-bib-body { }
    div.csl-entry {
      clear: both;
    }
    .hanging-indent div.csl-entry {
      margin-left:2em;
      text-indent:-2em;
    }
    div.csl-left-margin {
      min-width:2em;
      float:left;
    }
    div.csl-right-inline {
      margin-left:2em;
      padding-left:1em;
    }
    div.csl-indent {
      margin-left: 2em;
    }  </style>
  <link rel="stylesheet" href="../../site_libs/revealjs/dist/theme/quarto.css">
  <link href="../../site_libs/revealjs/plugin/quarto-line-highlight/line-highlight.css" rel="stylesheet">
  <link href="../../site_libs/revealjs/plugin/reveal-menu/menu.css" rel="stylesheet">
  <link href="../../site_libs/revealjs/plugin/reveal-menu/quarto-menu.css" rel="stylesheet">
  <link href="../../site_libs/revealjs/plugin/reveal-chalkboard/font-awesome/css/all.css" rel="stylesheet">
  <link href="../../site_libs/revealjs/plugin/reveal-chalkboard/style.css" rel="stylesheet">
  <link href="../../site_libs/revealjs/plugin/quarto-support/footer.css" rel="stylesheet">
  <style type="text/css">

  .callout {
    margin-top: 1em;
    margin-bottom: 1em;  
    border-radius: .25rem;
  }

  .callout.callout-style-simple { 
    padding: 0em 0.5em;
    border-left: solid #acacac .3rem;
    border-right: solid 1px silver;
    border-top: solid 1px silver;
    border-bottom: solid 1px silver;
    display: flex;
  }

  .callout.callout-style-default {
    border-left: solid #acacac .3rem;
    border-right: solid 1px silver;
    border-top: solid 1px silver;
    border-bottom: solid 1px silver;
  }

  .callout .callout-body-container {
    flex-grow: 1;
  }

  .callout.callout-style-simple .callout-body {
    font-size: 1rem;
    font-weight: 400;
  }

  .callout.callout-style-default .callout-body {
    font-size: 0.9rem;
    font-weight: 400;
  }

  .callout.callout-titled.callout-style-simple .callout-body {
    margin-top: 0.2em;
  }

  .callout:not(.callout-titled) .callout-body {
      display: flex;
  }

  .callout:not(.no-icon).callout-titled.callout-style-simple .callout-content {
    padding-left: 1.6em;
  }

  .callout.callout-titled .callout-header {
    padding-top: 0.2em;
    margin-bottom: -0.2em;
  }

  .callout.callout-titled .callout-title  p {
    margin-top: 0.5em;
    margin-bottom: 0.5em;
  }
    
  .callout.callout-titled.callout-style-simple .callout-content  p {
    margin-top: 0;
  }

  .callout.callout-titled.callout-style-default .callout-content  p {
    margin-top: 0.7em;
  }

  .callout.callout-style-simple div.callout-title {
    border-bottom: none;
    font-size: .9rem;
    font-weight: 600;
    opacity: 75%;
  }

  .callout.callout-style-default  div.callout-title {
    border-bottom: none;
    font-weight: 600;
    opacity: 85%;
    font-size: 0.9rem;
    padding-left: 0.5em;
    padding-right: 0.5em;
  }

  .callout.callout-style-default div.callout-content {
    padding-left: 0.5em;
    padding-right: 0.5em;
  }

  .callout.callout-style-simple .callout-icon::before {
    height: 1rem;
    width: 1rem;
    display: inline-block;
    content: "";
    background-repeat: no-repeat;
    background-size: 1rem 1rem;
  }

  .callout.callout-style-default .callout-icon::before {
    height: 0.9rem;
    width: 0.9rem;
    display: inline-block;
    content: "";
    background-repeat: no-repeat;
    background-size: 0.9rem 0.9rem;
  }

  .callout-title {
    display: flex
  }
    
  .callout-icon::before {
    margin-top: 1rem;
    padding-right: .5rem;
  }

  .callout.no-icon::before {
    display: none !important;
  }

  .callout.callout-titled .callout-body > .callout-content > :last-child {
    margin-bottom: 0.5rem;
  }

  .callout.callout-titled .callout-icon::before {
    margin-top: .5rem;
    padding-right: .5rem;
  }

  .callout:not(.callout-titled) .callout-icon::before {
    margin-top: 1rem;
    padding-right: .5rem;
  }

  /* Callout Types */

  div.callout-note {
    border-left-color: #4582ec !important;
  }

  div.callout-note .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAAEU0lEQVRYCcVXTWhcVRQ+586kSUMMxkyaElstCto2SIhitS5Ek8xUKV2poatCcVHtUlFQk8mbaaziwpWgglJwVaquitBOfhQXFlqlzSJpFSpIYyXNjBNiTCck7x2/8/LeNDOZxDuEkgOXe++553zfefee+/OYLOXFk3+1LLrRdiO81yNqZ6K9cG0P3MeFaMIQjXssE8Z1JzLO9ls20MBZX7oG8w9GxB0goaPrW5aNMp1yOZIa7Wv6o2ykpLtmAPs/vrG14Z+6d4jpbSKuhdcSyq9wGMPXjonwmESXrriLzFGOdDBLB8Y6MNYBu0dRokSygMA/mrun8MGFN3behm6VVAwg4WR3i6FvYK1T7MHo9BK7ydH+1uurECoouk5MPRyVSBrBHMYwVobG2aOXM07sWrn5qgB60rc6mcwIDJtQrnrEr44kmy+UO9r0u9O5/YbkS9juQckLed3DyW2XV/qWBBB3ptvI8EUY3I9p/67OW+g967TNr3Sotn3IuVlfMLVnsBwH4fsnebJvyGm5GeIUA3jljERmrv49SizPYuq+z7c2H/jlGC+Ghhupn/hcapqmcudB9jwJ/3jvnvu6vu5lVzF1fXyZuZZ7U8nRmVzytvT+H3kilYvH09mLWrQdwFSsFEsxFVs5fK7A0g8gMZjbif4ACpKbjv7gNGaD8bUrlk8x+KRflttr22JEMRUbTUwwDQScyzPgedQHZT0xnx7ujw2jfVfExwYHwOsDTjLdJ2ebmeQIlJ7neo41s/DrsL3kl+W2lWvAga0tR3zueGr6GL78M3ifH0rGXrBC2aAR8uYcIA5gwV8zIE8onoh8u0Fca/ciF7j1uOzEnqcIm59sEXoGc0+z6+H45V1CvAvHcD7THztu669cnp+L0okAeIc6zjbM/24LgGM1gZk7jnRu1aQWoU9sfUOuhrmtaPIO3YY1KLLWZaEO5TKUbMY5zx8W9UJ6elpLwKXbsaZ4EFl7B4bMtDv0iRipKoDQT2sNQI9b1utXFdYisi+wzZ/ri/1m7QfDgEuvgUUEIJPq3DhX/5DWNqIXDOweC2wvIR90Oq3lDpdMIgD2r0dXvGdsEW5H6x6HLRJYU7C69VefO1x8Gde1ZFSJLfWS1jbCnhtOPxmpfv2LXOA2Xk2tvnwKKPFuZ/oRmwBwqRQDcKNeVQkYcOjtWVBuM/JuYw5b6isojIkYxyYAFn5K7ZBF10fea52y8QltAg6jnMqNHFBmGkQ1j+U43HMi2xMar1Nv0zGsf1s8nUsmUtPOOrbFIR8bHFDMB5zL13Gmr/kGlCkUzedTzzmzsaJXhYawnA3UmARpiYj5ooJZiUoxFRtK3X6pgNPv+IZVPcnwbOl6f+aBaO1CNvPW9n9LmCp01nuSaTRF2YxHqZ8DYQT6WsXT+RD6eUztwYLZ8rM+rcPxamv1VQzFUkzFXvkiVrySGQgJNvXHJAxiU3/NwiC03rSf05VBaPtu/Z7/B8Yn/w7eguloAAAAAElFTkSuQmCC');
  }

  div.callout-note.callout-style-default .callout-title {
    background-color: #dae6fb
  }

  div.callout-important {
    border-left-color: #d9534f !important;
  }

  div.callout-important .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAAEKklEQVRYCcVXTWhcVRS+575MJym48A+hSRFr00ySRQhURRfd2HYjk2SSTokuBCkU2o0LoSKKraKIBTcuFCoidGFD08nkBzdREbpQ1EDNIv8qSGMFUboImMSZd4/f9zJv8ibJMC8xJQfO3HPPPef7zrvvvnvviIkpC9nsw0UttFunbUhpFzFtarSd6WJkStVMw5xyVqYTvkwfzuf/5FgtkVoB0729j1rjXwThS7Vio+Mo6DNnvLfahoZ+i/o32lULuJ3NNiz7q6+pyAUkJaFF6JwaM2lUJlV0MlnQn5aTRbEu0SEqHUa0A4AdiGuB1kFXRfVyg5d87+Dg4DL6m2TLAub60ilj7A1Ec4odSAc8X95sHh7+ZRPCFo6Fnp7HfU/fBng/hi10CjCnWnJjsxvDNxWw0NfV6Rv5GgP3I3jGWXumdTD/3cbEOP2ZbOZp69yniG3FQ9z1jD7bnBu9Fc2tKGC2q+uAJOQHBDRiZX1x36o7fWBs7J9ownbtO+n0/qWkvW7UPIfc37WgT6ZGR++EOJyeQDSb9UB+DZ1G6DdLDzyS+b/kBCYGsYgJbSQHuThGKRcw5xdeQf8YdNHsc6ePXrlSYMBuSIAFTGAtQo+VuALo4BX83N190NWZWbynBjhOHsmNfFWLeL6v+ynsA58zDvvAC8j5PkbOcXCMg2PZFk3q8MjI7WAG/Dp9AwP7jdGBOOQkAvlFUB+irtm16I1Zw9YBcpGTGXYmk3kQIC/Cds55l+iMI3jqhjAuaoe+am2Jw5GT3Nbz3CkE12NavmzN5+erJW7046n/CH1RO/RVa8lBLozXk9uqykkGAyRXLWlLv5jyp4RFsG5vGVzpDLnIjTWgnRy2Rr+tDKvRc7Y8AyZq10jj8DqXdnIRNtFZb+t/ZRtXcDiVnzpqx8mPcDWxgARUqx0W1QB9MeUZiNrV4qP+Ehc+BpNgATsTX8ozYKL2NtFYAHc84fG7ndxUPr+AR/iQSns7uSUufAymwDOb2+NjK27lEFocm/EE2WpyIy/Hi66MWuMKJn8RvxIcj87IM5Vh9663ziW36kR0HNenXuxmfaD8JC7tfKbrhFr7LiZCrMjrzTeGx+PmkosrkNzW94ObzwocJ7A1HokLolY+AvkTiD/q1H0cN48c5EL8Crkttsa/AXQVDmutfyku0E7jShx49XqV3MFK8IryDhYVbj7Sj2P2eBxwcXoe8T8idsKKPRcnZw1b+slFTubwUwhktrfnAt7J++jwQtLZcm3sr9LQrjRzz6cfMv9aLvgmnAGvpoaGLxM4mAEaLV7iAzQ3oU0IvD5x9ix3yF2RAAuYAOO2f7PEFWCXZ4C9Pb2UsgDeVnFSpbFK7/IWu7TPTvBqzbGdCHOJQSxiEjt6IyZmxQyEJHv6xyQsYk//moVFsN2zP6fRImjfq7/n/wFDguUQFNEwugAAAABJRU5ErkJggg==');
  }

  div.callout-important.callout-style-default .callout-title {
    background-color: #f7dddc
  }

  div.callout-warning {
    border-left-color: #f0ad4e !important;
  }

  div.callout-warning .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAAETklEQVRYCeVWW2gcVRg+58yaTUnizqbipZeX4uWhBEniBaoUX1Ioze52t7sRq6APio9V9MEaoWlVsFasRq0gltaAPuxms8lu0gcviE/FFOstVbSIxgcv6SU7EZqmdc7v9+9mJtNks51NTUH84ed889/PP+cmxP+d5FIbMJmNbpREu4WUkiTtCicKny0l1pIKmBzovF2S+hIJHX8iEu3hZJ5lNZGqyRrGSIQpq15AzF28jgpeY6yk6GVdrfFqdrD6Iw+QlB8g0YS2g7dyQmXM/IDhBhT0UCiRf59lfqmmDvzRt6kByV/m4JjtzuaujMUM2c5Z2d6JdKrRb3K2q6mA+oYVz8JnDdKPmmNthzkAk/lN63sYPgevrguc72aZX/L9C6x09GYyxBgCX4NlvyGUHOKELlm5rXeR1kchuChJt4SSwyddZRXgvwMGvYo4QSlk3/zkHD8UHxwVJA6zjZZqP8v8kK8OWLnIZtLyCAJagYC4rTGW/9Pqj92N/c+LUaAj27movwbi19tk/whRCIE7Q9vyI6yvRpftAKVTdUjOW40X3h5OXsKCdmFcx0xlLJoSuQngnrJe7Kcjm4OMq9FlC7CMmScQANuNvjfP3PjGXDBaUQmbp296S5L4DrpbrHN1T87ZVEZVCzg1FF0Ft+dKrlLukI+/c9ENo+TvlTDbYFvuKPtQ9+l052rXrgKoWkDAFnvh0wTOmYn8R5f4k/jN/fZiCM1tQx9jQQ4ANhqG4hiL0qIFTGViG9DKB7GYzgubnpofgYRwO+DFjh0Zin2m4b/97EDkXkc+f6xYAPX0KK2I/7fUQuwzuwo/L3AkcjugPNixC8cHf0FyPjWlItmLxWw4Ou9YsQCr5fijMGoD/zpdRy95HRysyXA74MWOnscpO4j2y3HAVisw85hX5+AFBRSHt4ShfLFkIMXTqyKFc46xdzQM6XbAi702a7sy04J0+feReMFKp5q9esYLCqAZYw/k14E/xcLLsFElaornTuJB0svMuJINy8xkIYuL+xPAlWRceH6+HX7THJ0djLUom46zREu7tTkxwmf/FdOZ/sh6Q8qvEAiHpm4PJ4a/doJe0gH1t+aHRgCzOvBvJedEK5OFE5jpm4AGP2a8Dxe3gGJ/pAutug9Gp6he92CsSsWBaEcxGx0FHytmIpuqGkOpldqNYQK8cSoXvd+xLxXADw0kf6UkJNFtdo5MOgaLjiQOQHcn+A6h5NuL2s0qsC2LOM75PcF3yr5STuBSAcGG+meA14K/CI21HcS4LBT6tv0QAh8Dr5l93AhZzG5ZJ4VxAqdZUEl9z7WJ4aN+svMvwHHL21UKTd1mqvChH7/Za5xzXBBKrUcB0TQ+Ulgkfbi/H/YT5EptrGzsEK7tR1B7ln9BBwckYfMiuSqklSznIuoIIOM42MQO+QnduCoFCI0bpkzjCjddHPN/F+2Yu+sd9bKNpVwHhbS3LluK/0zgfwD0xYI5dXuzlQAAAABJRU5ErkJggg==');
  }

  div.callout-warning.callout-style-default .callout-title {
    background-color: #fcefdc
  }

  div.callout-tip {
    border-left-color: #02b875 !important;
  }

  div.callout-tip .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAADr0lEQVRYCe1XTWgTQRj9ZjZV8a9SPIkKgj8I1bMHsUWrqYLVg4Ue6v9BwZOxSYsIerFao7UiUryIqJcqgtpimhbBXoSCVxUFe9CTiogUrUp2Pt+3aUI2u5vdNh4dmMzOzHvvezuz8xNFM0mjnbXaNu1MvFWRXkXEyE6aYOYJpdW4IXuA4r0fo8qqSMDBU0v1HJUgVieAXxzCsdE/YJTdFcVIZQNMyhruOMJKXYFoLfIfIvVIMWdsrd+Rpd86ZmyzzjJmLStqRn0v8lzkb4rVIXvnpScOJuAn2ACC65FkPzEdEy4TPWRLJ2h7z4cArXzzaOdKlbOvKKX25Wl00jSnrwVxAg3o4dRxhO13RBSdNvH0xSARv3adTXbBdTf64IWO2vH0LT+cv4GR1DJt+DUItaQogeBX/chhbTBxEiZ6gftlDNXTrvT7co4ub5A6gp9HIcHvzTa46OS5fBeP87Qm0fQkr4FsYgVQ7Qg+ZayaDg9jhg1GkWj8RG6lkeSacrrHgDaxdoBiZPg+NXV/KifMuB6//JmYH4CntVEHy/keA6x4h4CU5oFy8GzrBS18cLJMXcljAKB6INjWsRcuZBWVaS3GDrqB7rdapVIeA+isQ57Eev9eCqzqOa81CY05VLd6SamW2wA2H3SiTbnbSxmzfp7WtKZkqy4mdyAlGx7ennghYf8voqp9cLSgKdqNfa6RdRsAAkPwRuJZNbpByn+RrJi1RXTwdi8RQF6ymDwGMAtZ6TVE+4uoKh+MYkcLsT0Hk8eAienbiGdjJHZTpmNjlbFJNKDVAp2fJlYju6IreQxQ08UJDNYdoLSl6AadO+fFuCQqVMB1NJwPm69T04Wv5WhfcWyfXQB+wXRs1pt+nCknRa0LVzSA/2B+a9+zQJadb7IyyV24YAxKp2Jqs3emZTuNnKxsah+uabKbMk7CbTgJx/zIgQYErIeTKRQ9yD9wxVof5YolPHqaWo7TD6tJlh7jQnK5z2n3+fGdggIOx2kaa2YI9QWarc5Ce1ipNWMKeSG4DysFF52KBmTNMmn5HqCFkwy34rDg05gDwgH3bBi+sgFhN/e8QvRn8kbamCOhgrZ9GJhFDgfcMHzFb6BAtjKpFhzTjwv1KCVuxHvCbsSiEz4CANnj84cwHdFXAbAOJ4LTSAawGWFn5tDhLMYz6nWeU2wJfIhmIJBefcd/A5FWQWGgrWzyORZ3Q6HuV+Jf0Bj+BTX69fm1zWgK7By1YTXchFDORywnfQ7GpzOo6S+qECrsx2ifVQAAAABJRU5ErkJggg==');
  }

  div.callout-tip.callout-style-default .callout-title {
    background-color: #ccf1e3
  }

  div.callout-caution {
    border-left-color: #fd7e14 !important;
  }

  div.callout-caution .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAACV0lEQVRYCdVWzWoUQRCuqp2ICBLJXgITZL1EfQDBW/bkzUMUD7klD+ATSHBEfAIfQO+iXsWDxJsHL96EHAwhgzlkg8nBg25XWb0zIb0zs9muYYWkoKeru+vn664fBqElyZNuyh167NXJ8Ut8McjbmEraKHkd7uAnAFku+VWdb3reSmRV8PKSLfZ0Gjn3a6Xlcq9YGb6tADjn+lUfTXtVmaZ1KwBIvFI11rRXlWlatwIAAv2asaa9mlB9wwygiDX26qaw1yYPzFXg2N1GgG0FMF8Oj+VIx7E/03lHx8UhvYyNZLN7BwSPgekXXLribw7w5/c8EF+DBK5idvDVYtEEwMeYefjjLAdEyQ3M9nfOkgnPTEkYU+sxMq0BxNR6jExrAI31H1rzvLEfRIdgcv1XEdj6QTQAS2wtstEALLG1yEZ3QhH6oDX7ExBSFEkFINXH98NTrme5IOaaA7kIfiu2L8A3qhH9zRbukdCqdsA98TdElyeMe5BI8Rs2xHRIsoTSSVFfCFCWGPn9XHb4cdobRIWABNf0add9jakDjQJpJ1bTXOJXnnRXHRf+dNL1ZV1MBRCXhMbaHqGI1JkKIL7+i8uffuP6wVQAzO7+qVEbF6NbS0LJureYcWXUUhH66nLR5rYmva+2tjRFtojkM2aD76HEGAD3tPtKM309FJg5j/K682ywcWJ3PASCcycH/22u+Bh7Aa0ehM2Fu4z0SAE81HF9RkB21c5bEn4Dzw+/qNOyXr3DCTQDMBOdhi4nAgiFDGCinIa2owCEChUwD8qzd03PG+qdW/4fDzjUMcE1ZpIAAAAASUVORK5CYII=');
  }

  div.callout-caution.callout-style-default .callout-title {
    background-color: #ffe5d0
  }

  </style>
  <style type="text/css">
    .reveal div.sourceCode {
      margin: 0;
      overflow: auto;
    }
    .reveal div.hanging-indent {
      margin-left: 1em;
      text-indent: -1em;
    }
    .reveal .slide:not(.center) {
      height: 100%;
    }
    .reveal .slide.scrollable {
      overflow-y: auto;
    }
    .reveal .footnotes {
      height: 100%;
      overflow-y: auto;
    }
    .reveal .slide .absolute {
      position: absolute;
      display: block;
    }
    .reveal .footnotes ol {
      counter-reset: ol;
      list-style-type: none; 
      margin-left: 0;
    }
    .reveal .footnotes ol li:before {
      counter-increment: ol;
      content: counter(ol) ". "; 
    }
    .reveal .footnotes ol li > p:first-child {
      display: inline-block;
    }
    .reveal .slide ul,
    .reveal .slide ol {
      margin-bottom: 0.5em;
    }
    .reveal .slide ul li,
    .reveal .slide ol li {
      margin-top: 0.4em;
      margin-bottom: 0.2em;
    }
    .reveal .slide ul[role="tablist"] li {
      margin-bottom: 0;
    }
    .reveal .slide ul li > *:first-child,
    .reveal .slide ol li > *:first-child {
      margin-block-start: 0;
    }
    .reveal .slide ul li > *:last-child,
    .reveal .slide ol li > *:last-child {
      margin-block-end: 0;
    }
    .reveal .slide .columns:nth-child(3) {
      margin-block-start: 0.8em;
    }
    .reveal blockquote {
      box-shadow: none;
    }
    .reveal .tippy-content>* {
      margin-top: 0.2em;
      margin-bottom: 0.7em;
    }
    .reveal .tippy-content>*:last-child {
      margin-bottom: 0.2em;
    }
    .reveal .slide > img.stretch.quarto-figure-center,
    .reveal .slide > img.r-stretch.quarto-figure-center {
      display: block;
      margin-left: auto;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-left,
    .reveal .slide > img.r-stretch.quarto-figure-left  {
      display: block;
      margin-left: 0;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-right,
    .reveal .slide > img.r-stretch.quarto-figure-right  {
      display: block;
      margin-left: auto;
      margin-right: 0; 
    }
  </style>
<meta property="og:title" content="Biologia della Conservazione 23/24 - Biologia della Conservazione">
<meta property="og:description" content="Class 7 - Genetic Drift/Effective Population Size">
<meta property="og:site-name" content="Biologia della Conservazione 23/24">
<meta property="og:locale" content="en_US">
</head>
<body class="quarto-light">
  <div class="reveal">
    <div class="slides">

<section id="title-slide" class="quarto-title-block center">
  <h1 class="title">Biologia della Conservazione</h1>
  <p class="subtitle">Class 7 - Genetic Drift/Effective Population Size</p>

<div class="quarto-title-authors">
</div>

  <p class="date">2024-04-11</p>
</section>
<section id="topics" class="slide level2">
<h2>Topics</h2>
<div class="columns">
<div class="column" style="width:50%;">
<ol type="1">
<li><span style="color: green;">Introduction to conservation biology</span></li>
<li><span style="color: green;">Biodiversity</span></li>
<li><span style="color: green;">Biodiversity and extinctions</span></li>
<li><span style="color: green;">Populations</span></li>
<li><span style="color: green;">Basics of conservation genetics</span></li>
<li><span style="color: orange;">Genetic drift</span></li>
<li><span style="color: orange;">Effective population size</span></li>
<li><span style="color: blue;">Inbreeding depression</span></li>
<li><span style="color: blue;">Population structure</span></li>
<li><span style="color: blue;">Hybridization</span></li>
<li><span style="color: blue;">Applying genetic information to conservation</span></li>
</ol>
</div><div class="column" style="width:50%;">
<ol start="12" type="1">
<li><span style="color: blue;">Managing genetic diversity in captive populations</span></li>
<li><span style="color: blue;">Managing genetic diversity in the wild</span></li>
<li><span style="color: blue;">Population viability analysis</span></li>
<li><span style="color: blue;">Invasive species</span></li>
<li><span style="color: blue;">CITES and trade regulations of endangered species</span></li>
<li><span style="color: blue;">Conservation and climate change</span></li>
<li><span style="color: blue;">Future climate impacts</span></li>
<li><span style="color: blue;">Conservation economics</span></li>
<li><span style="color: blue;">Conservation technology</span></li>
</ol>
</div>
</div>
</section>
<section class="slide level2">

<h3 id="sampling-error">Sampling error</h3>

<img data-src="./samplingError.png" class="r-stretch quarto-figure-center"></section>
<section class="slide level2">

<h3 id="when-not-to-use-the-chi2-test">When not to use the <span class="math inline">\(\chi^2\)</span> test</h3>
<ul>
<li>The expected count of individuals is &lt; 5 in any of the possible genotype considered</li>
<li>Very low allele frequencies</li>
<li>The sample size is too small</li>
<li>There are too many alleles at a locus</li>
</ul>
<h3 id="a-valid-alternative-fishers-exact-test">A valid alternative: Fisher’s exact test</h3>
<ul>
<li>Based on the calculation of probabilities</li>
<li><span class="math inline">\(P(genotype\ counts|allele \ counts)\)</span> assuming HWE</li>
<li>Let’s work out an example with a small population: 4 individuals <span class="math inline">\(AA\)</span>, 2 individuals <span class="math inline">\(Aa\)</span> and 1 individuals <span class="math inline">\(aa\)</span></li>
<li>What genotype configurations could give us the same allele count?</li>
</ul>
</section>
<section class="slide level2">

<table>
<thead>
<tr class="header">
<th style="text-align: center;"><span class="math inline">\(n_{AA}\)</span></th>
<th style="text-align: center;"><span class="math inline">\(n_{Aa}\)</span></th>
<th style="text-align: center;"><span class="math inline">\(n_{aa}\)</span></th>
<th style="text-align: center;"><span class="math inline">\(n_{A}\)</span></th>
<th style="text-align: center;"><span class="math inline">\(n_{a}\)</span></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: center;">5</td>
<td style="text-align: center;">0</td>
<td style="text-align: center;">2</td>
<td style="text-align: center;">10</td>
<td style="text-align: center;">4</td>
</tr>
<tr class="even">
<td style="text-align: center;">4</td>
<td style="text-align: center;">2</td>
<td style="text-align: center;">1</td>
<td style="text-align: center;">10</td>
<td style="text-align: center;">4</td>
</tr>
<tr class="odd">
<td style="text-align: center;">3</td>
<td style="text-align: center;">4</td>
<td style="text-align: center;">0</td>
<td style="text-align: center;">10</td>
<td style="text-align: center;">4</td>
</tr>
</tbody>
</table>
<div class="fragment">
<ul>
<li>Can we estimate the theoretical probability of each combination of genotypes?</li>
</ul>
</div>
<div class="fragment">
<p><span class="math display">\[
P(n_{Aa}|n_A, n_a, HWE)=\frac{n!}{n_{AA}!*n_{Aa}!*n_{aa}!}*\frac{2^{n_{Aa}}*n_A!*n_a!}{(2n)!}
\]</span></p>
</div>
<div class="fragment">
<table>
<thead>
<tr class="header">
<th style="text-align: center;"><span class="math inline">\(n_{AA}\)</span></th>
<th style="text-align: center;"><span class="math inline">\(n_{Aa}\)</span></th>
<th style="text-align: center;"><span class="math inline">\(n_{aa}\)</span></th>
<th style="text-align: center;">Theoretical probability</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: center;">5</td>
<td style="text-align: center;">0</td>
<td style="text-align: center;">2</td>
<td style="text-align: center;">0.021</td>
</tr>
<tr class="even">
<td style="text-align: center;">4</td>
<td style="text-align: center;">2</td>
<td style="text-align: center;">1</td>
<td style="text-align: center;">0.419</td>
</tr>
<tr class="odd">
<td style="text-align: center;">3</td>
<td style="text-align: center;">4</td>
<td style="text-align: center;">0</td>
<td style="text-align: center;">0.559</td>
</tr>
</tbody>
</table>
</div>
</section>
<section class="slide level2">

<h3 id="other-options-for-testing-null-hypothesis">Other options for testing null hypothesis</h3>
<ul>
<li>Likelihood, the probability of observing the data given some parameter value under a certain statistical model, particularly useful when there are many alleles at a single locus</li>
<li>Permutation tests, randomization of the dataset by sampling, particularly useful when many genotypes are expected to have very low frequency</li>
</ul>
<h3 id="multiple-tests">Multiple tests</h3>
<ul>
<li><p>If we examine 10 loci in 10 population samples, 100 tests of HW proportions will be performed. If all of these loci are in HW proportions (<em>i.e.</em>, our null hypothesis is true at all loci in all populations), we expect to find five significant tests if we use the 5% significance level <span class="citation" data-cites="Allendorf2022">(<a href="#/references" role="doc-biblioref" onclick="">Allendorf et al., 2022, pp. 102–103</a>)</span>.</p></li>
<li><p>The Bonferroni correction allows to adjust the significance level (say 5%) by dividing it by the number of tests performed. In the previous example <span class="math inline">\(0.05/100 = 0.0005\)</span>. The critical <span class="math inline">\(\chi^2\)</span> value for <span class="math inline">\(P = 0.0005\)</span> with one degree of freedom is 12.1.</p></li>
</ul>
</section>
<section class="slide level2">

<h3 id="what-we-learnt-from-the-hw-model">What we learnt from the HW model</h3>
<ul>
<li>Genetic change will not occur in populations if all the assumptions of the HW equilibrium are met (and we already know that the HW assumptions are almost never met in nature)</li>
<li>What is the effect of allele and genotype frequencies when the assumptions are not met?</li>
<li>We will start looking at two interrelated factors: genetic drift and population size</li>
</ul>
<h3 id="genetic-drift">Genetic Drift</h3>
<ul>
<li>Genetic drift is random change in allele frequencies from generation to generation because of sampling error. That is, the finite number of genes transmitted to progeny will be an imperfect sample of the allele frequencies in the previous generation <span class="citation" data-cites="Allendorf2022">(<a href="#/references" role="doc-biblioref" onclick="">Allendorf et al., 2022, p. 114</a>)</span></li>
<li>Genetic drift is the primary cause bringing about allele frequency changes throughout the genome over time <span class="citation" data-cites="Chen2018 Allendorf2022">(<a href="#/references" role="doc-biblioref" onclick="">Allendorf et al., 2022</a>; <a href="#/references" role="doc-biblioref" onclick="">Chen et al., 2018</a>)</span></li>
</ul>
</section>
<section class="slide level2">

<h3 id="genetic-drift-1">Genetic drift</h3>
<ul>
<li>The best way to understand how strong the change effect can be in a population is to model the sampling process in the absence of selection, mutation and migration.</li>
<li>We start sampling alleles to generate an offspring from a selfing diploid individual with genotype <span class="math inline">\(A_1A_2\)</span></li>
<li>Our gamete pool is <span class="math inline">\(A_1\)</span> with frequency <span class="math inline">\(p = \frac{1}{2}=0.5\)</span> and <span class="math inline">\(A_2\)</span> with frequency <span class="math inline">\(q=1-p=0.5\)</span></li>
<li>This example is very similar to the simple fair-coin toss! Imagine the coin has <span class="math inline">\(A_1\)</span> on one side and <span class="math inline">\(A_2\)</span> on the other side. To obtain a diploid individual we need to toss the coin twice. Every outcome combination has probability <span class="math inline">\(1/4\)</span></li>
<li>If only one offspring (<span class="math inline">\(N = 1\)</span>) is produced there is 50% chance that one or the other allele would be lost</li>
</ul>
</section>
<section class="slide level2">

<h3 id="genetic-drift-2">Genetic Drift</h3>
<ul>
<li>With many offspring (<span class="math inline">\(n &gt; 1\)</span>) there are many more coin tosses and the probability that they all are <span class="math inline">\(A_1\)</span> or <span class="math inline">\(A_2\)</span> now becomes <span class="math inline">\((\frac{1}{2})^{2n}\)</span>, and it will be less likely to loose one of the two alleles</li>
</ul>
<h3 id="exp.-distribution-of-allele-frequencies-in-population-of-size-n-2">Exp. distribution of allele frequencies in population of size <span class="math inline">\(n = 2\)</span></h3>
<div class="columns">
<div class="column" style="width:50%;">
<ul>
<li>Population with <span class="math inline">\(p = 0.6\)</span> and <span class="math inline">\(q =0.4\)</span></li>
<li>We sample <span class="math inline">\(2\)</span> individuals (<em>i.e.</em>, <span class="math inline">\(2n = 4\)</span>gametes)</li>
<li>The expected distribution of alleles is the binomial expansion <span class="math inline">\((p+q)^4\)</span></li>
<li>There are 5 possible outcomes</li>
</ul>
</div><div class="column" style="width:50%;">
<table>
<thead>
<tr class="header">
<th>Outcome</th>
<th>Expected frequency</th>
<th>Probability</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><span class="math inline">\(4 A_1,0A_2\)</span></td>
<td><span class="math inline">\(p^4\)</span></td>
<td><span class="math inline">\(0.6^4=0.129\)</span></td>
</tr>
<tr class="even">
<td><span class="math inline">\(3 A_1,1A_2\)</span></td>
<td><span class="math inline">\(4p^3q\)</span></td>
<td><span class="math inline">\(4*0.6^3*0.4=0.345\)</span></td>
</tr>
<tr class="odd">
<td><span class="math inline">\(2 A_1,2A_2\)</span></td>
<td><span class="math inline">\(6p^2q^2\)</span></td>
<td><span class="math inline">\(6*0.6^2*0.4^2=0.345\)</span></td>
</tr>
<tr class="even">
<td><span class="math inline">\(1 A_1,3A_2\)</span></td>
<td><span class="math inline">\(4pq^3\)</span></td>
<td><span class="math inline">\(4*0.6*0.4^3=0.153\)</span></td>
</tr>
<tr class="odd">
<td><span class="math inline">\(0 A_1,4A_2\)</span></td>
<td><span class="math inline">\(q^4\)</span></td>
<td><span class="math inline">\(0.4^4=0.025\)</span></td>
</tr>
</tbody>
</table>
</div>
</div>
</section>
<section class="slide level2">

<ul>
<li>The probability that a population has <span class="math inline">\(k\)</span> copies of allele <span class="math inline">\(A_1\)</span> and <span class="math inline">\(2n-k\)</span> copies of <span class="math inline">\(A_2\)</span> allele is <span class="math inline">\(\binom{2n}{k}p^kq^{2n-k}\)</span> where <span class="math inline">\(\binom{2n}{k}\)</span> is the binomial function <span class="math inline">\(\frac{2n!}{k!(2n-k)!}\)</span></li>
</ul>
<div class="fragment">
<div class="cell">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="class_7_files/figure-revealjs/expDist-1.png" width="960"></p>
<figcaption>Expected distribution of allele frequency from 100 replicates of sampling 10, 100 and 1000 individuals from a population with p = 0.6</figcaption>
</figure>
</div>
</div>
</div>
</div>
</section>
<section class="slide level2">

<ul>
<li>The variance around the mean allele frequency is the binomial sampling variance: <span class="math inline">\(\sigma^2_p =\frac{p_0q_0}{2n}\)</span></li>
<li>Variances are higher in smaller than larger populations</li>
<li>In small population the process of sampling happens every single generation, and its effect on allele frequency is cumulative:</li>
<li>Genetic drift will eventually cause all except one allele to go to fixation.</li>
<li>The probability that a gamete does not contain allele <span class="math inline">\(A_1\)</span> (with frequency <span class="math inline">\(p\)</span> in the parental population) is <span class="math inline">\(1-p\)</span></li>
</ul>
<p><span class="math display">\[
P(fixation \ A_2)=(1-p)^{2n}
\]</span></p>
</section>
<section class="slide level2">

<div class="callout callout-important callout-titled callout-style-default">
<div class="callout-body">
<div class="callout-title">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<p><strong>Problem</strong></p>
</div>
<div class="callout-content">
<p>Diploid population with <span class="math inline">\(n = 4\)</span> with allele <span class="math inline">\(A_1\)</span> with a frequency <span class="math inline">\(p = 0.25\)</span> and allele <span class="math inline">\(A_2\)</span> with a frequency <span class="math inline">\(q = 0.75\)</span>. What is the probability of losing allele <span class="math inline">\(A_1\)</span> in one generation of sampling to recreate a population of the same size?</p>
</div>
</div>
</div>
<div class="fragment">
<div class="callout callout-tip callout-titled callout-style-default">
<div class="callout-body">
<div class="callout-title">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<p><strong>Solution</strong></p>
</div>
<div class="callout-content">
<p><span class="math inline">\((1-0.25)^8=\)</span> 0.1001129</p>
</div>
</div>
</div>
</div>
</section>
<section class="slide level2">

<h3 id="genetic-drift-3">Genetic Drift</h3>

<img data-src="class_7_files/figure-revealjs/drift1-1.png" width="960" class="r-stretch quarto-figure-center"><p class="caption">Allele frequency distribution at t = 0, with N = 100 <span class="citation" data-cites="Russo2021">(<a href="#/references" role="doc-biblioref" onclick="">Russo, Colosimo, Fanelli, Franceschini, &amp; Parisi, 2021</a>)</span>.</p></section>
<section class="slide level2">

<h3 id="genetic-drift-4">Genetic Drift</h3>

<img data-src="class_7_files/figure-revealjs/drift2-1.png" width="960" class="r-stretch quarto-figure-center"><p class="caption">Allele frequency distribution at t = 1, with N = 100 <span class="citation" data-cites="Russo2021">(<a href="#/references" role="doc-biblioref" onclick="">Russo et al., 2021</a>)</span>.</p></section>
<section class="slide level2">

<h3 id="genetic-drift-5">Genetic Drift</h3>

<img data-src="class_7_files/figure-revealjs/drift3-1.png" width="960" class="r-stretch quarto-figure-center"><p class="caption">Allele frequencies variation after 200 generations while maintaining N = 100 <span class="citation" data-cites="Russo2021">(<a href="#/references" role="doc-biblioref" onclick="">Russo et al., 2021</a>)</span>.</p></section>
<section class="slide level2">

<h3 id="genetic-drift-6">Genetic Drift</h3>
<h4 id="simulating-the-average-time-of-allele-fixation-in-our-example">Simulating the average time of allele fixation in our example</h4>
<div class="cell" data-layout-align="center" data-hash="class_7_cache/revealjs/drift4_04abd7597ea54141f3ba03bafcf83714">
<details>
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1"></a><span class="co"># simulare il tempo medio di fissazione degli alleli</span></span>
<span id="cb1-2"><a href="#cb1-2"></a><span class="co">#pop.ls &lt;- list(geneA = c(rep("A",50), rep("a",50)), </span></span>
<span id="cb1-3"><a href="#cb1-3"></a><span class="co">#               geneB = c(rep("B",50), rep("b",50)),</span></span>
<span id="cb1-4"><a href="#cb1-4"></a><span class="co">#               geneC = c(rep("C",50), rep("c",50)))</span></span>
<span id="cb1-5"><a href="#cb1-5"></a></span>
<span id="cb1-6"><a href="#cb1-6"></a></span>
<span id="cb1-7"><a href="#cb1-7"></a><span class="co"># pop.ls  =  lista contenente i geni della popolazione</span></span>
<span id="cb1-8"><a href="#cb1-8"></a><span class="co"># num.sim = numero di volte la simulazione deve girare</span></span>
<span id="cb1-9"><a href="#cb1-9"></a>sim.gen2fix <span class="ot">&lt;-</span> <span class="cf">function</span> (pop.ls, num.sims) {</span>
<span id="cb1-10"><a href="#cb1-10"></a>  </span>
<span id="cb1-11"><a href="#cb1-11"></a>  <span class="co"># numero totale di loci per ogni gene   </span></span>
<span id="cb1-12"><a href="#cb1-12"></a>  n <span class="ot">&lt;-</span> <span class="fu">length</span>(pop.ls[[<span class="dv">1</span>]])</span>
<span id="cb1-13"><a href="#cb1-13"></a>  <span class="co"># numero di geni differenti nella popolazione</span></span>
<span id="cb1-14"><a href="#cb1-14"></a>  n.genes <span class="ot">&lt;-</span> <span class="fu">length</span>(pop.ls)</span>
<span id="cb1-15"><a href="#cb1-15"></a>  <span class="co"># lista vuota per i risultati della simulazione</span></span>
<span id="cb1-16"><a href="#cb1-16"></a>  generations <span class="ot">&lt;-</span> <span class="fu">vector</span>(<span class="at">mode =</span> <span class="st">"list"</span>, <span class="at">length =</span> <span class="fu">length</span>(pop.ls))</span>
<span id="cb1-17"><a href="#cb1-17"></a>  <span class="cf">for</span> (h <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>n.genes){</span>
<span id="cb1-18"><a href="#cb1-18"></a>    generations[[h]] <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="dv">0</span>, num.sims)</span>
<span id="cb1-19"><a href="#cb1-19"></a>  }</span>
<span id="cb1-20"><a href="#cb1-20"></a>  <span class="fu">names</span>(generations) <span class="ot">&lt;-</span> <span class="fu">names</span>(pop.ls)</span>
<span id="cb1-21"><a href="#cb1-21"></a>  </span>
<span id="cb1-22"><a href="#cb1-22"></a>  <span class="co"># ciclo di simulazione </span></span>
<span id="cb1-23"><a href="#cb1-23"></a>  <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>num.sims) {</span>
<span id="cb1-24"><a href="#cb1-24"></a>    temp.pop.ls <span class="ot">&lt;-</span> pop.ls</span>
<span id="cb1-25"><a href="#cb1-25"></a>    <span class="cf">for</span> (y <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>n.genes) {</span>
<span id="cb1-26"><a href="#cb1-26"></a>      <span class="cf">while</span> (<span class="fu">max</span>(<span class="fu">table</span>(temp.pop.ls[[y]])) <span class="sc">&lt;</span> n) {</span>
<span id="cb1-27"><a href="#cb1-27"></a>        temp.pop.ls[[y]] <span class="ot">&lt;-</span> <span class="fu">sample</span>(temp.pop.ls[[y]],</span>
<span id="cb1-28"><a href="#cb1-28"></a>                                   <span class="dv">100</span>, <span class="at">replace =</span> T)</span>
<span id="cb1-29"><a href="#cb1-29"></a>        generations[[y]][i] <span class="ot">&lt;-</span> generations[[y]][i] <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb1-30"><a href="#cb1-30"></a>      }</span>
<span id="cb1-31"><a href="#cb1-31"></a>    }</span>
<span id="cb1-32"><a href="#cb1-32"></a>  }</span>
<span id="cb1-33"><a href="#cb1-33"></a>  </span>
<span id="cb1-34"><a href="#cb1-34"></a>    <span class="co"># statistiche descrittive dei risultati</span></span>
<span id="cb1-35"><a href="#cb1-35"></a>  df <span class="ot">&lt;-</span> <span class="fu">rbind</span>(<span class="fu">data.frame</span>(<span class="fu">lapply</span>(generations, mean)),</span>
<span id="cb1-36"><a href="#cb1-36"></a>        <span class="fu">data.frame</span>(<span class="fu">lapply</span>(generations, sd))) </span>
<span id="cb1-37"><a href="#cb1-37"></a>  <span class="fu">rownames</span>(df) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"mean"</span>, <span class="st">"sd"</span>)</span>
<span id="cb1-38"><a href="#cb1-38"></a></span>
<span id="cb1-39"><a href="#cb1-39"></a>  <span class="co"># ciclo per plottare i risultati</span></span>
<span id="cb1-40"><a href="#cb1-40"></a>  <span class="co"># modificare se i geni da plottare sono &gt; 4</span></span>
<span id="cb1-41"><a href="#cb1-41"></a>  <span class="fu">par</span>(<span class="at">mfrow =</span> <span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">2</span>))</span>
<span id="cb1-42"><a href="#cb1-42"></a>  <span class="cf">for</span> (z <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>n.genes) {</span>
<span id="cb1-43"><a href="#cb1-43"></a>    graph <span class="ot">&lt;-</span> generations[[z]]</span>
<span id="cb1-44"><a href="#cb1-44"></a>    <span class="fu">hist</span>(graph, </span>
<span id="cb1-45"><a href="#cb1-45"></a>         <span class="at">main =</span> <span class="fu">paste</span>(<span class="st">"Num. of generations to fix "</span>,</span>
<span id="cb1-46"><a href="#cb1-46"></a>                             <span class="fu">names</span>(generations[z])),</span>
<span id="cb1-47"><a href="#cb1-47"></a>         <span class="at">xlab =</span> <span class="st">"Generations"</span>)</span>
<span id="cb1-48"><a href="#cb1-48"></a>    <span class="fu">abline</span>(<span class="at">v=</span><span class="fu">mean</span>(graph), <span class="at">lwd=</span><span class="dv">2</span>, <span class="at">lty=</span><span class="dv">2</span>, <span class="at">col=</span><span class="st">"red"</span>)</span>
<span id="cb1-49"><a href="#cb1-49"></a>  }</span>
<span id="cb1-50"><a href="#cb1-50"></a>  </span>
<span id="cb1-51"><a href="#cb1-51"></a>  <span class="co">#return(df)</span></span>
<span id="cb1-52"><a href="#cb1-52"></a>}</span>
<span id="cb1-53"><a href="#cb1-53"></a></span>
<span id="cb1-54"><a href="#cb1-54"></a><span class="fu">sim.gen2fix</span>(pop.ls, <span class="dv">5000</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>

</div>
<img data-src="class_7_files/figure-revealjs/drift4-1.png" width="960" class="r-stretch quarto-figure-center"></section>
<section class="slide level2">

<h3 id="size-matters">Size matters</h3>

<img data-src="class_7_files/figure-revealjs/drift5-1.png" width="960" class="r-stretch"></section>
<section class="slide level2">

<h3 id="population-size">Population size</h3>
<blockquote>
<p>There can be no doubt that genetic variation is related to population size, as <span class="citation" data-cites="Soule1973">Soule (<a href="#/references" role="doc-biblioref" onclick="">1973</a>)</span> proposed. Small population size reduces the evolutionary potential of wildlife species <span class="citation" data-cites="Frankham1996">(<a href="#/references" role="doc-biblioref" onclick="">Frankham, 1996</a>)</span>.</p>
</blockquote>
<ul>
<li>The importance of maintaining a large population size is pretty intuitive, but may not be enough!</li>
<li>Many populations of insects and small mammals naturally fluctuate in their “census” population size, but it is when they are at their minimum that the genetic processes affect them the most</li>
<li>Often the census population counts juveniles and senescent individuals that do not really contribute to the reproductive output of next generations</li>
<li>Within adult and reproductive individuals there may be significant variance in the number of offspring sired by males and females</li>
</ul>
</section>
<section class="slide level2">

<h3 id="population-size-1">Population size</h3>
<ul>
<li>Different reproductive strategies (monogamy VS. polygamy) may as well influence the genetic output generation after generation</li>
<li>Therefore it is necessary to distinguish between census population and an hypothetical idealized population</li>
<li>We can standardize this concept of idealized population by describing it in term of <strong>effective population size</strong> (<span class="math inline">\(N_e\)</span>), and it can be defined as the actual number of breeding individuals needed in order to see the same variation of a critical parameter (such as genetic diversity) measured in the real population</li>
<li>The effective size of a population determines the rate of change in composition of a population caused by genetic drift (the random sampling of genetic variants) <span class="citation" data-cites="Charlesworth2009a">(<a href="#/references" role="doc-biblioref" onclick="">Charlesworth, 2009, p. 195</a>)</span></li>
<li>To better understand the origin of this concept of the idealized population see <span class="citation" data-cites="Wright1931">Wright (<a href="#/references" role="doc-biblioref" onclick="">1931</a>)</span></li>
</ul>
</section>
<section class="slide level2">

<h3 id="wright-fisher-population-model">Wright-Fisher population model</h3>
<p>The ideal population:</p>
<ul>
<li>All individual in the population are considered adults and capable of breeding</li>
<li>Individuals are hermafroditic (self fertilization is possible)</li>
<li>The number of breeding individual is constant in all generations and they breed randomly</li>
<li>New individuals are formed each generation by random sampling, with replacement, of the gametes from the parents</li>
<li>The parents die after mating so each generation is only composed by new individuals (discrete generations)</li>
<li>There is no selection at any life stage and mutation is ignored</li>
<li>When <span class="math inline">\(N\)</span> is sufficiently large this means that each individual contributes to the next generation a number of offspring that is approximated by a Poisson distribution</li>
</ul>
</section>
<section class="slide level2">

<h3 id="wright-fisher-population-model-1">Wright-Fisher population model</h3>
<ul>
<li>Any characteristic of a real population that deviates from the characteristic of the ideal population will make the census size <span class="math inline">\(N\)</span> to be different than the effective population size <span class="math inline">\(N_e\)</span></li>
<li>An important measure in conservation is the ratio <span class="math inline">\(\frac{N_e}{N}\)</span></li>
<li>There are multiple evidence that in most natural population this ratio is usually much less than 1.</li>
<li>The primary factors influencing this ratio are:
<ul>
<li>fluctuating population size over multiple generations</li>
<li>high variance in family size</li>
<li>unequal ratios between sexes</li>
</ul></li>
</ul>
</section>
<section class="slide level2">

<h3 id="estimating-n_e">Estimating <span class="math inline">\(N_e\)</span></h3>
<ul>
<li>To estimate <span class="math inline">\(N_e\)</span> it is possible to use demographic approaches and genetic approaches</li>
</ul>
<h4 id="demographic-approaches">Demographic approaches</h4>
<ul>
<li>Demographic methodologies require extensive data and sometimes are unattainable in wild populations</li>
</ul>
<div class="panel-tabset">
<ul id="tabset-1" class="panel-tabset-tabby"><li><a data-tabby-default="" href="#tabset-1-1">Fluctuations in population size</a></li><li><a href="#tabset-1-2">Variation in family size</a></li><li><a href="#tabset-1-3">Unequal sex-ratio</a></li></ul>
<div class="tab-content">
<div id="tabset-1-1">
<div class="columns">
<div class="column" style="width:50%;">
<ul>
<li>Natural populations continuously undergo fluctuations in size over multiple generations <span class="citation" data-cites="Reed2003b">(<a href="#/references" role="doc-biblioref" onclick="">Reed, OGrady, Ballou, &amp; Frankham, 2003</a>)</span></li>
<li>We can use the harmonic mean to estimate <span class="math inline">\(N_e\)</span> over <span class="math inline">\(t\)</span> generations</li>
</ul>
</div><div class="column" style="width:50%;">
<p><span class="math display">\[
N_e \sim \frac{t}{\sum(\frac{1}{N_{ei}})} \\ \sim \frac{3}{(1/175000)+(1/20)+(1/175000)} \sim 60
\]</span></p>
</div>
</div>
</div>
<div id="tabset-1-2">
<div class="columns">
<div class="column" style="width:50%;">
<ul>
<li>The effect on <span class="math inline">\(N_e\)</span> of variation in family sizes in an otherwise ideal population can be estimated using:</li>
</ul>
<p><span class="math display">\[
N_e = \frac{4N-2}{V_k +2}
\]</span></p>
</div><div class="column" style="width:50%;">
<ul>
<li><span class="math inline">\(k\)</span> is the mean family size</li>
<li>In Wright-Fisher population <span class="math inline">\(V_k = 2\)</span> so that <span class="math inline">\(N_e \sim N\)</span></li>
<li>In Darwin’s finches the value of <span class="math inline">\(V_k\)</span> has been estimated at 6.74, which reduces dramatically <span class="math inline">\(N_e\)</span> <span class="citation" data-cites="Frankham1995a">(<a href="#/references" role="doc-biblioref" onclick="">Frankham, 1995</a>)</span></li>
</ul>
</div>
</div>
</div>
<div id="tabset-1-3">
<div class="columns">
<div class="column" style="width:50%;">
<ul>
<li>According to <span class="citation" data-cites="Donald2007">Donald (<a href="#/references" role="doc-biblioref" onclick="">2007</a>)</span> there usually is an excess of females in mammals and a deficiency in males <span class="math display">\[
N_e=\frac{4N_{ef}N_{em}}{N_{ef}+N_{em}}
\]</span></li>
</ul>
</div><div class="column" style="width:50%;">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="./unequalsexratio.jpg" width="250"></p>
<figcaption>Effects of unequal sex ratio on <span class="math inline">\(N_e/N\)</span>. After <span class="citation" data-cites="Frankham2010">Frankham, Ballou, &amp; Briscoe (<a href="#/references" role="doc-biblioref" onclick="">2010</a>)</span>, p.249</figcaption>
</figure>
</div>
</div>
</div>
</div>
</div>
</div>
</section>
<section class="slide level2">

<h4 id="genetic-approaches">Genetic approaches</h4>
<ul>
<li>DNA retains information on the evolutionary history of individuals</li>
<li>We can use basic probability theory to extrapolate information, including information on <span class="math inline">\(N_e\)</span> from DNA markers</li>
<li>The coalescent approach uses the sampling theory for neutral alleles and allows us to trace back in time this information <span class="citation" data-cites="Rosenberg2002">(<a href="#/references" role="doc-biblioref" onclick="">Rosenberg &amp; Nordborg, 2002</a>)</span></li>
<li>Under the neutral theory of evolution 2 alleles may descend from the same ancestral allele in the previous generation with probability <span class="math inline">\(1/N_{ef}\)</span> for mtDNA, or <span class="math inline">\(1/2N_e\)</span> for a general diploid locus</li>
<li>Alternatively 2 alleles may derive from 2 different alleles in the previous generation with probability <span class="math inline">\(1-\frac{1}{N_{ef}}\)</span></li>
</ul>
</section>
<section class="slide level2">

<ul>
<li>Under this neutral model of genetic drift the coalescence process takes a characteristic time:</li>
</ul>
<p><span class="math display">\[
T_k = \frac{4N_e}{k(k-1)}generations
\]</span> where <span class="math inline">\(T_k\)</span> is the average time in generations back to the previous coalescent event when there were <span class="math inline">\(k-1\)</span> alleles <span class="citation" data-cites="Hedrick2005">(<a href="#/references" role="doc-biblioref" onclick="">Hedrick, 2005</a>)</span></p>
<ul>
<li>The time for all <span class="math inline">\(k\)</span> alleles in the population to coalesce back to one allele is: <span class="math inline">\(4N_e[1-(\frac{1}{k})]\)</span></li>
</ul>
</section>
<section class="slide level2">

<h3 id="the-coalescent-theory">The coalescent theory</h3>
<ul>
<li>The coalescent allows to study the loss of genetic diversity and to use this information to infer effective population size</li>
<li>To coalesce means “unite” or “fuse”</li>
<li>To better understand this approach we can look at the example of mitochondrial DNA (mtDNA)</li>
<li>mtDNA is haploid, maternally inherited and it does not undergo recombination</li>
<li>Coalescent theory provides a powerful framework to study the effects of genetic drift, natural selection, mutation, and gene flow in natural populations <span class="citation" data-cites="Allendorf2022 Rosenberg2002">(<a href="#/references" role="doc-biblioref" onclick="">Allendorf et al., 2022</a>; <a href="#/references" role="doc-biblioref" onclick="">Rosenberg &amp; Nordborg, 2002</a>)</span></li>
</ul>
</section>
<section class="slide level2">


<img data-src="class_7_files/figure-revealjs/Ne-1.png" width="960" class="r-stretch"></section>
<section id="references" class="slide level2 smaller scrollable">
<h2>References</h2>


<img src="../../figures/Tor-Vergata_logo_cmyk.png" class="slide-logo r-stretch"><div class="footer footer-default">
<p><a href="https://giulianocolosimo.github.io/biologia_della_conservazione_23_24/">Biologia della conservazione 23/24 - Giuliano Colosimo, PhD</a></p>
</div>
<div id="refs" class="references csl-bib-body hanging-indent" data-line-spacing="2" role="list">
<div id="ref-Allendorf2022" class="csl-entry" role="listitem">
Allendorf, F. W., Funk, W. C., Aitken, S. N., Byrne, M., Luikart, G., &amp; Antunes, A. (2022). <em><span class="nocase">Conservation and the Genomics of Populations</span></em>. Oxford University Press. Retrieved from <a href="https://doi.org/10.1093/oso/9780198856566.001.0001">https://doi.org/10.1093/oso/9780198856566.001.0001</a>
</div>
<div id="ref-Charlesworth2009a" class="csl-entry" role="listitem">
Charlesworth, B. (2009). <span class="nocase">Fundamental concepts in genetics: Effective population size and patterns of molecular evolution and variation</span>. <em>Nature Reviews Genetics</em>, <em>10</em>(3), 195–205. Retrieved from <a href="http://www.nature.com/doifinder/10.1038/nrg2526">http://www.nature.com/doifinder/10.1038/nrg2526</a>
</div>
<div id="ref-Chen2018" class="csl-entry" role="listitem">
Chen, N., Juric, I., Cosgrove, E. J., Bowman, R., Fitzpatrick, J. W., Schoech, S. J., Clark, A. G., et al. (2018). Allele frequency dynamics in a pedigreed natural population. <em>Proceedings of the National Academy of Sciences</em>, <em>116</em>(6), 2158–2164. Proceedings of the National Academy of Sciences. Retrieved from <a href="https://doi.org/10.1073%2Fpnas.1813852116">https://doi.org/10.1073%2Fpnas.1813852116</a>
</div>
<div id="ref-Donald2007" class="csl-entry" role="listitem">
Donald, P. F. (2007). Adult sex ratios in wild bird populations. <em>Ibis</em>, <em>149</em>(4), 671–692. Retrieved from <a href="https://onlinelibrary.wiley.com/doi/abs/10.1111/j.1474-919X.2007.00724.x">https://onlinelibrary.wiley.com/doi/abs/10.1111/j.1474-919X.2007.00724.x</a>
</div>
<div id="ref-Frankham1995a" class="csl-entry" role="listitem">
Frankham, R. (1995). <a href="https://doi.org/10.1017/S0016672308009695"><span class="nocase">Effective population size/adult population size ratios in wildlife: a review.</span></a> <em>Genetical research</em>, <em>89</em>(5-6), 491–503.
</div>
<div id="ref-Frankham1996" class="csl-entry" role="listitem">
Frankham, R. (1996). <a href="https://doi.org/10.1046/j.1523-1739.1996.10061500.x"><span class="nocase">Relationship of genetic variation to population size in wildlife</span></a>. <em>Conserv Biol</em>, <em>10</em>(6), 1500–1508.
</div>
<div id="ref-Frankham2010" class="csl-entry" role="listitem">
Frankham, R., Ballou, J. D., &amp; Briscoe, D. A. (2010). <em>Introduction to conservation genetics</em>. Cambridge University Press. Retrieved from <a href="https://doi.org/10.1017%2Fcbo9780511809002">https://doi.org/10.1017%2Fcbo9780511809002</a>
</div>
<div id="ref-Hedrick2005" class="csl-entry" role="listitem">
Hedrick, P. W. (2005). <em><span class="nocase">Genetics of Populations</span></em> (Third., pp. 1–737). Sudbury, MA: Jones; Bartlett.
</div>
<div id="ref-Reed2003b" class="csl-entry" role="listitem">
Reed, D. H., OGrady, J. J., Ballou, J. D., &amp; Frankham, R. (2003). The frequency and severity of catastrophic die-offs in vertebrates. <em>Animal Conservation</em>, <em>6</em>(2), 109–114. Wiley. Retrieved from <a href="https://doi.org/10.1017%2Fs1367943003003147">https://doi.org/10.1017%2Fs1367943003003147</a>
</div>
<div id="ref-Rosenberg2002" class="csl-entry" role="listitem">
Rosenberg, N. A., &amp; Nordborg, M. (2002). Genealogical trees, coalescent theory and the analysis of genetic polymorphisms. <em>Nature Reviews Genetics</em>, <em>3</em>(5), 380–390. Springer Science; Business Media <span>LLC</span>. Retrieved from <a href="https://doi.org/10.1038%2Fnrg795">https://doi.org/10.1038%2Fnrg795</a>
</div>
<div id="ref-Russo2021" class="csl-entry" role="listitem">
Russo, T., Colosimo, G., Fanelli, C., Franceschini, S., &amp; Parisi, A. (2021). Un viaggio nell’ambiente r. Retrieved from <a href="https://tommaso-russo.github.io/libroR/index.html">https://tommaso-russo.github.io/libroR/index.html</a>
</div>
<div id="ref-Soule1973" class="csl-entry" role="listitem">
Soule, M. (1973). The epistasis cycle: A theory of marginal populations. <em>Annual Review of Ecology and Systematics</em>, <em>4</em>, 165–187. Annual Reviews. Retrieved April 5, 2023, from <a href="http://www.jstor.org/stable/2096809">http://www.jstor.org/stable/2096809</a>
</div>
<div id="ref-Wright1931" class="csl-entry" role="listitem">
Wright, S. (1931). <a href="https://doi.org/10.1016/S0092-8240(05)80011-4"><span class="nocase">Evolution in Mendelian populations</span></a>. <em>Bulletin of Mathematical Biology</em>, <em>52</em>(1-2), 241–295.
</div>
</div>
</section>
    </div>
  </div>

  <script>window.backupDefine = window.define; window.define = undefined;</script>
  <script src="../../site_libs/revealjs/dist/reveal.js"></script>
  <!-- reveal.js plugins -->
  <script src="../../site_libs/revealjs/plugin/quarto-line-highlight/line-highlight.js"></script>
  <script src="../../site_libs/revealjs/plugin/pdf-export/pdfexport.js"></script>
  <script src="../../site_libs/revealjs/plugin/reveal-menu/menu.js"></script>
  <script src="../../site_libs/revealjs/plugin/reveal-menu/quarto-menu.js"></script>
  <script src="../../site_libs/revealjs/plugin/reveal-chalkboard/plugin.js"></script>
  <script src="../../site_libs/revealjs/plugin/multiplex/socket.io.js"></script>
  <script src="../../site_libs/revealjs/plugin/multiplex/multiplex.js"></script>
  <script src="../../site_libs/revealjs/plugin/quarto-support/support.js"></script>
  

  <script src="../../site_libs/revealjs/plugin/notes/notes.js"></script>
  <script src="../../site_libs/revealjs/plugin/search/search.js"></script>
  <script src="../../site_libs/revealjs/plugin/zoom/zoom.js"></script>
  <script src="../../site_libs/revealjs/plugin/math/math.js"></script>
  <script>window.define = window.backupDefine; window.backupDefine = undefined;</script>

  <script>

      // Full list of configuration options available at:
      // https://revealjs.com/config/
      Reveal.initialize({
'controlsAuto': true,
'previewLinksAuto': false,
'smaller': false,
'pdfSeparateFragments': false,
'autoAnimateEasing': "ease",
'autoAnimateDuration': 1,
'autoAnimateUnmatched': true,
'menu': {"side":"left","useTextContentForMissingTitles":true,"markers":false,"loadIcons":false,"custom":[{"title":"Tools","icon":"<i class=\"fas fa-gear\"></i>","content":"<ul class=\"slide-menu-items\">\n<li class=\"slide-tool-item active\" data-item=\"0\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.fullscreen(event)\"><kbd>f</kbd> Fullscreen</a></li>\n<li class=\"slide-tool-item\" data-item=\"1\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.speakerMode(event)\"><kbd>s</kbd> Speaker View</a></li>\n<li class=\"slide-tool-item\" data-item=\"2\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.overview(event)\"><kbd>o</kbd> Slide Overview</a></li>\n<li class=\"slide-tool-item\" data-item=\"3\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.togglePdfExport(event)\"><kbd>e</kbd> PDF Export Mode</a></li>\n<li class=\"slide-tool-item\" data-item=\"4\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.toggleChalkboard(event)\"><kbd>b</kbd> Toggle Chalkboard</a></li>\n<li class=\"slide-tool-item\" data-item=\"5\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.toggleNotesCanvas(event)\"><kbd>c</kbd> Toggle Notes Canvas</a></li>\n<li class=\"slide-tool-item\" data-item=\"6\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.downloadDrawings(event)\"><kbd>d</kbd> Download Drawings</a></li>\n<li class=\"slide-tool-item\" data-item=\"7\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.keyboardHelp(event)\"><kbd>?</kbd> Keyboard Help</a></li>\n</ul>"}],"openButton":true},
'chalkboard': {"buttons":true,"boardmarkerWidth":5},
'multiplex': {"secret":null,"id":"3b6ee20dda0f1e99","url":"https://reveal-multiplex.glitch.me/"},
'smaller': false,
 
        // Display controls in the bottom right corner
        controls: false,

        // Help the user learn the controls by providing hints, for example by
        // bouncing the down arrow when they first encounter a vertical slide
        controlsTutorial: false,

        // Determines where controls appear, "edges" or "bottom-right"
        controlsLayout: 'edges',

        // Visibility rule for backwards navigation arrows; "faded", "hidden"
        // or "visible"
        controlsBackArrows: 'faded',

        // Display a presentation progress bar
        progress: true,

        // Display the page number of the current slide
        slideNumber: 'c/t',

        // 'all', 'print', or 'speaker'
        showSlideNumber: 'all',

        // Add the current slide number to the URL hash so that reloading the
        // page/copying the URL will return you to the same slide
        hash: true,

        // Start with 1 for the hash rather than 0
        hashOneBasedIndex: false,

        // Flags if we should monitor the hash and change slides accordingly
        respondToHashChanges: true,

        // Push each slide change to the browser history
        history: true,

        // Enable keyboard shortcuts for navigation
        keyboard: true,

        // Enable the slide overview mode
        overview: true,

        // Disables the default reveal.js slide layout (scaling and centering)
        // so that you can use custom CSS layout
        disableLayout: false,

        // Vertical centering of slides
        center: false,

        // Enables touch navigation on devices with touch input
        touch: true,

        // Loop the presentation
        loop: false,

        // Change the presentation direction to be RTL
        rtl: false,

        // see https://revealjs.com/vertical-slides/#navigation-mode
        navigationMode: 'linear',

        // Randomizes the order of slides each time the presentation loads
        shuffle: false,

        // Turns fragments on and off globally
        fragments: true,

        // Flags whether to include the current fragment in the URL,
        // so that reloading brings you to the same fragment position
        fragmentInURL: false,

        // Flags if the presentation is running in an embedded mode,
        // i.e. contained within a limited portion of the screen
        embedded: false,

        // Flags if we should show a help overlay when the questionmark
        // key is pressed
        help: true,

        // Flags if it should be possible to pause the presentation (blackout)
        pause: true,

        // Flags if speaker notes should be visible to all viewers
        showNotes: false,

        // Global override for autoplaying embedded media (null/true/false)
        autoPlayMedia: null,

        // Global override for preloading lazy-loaded iframes (null/true/false)
        preloadIframes: null,

        // Number of milliseconds between automatically proceeding to the
        // next slide, disabled when set to 0, this value can be overwritten
        // by using a data-autoslide attribute on your slides
        autoSlide: 0,

        // Stop auto-sliding after user input
        autoSlideStoppable: true,

        // Use this method for navigation when auto-sliding
        autoSlideMethod: null,

        // Specify the average time in seconds that you think you will spend
        // presenting each slide. This is used to show a pacing timer in the
        // speaker view
        defaultTiming: null,

        // Enable slide navigation via mouse wheel
        mouseWheel: false,

        // The display mode that will be used to show slides
        display: 'block',

        // Hide cursor if inactive
        hideInactiveCursor: true,

        // Time before the cursor is hidden (in ms)
        hideCursorTime: 5000,

        // Opens links in an iframe preview overlay
        previewLinks: true,

        // Transition style (none/fade/slide/convex/concave/zoom)
        transition: 'fade',

        // Transition speed (default/fast/slow)
        transitionSpeed: 'default',

        // Transition style for full page slide backgrounds
        // (none/fade/slide/convex/concave/zoom)
        backgroundTransition: 'none',

        // Number of slides away from the current that are visible
        viewDistance: 3,

        // Number of slides away from the current that are visible on mobile
        // devices. It is advisable to set this to a lower number than
        // viewDistance in order to save resources.
        mobileViewDistance: 2,

        // The "normal" size of the presentation, aspect ratio will be preserved
        // when the presentation is scaled to fit different resolutions. Can be
        // specified using percentage units.
        width: 1600,

        height: 900,

        // Factor of the display size that should remain empty around the content
        margin: 0.1,

        math: {
          mathjax: 'https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.0/MathJax.js',
          config: 'TeX-AMS_HTML-full',
          tex2jax: {
            inlineMath: [['\\(','\\)']],
            displayMath: [['\\[','\\]']],
            balanceBraces: true,
            processEscapes: false,
            processRefs: true,
            processEnvironments: true,
            preview: 'TeX',
            skipTags: ['script','noscript','style','textarea','pre','code'],
            ignoreClass: 'tex2jax_ignore',
            processClass: 'tex2jax_process'
          },
        },

        // reveal.js plugins
        plugins: [QuartoLineHighlight, PdfExport, RevealMenu, RevealChalkboard, QuartoSupport,

          RevealMath,
          
          RevealSearch,
          RevealZoom
        ]
      });
    </script>
    
    <script>
      // htmlwidgets need to know to resize themselves when slides are shown/hidden.
      // Fire the "slideenter" event (handled by htmlwidgets.js) when the current
      // slide changes (different for each slide format).
      (function () {
        // dispatch for htmlwidgets
        function fireSlideEnter() {
          const event = window.document.createEvent("Event");
          event.initEvent("slideenter", true, true);
          window.document.dispatchEvent(event);
        }

        function fireSlideChanged(previousSlide, currentSlide) {
          fireSlideEnter();

          // dispatch for shiny
          if (window.jQuery) {
            if (previousSlide) {
              window.jQuery(previousSlide).trigger("hidden");
            }
            if (currentSlide) {
              window.jQuery(currentSlide).trigger("shown");
            }
          }
        }

        // hookup for slidy
        if (window.w3c_slidy) {
          window.w3c_slidy.add_observer(function (slide_num) {
            // slide_num starts at position 1
            fireSlideChanged(null, w3c_slidy.slides[slide_num - 1]);
          });
        }

      })();
    </script>

    <script id="quarto-html-after-body" type="application/javascript">
    window.document.addEventListener("DOMContentLoaded", function (event) {
      const toggleBodyColorMode = (bsSheetEl) => {
        const mode = bsSheetEl.getAttribute("data-mode");
        const bodyEl = window.document.querySelector("body");
        if (mode === "dark") {
          bodyEl.classList.add("quarto-dark");
          bodyEl.classList.remove("quarto-light");
        } else {
          bodyEl.classList.add("quarto-light");
          bodyEl.classList.remove("quarto-dark");
        }
      }
      const toggleBodyColorPrimary = () => {
        const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
        if (bsSheetEl) {
          toggleBodyColorMode(bsSheetEl);
        }
      }
      toggleBodyColorPrimary();  
      const tabsets =  window.document.querySelectorAll(".panel-tabset-tabby")
      tabsets.forEach(function(tabset) {
        const tabby = new Tabby('#' + tabset.id);
      });
      const isCodeAnnotation = (el) => {
        for (const clz of el.classList) {
          if (clz.startsWith('code-annotation-')) {                     
            return true;
          }
        }
        return false;
      }
      const clipboard = new window.ClipboardJS('.code-copy-button', {
        text: function(trigger) {
          const codeEl = trigger.previousElementSibling.cloneNode(true);
          for (const childEl of codeEl.children) {
            if (isCodeAnnotation(childEl)) {
              childEl.remove();
            }
          }
          return codeEl.innerText;
        }
      });
      clipboard.on('success', function(e) {
        // button target
        const button = e.trigger;
        // don't keep focus
        button.blur();
        // flash "checked"
        button.classList.add('code-copy-button-checked');
        var currentTitle = button.getAttribute("title");
        button.setAttribute("title", "Copied!");
        let tooltip;
        if (window.bootstrap) {
          button.setAttribute("data-bs-toggle", "tooltip");
          button.setAttribute("data-bs-placement", "left");
          button.setAttribute("data-bs-title", "Copied!");
          tooltip = new bootstrap.Tooltip(button, 
            { trigger: "manual", 
              customClass: "code-copy-button-tooltip",
              offset: [0, -8]});
          tooltip.show();    
        }
        setTimeout(function() {
          if (tooltip) {
            tooltip.hide();
            button.removeAttribute("data-bs-title");
            button.removeAttribute("data-bs-toggle");
            button.removeAttribute("data-bs-placement");
          }
          button.setAttribute("title", currentTitle);
          button.classList.remove('code-copy-button-checked');
        }, 1000);
        // clear code selection
        e.clearSelection();
      });
      function tippyHover(el, contentFn) {
        const config = {
          allowHTML: true,
          content: contentFn,
          maxWidth: 500,
          delay: 100,
          arrow: false,
          appendTo: function(el) {
              return el.closest('section.slide') || el.parentElement;
          },
          interactive: true,
          interactiveBorder: 10,
          theme: 'light-border',
          placement: 'bottom-start'
        };
          config['offset'] = [0,0];
          config['maxWidth'] = 700;
        window.tippy(el, config); 
      }
      const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
      for (var i=0; i<noterefs.length; i++) {
        const ref = noterefs[i];
        tippyHover(ref, function() {
          // use id or data attribute instead here
          let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
          try { href = new URL(href).hash; } catch {}
          const id = href.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          return note.innerHTML;
        });
      }
      const findCites = (el) => {
        const parentEl = el.parentElement;
        if (parentEl) {
          const cites = parentEl.dataset.cites;
          if (cites) {
            return {
              el,
              cites: cites.split(' ')
            };
          } else {
            return findCites(el.parentElement)
          }
        } else {
          return undefined;
        }
      };
      var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
      for (var i=0; i<bibliorefs.length; i++) {
        const ref = bibliorefs[i];
        const citeInfo = findCites(ref);
        if (citeInfo) {
          tippyHover(citeInfo.el, function() {
            var popup = window.document.createElement('div');
            citeInfo.cites.forEach(function(cite) {
              var citeDiv = window.document.createElement('div');
              citeDiv.classList.add('hanging-indent');
              citeDiv.classList.add('csl-entry');
              var biblioDiv = window.document.getElementById('ref-' + cite);
              if (biblioDiv) {
                citeDiv.innerHTML = biblioDiv.innerHTML;
              }
              popup.appendChild(citeDiv);
            });
            return popup.innerHTML;
          });
        }
      }
    });
    </script>
    

</body></html>