<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.4.553">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="dcterms.date" content="2024-04-11">

<title>Biologia della Conservazione 23/24 - Biologia della Conservazione</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../">
<script src="../../site_libs/quarto-html/quarto.js"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-text-highlighting-styles">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-text-highlighting-styles">
<script src="../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-bootstrap" data-mode="light">
<link href="../../site_libs/bootstrap/bootstrap-dark.min.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-bootstrap" data-mode="dark">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>

  <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>

<meta property="og:title" content="Biologia della Conservazione 23/24 - Biologia della Conservazione">
<meta property="og:description" content="Class 7 - Genetic Drift/Effective Population Size">
<meta property="og:site_name" content="Biologia della Conservazione 23/24">
<meta property="og:locale" content="en_US">
</head>

<body class="nav-sidebar docked nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="../../index.html">
    <span class="navbar-title">Biologia della Conservazione 23/24</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item">
    <a class="nav-link active" href="../../index.html" aria-current="page"> 
<span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../calendar.html"> 
<span class="menu-text">Calendar</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../contact.html"> 
<span class="menu-text">Contact</span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
          <div class="quarto-navbar-tools">
  <a href="" class="quarto-color-scheme-toggle quarto-navigation-tool  px-1" onclick="window.quartoToggleColorScheme(); return false;" title="Toggle dark mode"><i class="bi"></i></a>
  <a href="" class="quarto-reader-toggle quarto-navigation-tool px-1" onclick="window.quartoToggleReader(); return false;" title="Toggle reader mode">
  <div class="quarto-reader-toggle-btn">
  <i class="bi"></i>
  </div>
</a>
</div>
      </div> <!-- /container-fluid -->
    </nav>
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item">Biologia della Conservazione</li></ol></nav>
        <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation docked overflow-auto">
    <div class="pt-lg-2 mt-2 text-center sidebar-header">
      <a href="../../index.html" class="sidebar-logo-link">
      <img src="../.././figures/avatar.jpg" alt="" class="sidebar-logo py-0 d-lg-inline d-none">
      </a>
      <div class="sidebar-tools-main">
    <a href="mailto:giuliano.colosimo@protonmail.com" title="" class="quarto-navigation-tool px-1" aria-label=""><i class="bi bi-envelope"></i></a>
</div>
      </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true">
 <span class="menu-text">Classes</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../.././classes/1/class_1.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Class 1 - Introduction to conservation biology</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../.././classes/2/class_2.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Class 2 - Biodiversity</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../.././classes/3/class_3.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Class 3 - Biodiversity &amp; Extinctions</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../.././classes/4/class_4.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Class 4 - Populations (1)</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../.././classes/5/class_5.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Class 5 - Populations (2) - Intro to Conservation Genetics</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../.././classes/6/class_6.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Class 6 - Conservation Genetics</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../.././classes/7/class_7.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Class 7 - Genetic Drift/Effective Population Size</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../.././classes/8/class_8.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Class 8 - Manuscript discussion-Genetic Drift/Effective Population Size</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../.././classes/9/class_9.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Class 9 - Manuscript discussion-Genetic Drift/Effective Population Size</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../.././classes/10/class_10.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Class 10 - Inbreeding</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../.././classes/11/class_11.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Class 11 - Arguments review for mid term</span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#topics" id="toc-topics" class="nav-link active" data-scroll-target="#topics">Topics</a>
  <ul class="collapse">
  <li><a href="#sampling-error" id="toc-sampling-error" class="nav-link" data-scroll-target="#sampling-error">Sampling error</a></li>
  <li><a href="#testing-hypothesis" id="toc-testing-hypothesis" class="nav-link" data-scroll-target="#testing-hypothesis">Testing hypothesis</a></li>
  <li><a href="#when-not-to-use-the-chi2-test" id="toc-when-not-to-use-the-chi2-test" class="nav-link" data-scroll-target="#when-not-to-use-the-chi2-test">When not to use the <span class="math inline">\(\chi^2\)</span> test</a></li>
  <li><a href="#a-valid-alternative-fishers-exact-test" id="toc-a-valid-alternative-fishers-exact-test" class="nav-link" data-scroll-target="#a-valid-alternative-fishers-exact-test">A valid alternative: Fisher’s exact test</a></li>
  <li><a href="#other-options-for-testing-null-hypothesis" id="toc-other-options-for-testing-null-hypothesis" class="nav-link" data-scroll-target="#other-options-for-testing-null-hypothesis">Other options for testing null hypothesis</a></li>
  <li><a href="#multiple-tests" id="toc-multiple-tests" class="nav-link" data-scroll-target="#multiple-tests">Multiple tests</a></li>
  <li><a href="#what-we-learnt-from-the-hw-model" id="toc-what-we-learnt-from-the-hw-model" class="nav-link" data-scroll-target="#what-we-learnt-from-the-hw-model">What we learnt from the HW model</a></li>
  <li><a href="#genetic-drift" id="toc-genetic-drift" class="nav-link" data-scroll-target="#genetic-drift">Genetic Drift</a></li>
  <li><a href="#genetic-drift-1" id="toc-genetic-drift-1" class="nav-link" data-scroll-target="#genetic-drift-1">Genetic drift</a></li>
  <li><a href="#genetic-drift-2" id="toc-genetic-drift-2" class="nav-link" data-scroll-target="#genetic-drift-2">Genetic Drift</a></li>
  <li><a href="#exp.-distribution-of-allele-frequencies-in-population-of-size-n-2" id="toc-exp.-distribution-of-allele-frequencies-in-population-of-size-n-2" class="nav-link" data-scroll-target="#exp.-distribution-of-allele-frequencies-in-population-of-size-n-2">Exp. distribution of allele frequencies in population of size <span class="math inline">\(n = 2\)</span></a></li>
  <li><a href="#genetic-drift-3" id="toc-genetic-drift-3" class="nav-link" data-scroll-target="#genetic-drift-3">Genetic Drift</a></li>
  <li><a href="#genetic-drift-4" id="toc-genetic-drift-4" class="nav-link" data-scroll-target="#genetic-drift-4">Genetic Drift</a></li>
  <li><a href="#genetic-drift-5" id="toc-genetic-drift-5" class="nav-link" data-scroll-target="#genetic-drift-5">Genetic Drift</a></li>
  <li><a href="#genetic-drift-6" id="toc-genetic-drift-6" class="nav-link" data-scroll-target="#genetic-drift-6">Genetic Drift</a></li>
  <li><a href="#size-matters" id="toc-size-matters" class="nav-link" data-scroll-target="#size-matters">Size matters</a></li>
  <li><a href="#population-size" id="toc-population-size" class="nav-link" data-scroll-target="#population-size">Population size</a></li>
  <li><a href="#population-size-1" id="toc-population-size-1" class="nav-link" data-scroll-target="#population-size-1">Population size</a></li>
  <li><a href="#wright-fisher-population-model" id="toc-wright-fisher-population-model" class="nav-link" data-scroll-target="#wright-fisher-population-model">Wright-Fisher population model</a></li>
  <li><a href="#wright-fisher-population-model-1" id="toc-wright-fisher-population-model-1" class="nav-link" data-scroll-target="#wright-fisher-population-model-1">Wright-Fisher population model</a></li>
  <li><a href="#estimating-n_e" id="toc-estimating-n_e" class="nav-link" data-scroll-target="#estimating-n_e">Estimating <span class="math inline">\(N_e\)</span></a></li>
  <li><a href="#the-coalescent-theory" id="toc-the-coalescent-theory" class="nav-link" data-scroll-target="#the-coalescent-theory">The coalescent theory</a></li>
  </ul></li>
  <li><a href="#references" id="toc-references" class="nav-link" data-scroll-target="#references">References</a></li>
  </ul>
<div class="toc-actions"><ul><li><a href="https://github.com/quarto-dev/quarto-demo/issues/new" class="toc-action"><i class="bi bi-github"></i>Report an issue</a></li></ul></div><div class="quarto-alternate-formats"><h2>Other Formats</h2><ul><li><a href="class_7.html"><i class="bi bi-file-slides"></i>RevealJS</a></li></ul></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Biologia della Conservazione</h1>
<p class="subtitle lead">Class 7 - Genetic Drift/Effective Population Size</p>
</div>



<div class="quarto-title-meta">

    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">April 11, 2024</p>
    </div>
  </div>
  
    
  </div>
  


</header>


<section id="topics" class="level2">
<h2 class="anchored" data-anchor-id="topics">Topics</h2>
<div class="columns">
<div class="column" style="width:50%;">
<ol type="1">
<li><span style="color: green;">Introduction to conservation biology</span></li>
<li><span style="color: green;">Biodiversity</span></li>
<li><span style="color: green;">Biodiversity and extinctions</span></li>
<li><span style="color: green;">Populations</span></li>
<li><span style="color: green;">Basics of conservation genetics</span></li>
<li><span style="color: orange;">Genetic drift</span></li>
<li><span style="color: orange;">Effective population size</span></li>
<li><span style="color: blue;">Inbreeding depression</span></li>
<li><span style="color: blue;">Population structure</span></li>
<li><span style="color: blue;">Hybridization</span></li>
<li><span style="color: blue;">Applying genetic information to conservation</span></li>
</ol>
</div><div class="column" style="width:50%;">
<ol start="12" type="1">
<li><span style="color: blue;">Managing genetic diversity in captive populations</span></li>
<li><span style="color: blue;">Managing genetic diversity in the wild</span></li>
<li><span style="color: blue;">Population viability analysis</span></li>
<li><span style="color: blue;">Invasive species</span></li>
<li><span style="color: blue;">CITES and trade regulations of endangered species</span></li>
<li><span style="color: blue;">Conservation and climate change</span></li>
<li><span style="color: blue;">Future climate impacts</span></li>
<li><span style="color: blue;">Conservation economics</span></li>
<li><span style="color: blue;">Conservation technology</span></li>
</ol>
</div>
</div>
<hr>
<section id="sampling-error" class="level3">
<h3 class="anchored" data-anchor-id="sampling-error">Sampling error</h3>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="./samplingError.png" class="img-fluid quarto-figure quarto-figure-center figure-img"></p>
</figure>
</div>
<hr>
</section>
<section id="testing-hypothesis" class="level3">
<h3 class="anchored" data-anchor-id="testing-hypothesis">Testing hypothesis</h3>
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="class_7_files/figure-revealjs/chiSq2-1.png" class="img-fluid figure-img" width="960"></p>
</figure>
</div>
</div>
</div>
<hr>
</section>
<section id="when-not-to-use-the-chi2-test" class="level3">
<h3 class="anchored" data-anchor-id="when-not-to-use-the-chi2-test">When not to use the <span class="math inline">\(\chi^2\)</span> test</h3>
<ul>
<li>The expected count of individuals is &lt; 5 in any of the possible genotype considered</li>
<li>Very low allele frequencies</li>
<li>The sample size is too small</li>
<li>There are too many alleles at a locus</li>
</ul>
</section>
<section id="a-valid-alternative-fishers-exact-test" class="level3">
<h3 class="anchored" data-anchor-id="a-valid-alternative-fishers-exact-test">A valid alternative: Fisher’s exact test</h3>
<ul>
<li>Based on the calculation of probabilities</li>
<li><span class="math inline">\(P(genotype\ counts|allele \ counts)\)</span> assuming HWE</li>
<li>Let’s work out an example with a small population: 4 individuals <span class="math inline">\(AA\)</span>, 2 individuals <span class="math inline">\(Aa\)</span> and 1 individuals <span class="math inline">\(aa\)</span></li>
<li>What genotype configurations could give us the same allele count?</li>
</ul>
<hr>
<table class="table">
<thead>
<tr class="header">
<th style="text-align: center;"><span class="math inline">\(n_{AA}\)</span></th>
<th style="text-align: center;"><span class="math inline">\(n_{Aa}\)</span></th>
<th style="text-align: center;"><span class="math inline">\(n_{aa}\)</span></th>
<th style="text-align: center;"><span class="math inline">\(n_{A}\)</span></th>
<th style="text-align: center;"><span class="math inline">\(n_{a}\)</span></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: center;">5</td>
<td style="text-align: center;">0</td>
<td style="text-align: center;">2</td>
<td style="text-align: center;">10</td>
<td style="text-align: center;">4</td>
</tr>
<tr class="even">
<td style="text-align: center;">4</td>
<td style="text-align: center;">2</td>
<td style="text-align: center;">1</td>
<td style="text-align: center;">10</td>
<td style="text-align: center;">4</td>
</tr>
<tr class="odd">
<td style="text-align: center;">3</td>
<td style="text-align: center;">4</td>
<td style="text-align: center;">0</td>
<td style="text-align: center;">10</td>
<td style="text-align: center;">4</td>
</tr>
</tbody>
</table>
<p>. . .</p>
<ul>
<li>Can we estimate the theoretical probability of each combination of genotypes?</li>
</ul>
<p>. . .</p>
<p><span class="math display">\[
P(n_{Aa}|n_A, n_a, HWE)=\frac{n!}{n_{AA}!*n_{Aa}!*n_{aa}!}*\frac{2^{n_{Aa}}*n_A!*n_a!}{(2n)!}
\]</span></p>
<p>. . .</p>
<table class="table">
<thead>
<tr class="header">
<th style="text-align: center;"><span class="math inline">\(n_{AA}\)</span></th>
<th style="text-align: center;"><span class="math inline">\(n_{Aa}\)</span></th>
<th style="text-align: center;"><span class="math inline">\(n_{aa}\)</span></th>
<th style="text-align: center;">Theoretical probability</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: center;">5</td>
<td style="text-align: center;">0</td>
<td style="text-align: center;">2</td>
<td style="text-align: center;">0.021</td>
</tr>
<tr class="even">
<td style="text-align: center;">4</td>
<td style="text-align: center;">2</td>
<td style="text-align: center;">1</td>
<td style="text-align: center;">0.419</td>
</tr>
<tr class="odd">
<td style="text-align: center;">3</td>
<td style="text-align: center;">4</td>
<td style="text-align: center;">0</td>
<td style="text-align: center;">0.559</td>
</tr>
</tbody>
</table>
<hr>
</section>
<section id="other-options-for-testing-null-hypothesis" class="level3">
<h3 class="anchored" data-anchor-id="other-options-for-testing-null-hypothesis">Other options for testing null hypothesis</h3>
<ul>
<li>Likelihood, the probability of observing the data given some parameter value under a certain statistical model, particularly useful when there are many alleles at a single locus</li>
<li>Permutation tests, randomization of the dataset by sampling, particularly useful when many genotypes are expected to have very low frequency</li>
</ul>
</section>
<section id="multiple-tests" class="level3">
<h3 class="anchored" data-anchor-id="multiple-tests">Multiple tests</h3>
<ul>
<li><p>If we examine 10 loci in 10 population samples, 100 tests of HW proportions will be performed. If all of these loci are in HW proportions (<em>i.e.</em>, our null hypothesis is true at all loci in all populations), we expect to find five significant tests if we use the 5% significance level <span class="citation" data-cites="Allendorf2022">(<a href="#ref-Allendorf2022" role="doc-biblioref">Allendorf et al., 2022, pp. 102–103</a>)</span>.</p></li>
<li><p>The Bonferroni correction allows to adjust the significance level (say 5%) by dividing it by the number of tests performed. In the previous example <span class="math inline">\(0.05/100 = 0.0005\)</span>. The critical <span class="math inline">\(\chi^2\)</span> value for <span class="math inline">\(P = 0.0005\)</span> with one degree of freedom is 12.1.</p></li>
</ul>
<hr>
</section>
<section id="what-we-learnt-from-the-hw-model" class="level3">
<h3 class="anchored" data-anchor-id="what-we-learnt-from-the-hw-model">What we learnt from the HW model</h3>
<ul>
<li>Genetic change will not occur in populations if all the assumptions of the HW equilibrium are met (and we already know that the HW assumptions are almost never met in nature)</li>
<li>What is the effect of allele and genotype frequencies when the assumptions are not met?</li>
<li>We will start looking at two interrelated factors: genetic drift and population size</li>
</ul>
</section>
<section id="genetic-drift" class="level3">
<h3 class="anchored" data-anchor-id="genetic-drift">Genetic Drift</h3>
<ul>
<li>Genetic drift is random change in allele frequencies from generation to generation because of sampling error. That is, the finite number of genes transmitted to progeny will be an imperfect sample of the allele frequencies in the previous generation <span class="citation" data-cites="Allendorf2022">(<a href="#ref-Allendorf2022" role="doc-biblioref">Allendorf et al., 2022, p. 114</a>)</span></li>
<li>Genetic drift is the primary cause bringing about allele frequency changes throughout the genome over time <span class="citation" data-cites="Chen2018 Allendorf2022">(<a href="#ref-Allendorf2022" role="doc-biblioref">Allendorf et al., 2022</a>; <a href="#ref-Chen2018" role="doc-biblioref">Chen et al., 2018</a>)</span></li>
</ul>
<hr>
</section>
<section id="genetic-drift-1" class="level3">
<h3 class="anchored" data-anchor-id="genetic-drift-1">Genetic drift</h3>
<ul>
<li>The best way to understand how strong the change effect can be in a population is to model the sampling process in the absence of selection, mutation and migration.</li>
<li>We start sampling alleles to generate an offspring from a selfing diploid individual with genotype <span class="math inline">\(A_1A_2\)</span></li>
<li>Our gamete pool is <span class="math inline">\(A_1\)</span> with frequency <span class="math inline">\(p = \frac{1}{2}=0.5\)</span> and <span class="math inline">\(A_2\)</span> with frequency <span class="math inline">\(q=1-p=0.5\)</span></li>
<li>This example is very similar to the simple fair-coin toss! Imagine the coin has <span class="math inline">\(A_1\)</span> on one side and <span class="math inline">\(A_2\)</span> on the other side. To obtain a diploid individual we need to toss the coin twice. Every outcome combination has probability <span class="math inline">\(1/4\)</span></li>
<li>If only one offspring (<span class="math inline">\(N = 1\)</span>) is produced there is 50% chance that one or the other allele would be lost</li>
</ul>
<hr>
</section>
<section id="genetic-drift-2" class="level3">
<h3 class="anchored" data-anchor-id="genetic-drift-2">Genetic Drift</h3>
<ul>
<li>With many offspring (<span class="math inline">\(n &gt; 1\)</span>) there are many more coin tosses and the probability that they all are <span class="math inline">\(A_1\)</span> or <span class="math inline">\(A_2\)</span> now becomes <span class="math inline">\((\frac{1}{2})^{2n}\)</span>, and it will be less likely to loose one of the two alleles</li>
</ul>
</section>
<section id="exp.-distribution-of-allele-frequencies-in-population-of-size-n-2" class="level3">
<h3 class="anchored" data-anchor-id="exp.-distribution-of-allele-frequencies-in-population-of-size-n-2">Exp. distribution of allele frequencies in population of size <span class="math inline">\(n = 2\)</span></h3>
<div class="columns">
<div class="column" style="width:50%;">
<ul>
<li>Population with <span class="math inline">\(p = 0.6\)</span> and <span class="math inline">\(q =0.4\)</span></li>
<li>We sample <span class="math inline">\(2\)</span> individuals (<em>i.e.</em>, <span class="math inline">\(2n = 4\)</span>gametes)</li>
<li>The expected distribution of alleles is the binomial expansion <span class="math inline">\((p+q)^4\)</span></li>
<li>There are 5 possible outcomes</li>
</ul>
</div><div class="column" style="width:50%;">
<table class="table">
<thead>
<tr class="header">
<th>Outcome</th>
<th>Expected frequency</th>
<th>Probability</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><span class="math inline">\(4 A_1,0A_2\)</span></td>
<td><span class="math inline">\(p^4\)</span></td>
<td><span class="math inline">\(0.6^4=0.129\)</span></td>
</tr>
<tr class="even">
<td><span class="math inline">\(3 A_1,1A_2\)</span></td>
<td><span class="math inline">\(4p^3q\)</span></td>
<td><span class="math inline">\(4*0.6^3*0.4=0.345\)</span></td>
</tr>
<tr class="odd">
<td><span class="math inline">\(2 A_1,2A_2\)</span></td>
<td><span class="math inline">\(6p^2q^2\)</span></td>
<td><span class="math inline">\(6*0.6^2*0.4^2=0.345\)</span></td>
</tr>
<tr class="even">
<td><span class="math inline">\(1 A_1,3A_2\)</span></td>
<td><span class="math inline">\(4pq^3\)</span></td>
<td><span class="math inline">\(4*0.6*0.4^3=0.153\)</span></td>
</tr>
<tr class="odd">
<td><span class="math inline">\(0 A_1,4A_2\)</span></td>
<td><span class="math inline">\(q^4\)</span></td>
<td><span class="math inline">\(0.4^4=0.025\)</span></td>
</tr>
</tbody>
</table>
</div>
</div>
<hr>
<ul>
<li>The probability that a population has <span class="math inline">\(k\)</span> copies of allele <span class="math inline">\(A_1\)</span> and <span class="math inline">\(2n-k\)</span> copies of <span class="math inline">\(A_2\)</span> allele is <span class="math inline">\(\binom{2n}{k}p^kq^{2n-k}\)</span> where <span class="math inline">\(\binom{2n}{k}\)</span> is the binomial function <span class="math inline">\(\frac{2n!}{k!(2n-k)!}\)</span></li>
</ul>
<p>. . .</p>
<div class="cell">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="class_7_files/figure-revealjs/expDist-1.png" class="img-fluid figure-img" width="960"></p>
<figcaption>Expected distribution of allele frequency from 100 replicates of sampling 10, 100 and 1000 individuals from a population with p = 0.6</figcaption>
</figure>
</div>
</div>
</div>
<hr>
<ul>
<li>The variance around the mean allele frequency is the binomial sampling variance: <span class="math inline">\(\sigma^2_p =\frac{p_0q_0}{2n}\)</span></li>
<li>Variances are higher in smaller than larger populations</li>
<li>In small population the process of sampling happens every single generation, and its effect on allele frequency is cumulative:</li>
<li>Genetic drift will eventually cause all except one allele to go to fixation.</li>
<li>The probability that a gamete does not contain allele <span class="math inline">\(A_1\)</span> (with frequency <span class="math inline">\(p\)</span> in the parental population) is <span class="math inline">\(1-p\)</span></li>
</ul>
<p><span class="math display">\[
P(fixation \ A_2)=(1-p)^{2n}
\]</span></p>
<hr>
<div class="callout callout-style-default callout-important callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Problem
</div>
</div>
<div class="callout-body-container callout-body">
<p>Diploid population with <span class="math inline">\(n = 4\)</span> with allele <span class="math inline">\(A_1\)</span> with a frequency <span class="math inline">\(p = 0.25\)</span> and allele <span class="math inline">\(A_2\)</span> with a frequency <span class="math inline">\(q = 0.75\)</span>. What is the probability of losing allele <span class="math inline">\(A_1\)</span> in one generation of sampling to recreate a population of the same size?</p>
</div>
</div>
<p>. . .</p>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Solution
</div>
</div>
<div class="callout-body-container callout-body">
<p><span class="math inline">\((1-0.25)^8=\)</span> 0.1001129</p>
</div>
</div>
<hr>
</section>
<section id="genetic-drift-3" class="level3">
<h3 class="anchored" data-anchor-id="genetic-drift-3">Genetic Drift</h3>
<div class="cell">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="class_7_files/figure-revealjs/drift1-1.png" class="img-fluid figure-img" width="960"></p>
<figcaption>Allele frequency distribution at t = 0, with N = 100 <span class="citation" data-cites="Russo2021">(<a href="#ref-Russo2021" role="doc-biblioref">Russo, Colosimo, Fanelli, Franceschini, &amp; Parisi, 2021</a>)</span>.</figcaption>
</figure>
</div>
</div>
</div>
<hr>
</section>
<section id="genetic-drift-4" class="level3">
<h3 class="anchored" data-anchor-id="genetic-drift-4">Genetic Drift</h3>
<div class="cell">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="class_7_files/figure-revealjs/drift2-1.png" class="img-fluid figure-img" width="960"></p>
<figcaption>Allele frequency distribution at t = 1, with N = 100 <span class="citation" data-cites="Russo2021">(<a href="#ref-Russo2021" role="doc-biblioref">Russo et al., 2021</a>)</span>.</figcaption>
</figure>
</div>
</div>
</div>
<hr>
</section>
<section id="genetic-drift-5" class="level3">
<h3 class="anchored" data-anchor-id="genetic-drift-5">Genetic Drift</h3>
<div class="cell">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="class_7_files/figure-revealjs/drift3-1.png" class="img-fluid figure-img" width="960"></p>
<figcaption>Allele frequencies variation after 200 generations while maintaining N = 100 <span class="citation" data-cites="Russo2021">(<a href="#ref-Russo2021" role="doc-biblioref">Russo et al., 2021</a>)</span>.</figcaption>
</figure>
</div>
</div>
</div>
<hr>
</section>
<section id="genetic-drift-6" class="level3">
<h3 class="anchored" data-anchor-id="genetic-drift-6">Genetic Drift</h3>
<section id="simulating-the-average-time-of-allele-fixation-in-our-example" class="level4">
<h4 class="anchored" data-anchor-id="simulating-the-average-time-of-allele-fixation-in-our-example">Simulating the average time of allele fixation in our example</h4>
<div class="cell" data-layout-align="center" data-hash="class_7_cache/revealjs/drift4_04abd7597ea54141f3ba03bafcf83714">
<details class="code-fold">
<summary>Show the code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co"># simulare il tempo medio di fissazione degli alleli</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="co">#pop.ls &lt;- list(geneA = c(rep("A",50), rep("a",50)), </span></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="co">#               geneB = c(rep("B",50), rep("b",50)),</span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="co">#               geneC = c(rep("C",50), rep("c",50)))</span></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="co"># pop.ls  =  lista contenente i geni della popolazione</span></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="co"># num.sim = numero di volte la simulazione deve girare</span></span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a>sim.gen2fix <span class="ot">&lt;-</span> <span class="cf">function</span> (pop.ls, num.sims) {</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a>  <span class="co"># numero totale di loci per ogni gene   </span></span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a>  n <span class="ot">&lt;-</span> <span class="fu">length</span>(pop.ls[[<span class="dv">1</span>]])</span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a>  <span class="co"># numero di geni differenti nella popolazione</span></span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a>  n.genes <span class="ot">&lt;-</span> <span class="fu">length</span>(pop.ls)</span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a>  <span class="co"># lista vuota per i risultati della simulazione</span></span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a>  generations <span class="ot">&lt;-</span> <span class="fu">vector</span>(<span class="at">mode =</span> <span class="st">"list"</span>, <span class="at">length =</span> <span class="fu">length</span>(pop.ls))</span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (h <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>n.genes){</span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a>    generations[[h]] <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="dv">0</span>, num.sims)</span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">names</span>(generations) <span class="ot">&lt;-</span> <span class="fu">names</span>(pop.ls)</span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb1-22"><a href="#cb1-22" aria-hidden="true" tabindex="-1"></a>  <span class="co"># ciclo di simulazione </span></span>
<span id="cb1-23"><a href="#cb1-23" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>num.sims) {</span>
<span id="cb1-24"><a href="#cb1-24" aria-hidden="true" tabindex="-1"></a>    temp.pop.ls <span class="ot">&lt;-</span> pop.ls</span>
<span id="cb1-25"><a href="#cb1-25" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> (y <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>n.genes) {</span>
<span id="cb1-26"><a href="#cb1-26" aria-hidden="true" tabindex="-1"></a>      <span class="cf">while</span> (<span class="fu">max</span>(<span class="fu">table</span>(temp.pop.ls[[y]])) <span class="sc">&lt;</span> n) {</span>
<span id="cb1-27"><a href="#cb1-27" aria-hidden="true" tabindex="-1"></a>        temp.pop.ls[[y]] <span class="ot">&lt;-</span> <span class="fu">sample</span>(temp.pop.ls[[y]],</span>
<span id="cb1-28"><a href="#cb1-28" aria-hidden="true" tabindex="-1"></a>                                   <span class="dv">100</span>, <span class="at">replace =</span> T)</span>
<span id="cb1-29"><a href="#cb1-29" aria-hidden="true" tabindex="-1"></a>        generations[[y]][i] <span class="ot">&lt;-</span> generations[[y]][i] <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb1-30"><a href="#cb1-30" aria-hidden="true" tabindex="-1"></a>      }</span>
<span id="cb1-31"><a href="#cb1-31" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb1-32"><a href="#cb1-32" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb1-33"><a href="#cb1-33" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb1-34"><a href="#cb1-34" aria-hidden="true" tabindex="-1"></a>    <span class="co"># statistiche descrittive dei risultati</span></span>
<span id="cb1-35"><a href="#cb1-35" aria-hidden="true" tabindex="-1"></a>  df <span class="ot">&lt;-</span> <span class="fu">rbind</span>(<span class="fu">data.frame</span>(<span class="fu">lapply</span>(generations, mean)),</span>
<span id="cb1-36"><a href="#cb1-36" aria-hidden="true" tabindex="-1"></a>        <span class="fu">data.frame</span>(<span class="fu">lapply</span>(generations, sd))) </span>
<span id="cb1-37"><a href="#cb1-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames</span>(df) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"mean"</span>, <span class="st">"sd"</span>)</span>
<span id="cb1-38"><a href="#cb1-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-39"><a href="#cb1-39" aria-hidden="true" tabindex="-1"></a>  <span class="co"># ciclo per plottare i risultati</span></span>
<span id="cb1-40"><a href="#cb1-40" aria-hidden="true" tabindex="-1"></a>  <span class="co"># modificare se i geni da plottare sono &gt; 4</span></span>
<span id="cb1-41"><a href="#cb1-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">par</span>(<span class="at">mfrow =</span> <span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">2</span>))</span>
<span id="cb1-42"><a href="#cb1-42" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (z <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>n.genes) {</span>
<span id="cb1-43"><a href="#cb1-43" aria-hidden="true" tabindex="-1"></a>    graph <span class="ot">&lt;-</span> generations[[z]]</span>
<span id="cb1-44"><a href="#cb1-44" aria-hidden="true" tabindex="-1"></a>    <span class="fu">hist</span>(graph, </span>
<span id="cb1-45"><a href="#cb1-45" aria-hidden="true" tabindex="-1"></a>         <span class="at">main =</span> <span class="fu">paste</span>(<span class="st">"Num. of generations to fix "</span>,</span>
<span id="cb1-46"><a href="#cb1-46" aria-hidden="true" tabindex="-1"></a>                             <span class="fu">names</span>(generations[z])),</span>
<span id="cb1-47"><a href="#cb1-47" aria-hidden="true" tabindex="-1"></a>         <span class="at">xlab =</span> <span class="st">"Generations"</span>)</span>
<span id="cb1-48"><a href="#cb1-48" aria-hidden="true" tabindex="-1"></a>    <span class="fu">abline</span>(<span class="at">v=</span><span class="fu">mean</span>(graph), <span class="at">lwd=</span><span class="dv">2</span>, <span class="at">lty=</span><span class="dv">2</span>, <span class="at">col=</span><span class="st">"red"</span>)</span>
<span id="cb1-49"><a href="#cb1-49" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb1-50"><a href="#cb1-50" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb1-51"><a href="#cb1-51" aria-hidden="true" tabindex="-1"></a>  <span class="co">#return(df)</span></span>
<span id="cb1-52"><a href="#cb1-52" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb1-53"><a href="#cb1-53" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-54"><a href="#cb1-54" aria-hidden="true" tabindex="-1"></a><span class="fu">sim.gen2fix</span>(pop.ls, <span class="dv">5000</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="class_7_files/figure-revealjs/drift4-1.png" class="img-fluid quarto-figure quarto-figure-center figure-img" width="960"></p>
</figure>
</div>
</div>
</div>
<hr>
</section>
</section>
<section id="size-matters" class="level3">
<h3 class="anchored" data-anchor-id="size-matters">Size matters</h3>
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="class_7_files/figure-revealjs/drift5-1.png" class="img-fluid figure-img" width="960"></p>
</figure>
</div>
</div>
</div>
<hr>
</section>
<section id="population-size" class="level3">
<h3 class="anchored" data-anchor-id="population-size">Population size</h3>
<blockquote class="blockquote">
<p>There can be no doubt that genetic variation is related to population size, as <span class="citation" data-cites="Soule1973">Soule (<a href="#ref-Soule1973" role="doc-biblioref">1973</a>)</span> proposed. Small population size reduces the evolutionary potential of wildlife species <span class="citation" data-cites="Frankham1996">(<a href="#ref-Frankham1996" role="doc-biblioref">Frankham, 1996</a>)</span>.</p>
</blockquote>
<ul>
<li>The importance of maintaining a large population size is pretty intuitive, but may not be enough!</li>
<li>Many populations of insects and small mammals naturally fluctuate in their “census” population size, but it is when they are at their minimum that the genetic processes affect them the most</li>
<li>Often the census population counts juveniles and senescent individuals that do not really contribute to the reproductive output of next generations</li>
<li>Within adult and reproductive individuals there may be significant variance in the number of offspring sired by males and females</li>
</ul>
<hr>
</section>
<section id="population-size-1" class="level3">
<h3 class="anchored" data-anchor-id="population-size-1">Population size</h3>
<ul>
<li>Different reproductive strategies (monogamy VS. polygamy) may as well influence the genetic output generation after generation</li>
<li>Therefore it is necessary to distinguish between census population and an hypothetical idealized population</li>
<li>We can standardize this concept of idealized population by describing it in term of <strong>effective population size</strong> (<span class="math inline">\(N_e\)</span>), and it can be defined as the actual number of breeding individuals needed in order to see the same variation of a critical parameter (such as genetic diversity) measured in the real population</li>
<li>The effective size of a population determines the rate of change in composition of a population caused by genetic drift (the random sampling of genetic variants) <span class="citation" data-cites="Charlesworth2009a">(<a href="#ref-Charlesworth2009a" role="doc-biblioref">Charlesworth, 2009, p. 195</a>)</span></li>
<li>To better understand the origin of this concept of the idealized population see <span class="citation" data-cites="Wright1931">Wright (<a href="#ref-Wright1931" role="doc-biblioref">1931</a>)</span></li>
</ul>
<hr>
</section>
<section id="wright-fisher-population-model" class="level3">
<h3 class="anchored" data-anchor-id="wright-fisher-population-model">Wright-Fisher population model</h3>
<p>The ideal population:</p>
<ul>
<li>All individual in the population are considered adults and capable of breeding</li>
<li>Individuals are hermafroditic (self fertilization is possible)</li>
<li>The number of breeding individual is constant in all generations and they breed randomly</li>
<li>New individuals are formed each generation by random sampling, with replacement, of the gametes from the parents</li>
<li>The parents die after mating so each generation is only composed by new individuals (discrete generations)</li>
<li>There is no selection at any life stage and mutation is ignored</li>
<li>When <span class="math inline">\(N\)</span> is sufficiently large this means that each individual contributes to the next generation a number of offspring that is approximated by a Poisson distribution</li>
</ul>
<hr>
</section>
<section id="wright-fisher-population-model-1" class="level3">
<h3 class="anchored" data-anchor-id="wright-fisher-population-model-1">Wright-Fisher population model</h3>
<ul>
<li>Any characteristic of a real population that deviates from the characteristic of the ideal population will make the census size <span class="math inline">\(N\)</span> to be different than the effective population size <span class="math inline">\(N_e\)</span></li>
<li>An important measure in conservation is the ratio <span class="math inline">\(\frac{N_e}{N}\)</span></li>
<li>There are multiple evidence that in most natural population this ratio is usually much less than 1.</li>
<li>The primary factors influencing this ratio are:
<ul>
<li>fluctuating population size over multiple generations</li>
<li>high variance in family size</li>
<li>unequal ratios between sexes</li>
</ul></li>
</ul>
<hr>
</section>
<section id="estimating-n_e" class="level3">
<h3 class="anchored" data-anchor-id="estimating-n_e">Estimating <span class="math inline">\(N_e\)</span></h3>
<ul>
<li>To estimate <span class="math inline">\(N_e\)</span> it is possible to use demographic approaches and genetic approaches</li>
</ul>
<section id="demographic-approaches" class="level4">
<h4 class="anchored" data-anchor-id="demographic-approaches">Demographic approaches</h4>
<ul>
<li>Demographic methodologies require extensive data and sometimes are unattainable in wild populations</li>
</ul>
<div class="tabset-margin-container"></div><div class="panel-tabset">
<ul class="nav nav-tabs" role="tablist"><li class="nav-item" role="presentation"><a class="nav-link active" id="tabset-1-1-tab" data-bs-toggle="tab" data-bs-target="#tabset-1-1" role="tab" aria-controls="tabset-1-1" aria-selected="true">Fluctuations in population size</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-1-2-tab" data-bs-toggle="tab" data-bs-target="#tabset-1-2" role="tab" aria-controls="tabset-1-2" aria-selected="false">Variation in family size</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-1-3-tab" data-bs-toggle="tab" data-bs-target="#tabset-1-3" role="tab" aria-controls="tabset-1-3" aria-selected="false">Unequal sex-ratio</a></li></ul>
<div class="tab-content">
<div id="tabset-1-1" class="tab-pane active" role="tabpanel" aria-labelledby="tabset-1-1-tab">
<div class="columns">
<div class="column" style="width:50%;">
<ul>
<li>Natural populations continuously undergo fluctuations in size over multiple generations <span class="citation" data-cites="Reed2003b">(<a href="#ref-Reed2003b" role="doc-biblioref">Reed, OGrady, Ballou, &amp; Frankham, 2003</a>)</span></li>
<li>We can use the harmonic mean to estimate <span class="math inline">\(N_e\)</span> over <span class="math inline">\(t\)</span> generations</li>
</ul>
</div><div class="column" style="width:50%;">
<p><span class="math display">\[
N_e \sim \frac{t}{\sum(\frac{1}{N_{ei}})} \\ \sim \frac{3}{(1/175000)+(1/20)+(1/175000)} \sim 60
\]</span></p>
</div>
</div>
</div>
<div id="tabset-1-2" class="tab-pane" role="tabpanel" aria-labelledby="tabset-1-2-tab">
<div class="columns">
<div class="column" style="width:50%;">
<ul>
<li>The effect on <span class="math inline">\(N_e\)</span> of variation in family sizes in an otherwise ideal population can be estimated using:</li>
</ul>
<p><span class="math display">\[
N_e = \frac{4N-2}{V_k +2}
\]</span></p>
</div><div class="column" style="width:50%;">
<ul>
<li><span class="math inline">\(k\)</span> is the mean family size</li>
<li>In Wright-Fisher population <span class="math inline">\(V_k = 2\)</span> so that <span class="math inline">\(N_e \sim N\)</span></li>
<li>In Darwin’s finches the value of <span class="math inline">\(V_k\)</span> has been estimated at 6.74, which reduces dramatically <span class="math inline">\(N_e\)</span> <span class="citation" data-cites="Frankham1995a">(<a href="#ref-Frankham1995a" role="doc-biblioref">Frankham, 1995</a>)</span></li>
</ul>
</div>
</div>
</div>
<div id="tabset-1-3" class="tab-pane" role="tabpanel" aria-labelledby="tabset-1-3-tab">
<div class="columns">
<div class="column" style="width:50%;">
<ul>
<li>According to <span class="citation" data-cites="Donald2007">Donald (<a href="#ref-Donald2007" role="doc-biblioref">2007</a>)</span> there usually is an excess of females in mammals and a deficiency in males <span class="math display">\[
N_e=\frac{4N_{ef}N_{em}}{N_{ef}+N_{em}}
\]</span></li>
</ul>
</div><div class="column" style="width:50%;">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="./unequalsexratio.jpg" class="img-fluid figure-img" width="250"></p>
<figcaption>Effects of unequal sex ratio on <span class="math inline">\(N_e/N\)</span>. After <span class="citation" data-cites="Frankham2010">Frankham, Ballou, &amp; Briscoe (<a href="#ref-Frankham2010" role="doc-biblioref">2010</a>)</span>, p.249</figcaption>
</figure>
</div>
</div>
</div>
</div>
</div>
</div>
<hr>
</section>
<section id="genetic-approaches" class="level4">
<h4 class="anchored" data-anchor-id="genetic-approaches">Genetic approaches</h4>
<ul>
<li>DNA retains information on the evolutionary history of individuals</li>
<li>We can use basic probability theory to extrapolate information, including information on <span class="math inline">\(N_e\)</span> from DNA markers</li>
<li>The coalescent approach uses the sampling theory for neutral alleles and allows us to trace back in time this information <span class="citation" data-cites="Rosenberg2002">(<a href="#ref-Rosenberg2002" role="doc-biblioref">Rosenberg &amp; Nordborg, 2002</a>)</span></li>
<li>Under the neutral theory of evolution 2 alleles may descend from the same ancestral allele in the previous generation with probability <span class="math inline">\(1/N_{ef}\)</span> for mtDNA, or <span class="math inline">\(1/2N_e\)</span> for a general diploid locus</li>
<li>Alternatively 2 alleles may derive from 2 different alleles in the previous generation with probability <span class="math inline">\(1-\frac{1}{N_{ef}}\)</span></li>
</ul>
<hr>
<ul>
<li>Under this neutral model of genetic drift the coalescence process takes a characteristic time:</li>
</ul>
<p><span class="math display">\[
T_k = \frac{4N_e}{k(k-1)}generations
\]</span> where <span class="math inline">\(T_k\)</span> is the average time in generations back to the previous coalescent event when there were <span class="math inline">\(k-1\)</span> alleles <span class="citation" data-cites="Hedrick2005">(<a href="#ref-Hedrick2005" role="doc-biblioref">Hedrick, 2005</a>)</span></p>
<ul>
<li>The time for all <span class="math inline">\(k\)</span> alleles in the population to coalesce back to one allele is: <span class="math inline">\(4N_e[1-(\frac{1}{k})]\)</span></li>
</ul>
<hr>
</section>
</section>
<section id="the-coalescent-theory" class="level3">
<h3 class="anchored" data-anchor-id="the-coalescent-theory">The coalescent theory</h3>
<ul>
<li>The coalescent allows to study the loss of genetic diversity and to use this information to infer effective population size</li>
<li>To coalesce means “unite” or “fuse”</li>
<li>To better understand this approach we can look at the example of mitochondrial DNA (mtDNA)</li>
<li>mtDNA is haploid, maternally inherited and it does not undergo recombination</li>
<li>Coalescent theory provides a powerful framework to study the effects of genetic drift, natural selection, mutation, and gene flow in natural populations <span class="citation" data-cites="Allendorf2022 Rosenberg2002">(<a href="#ref-Allendorf2022" role="doc-biblioref">Allendorf et al., 2022</a>; <a href="#ref-Rosenberg2002" role="doc-biblioref">Rosenberg &amp; Nordborg, 2002</a>)</span></li>
</ul>
<hr>
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="class_7_files/figure-revealjs/Ne-1.png" class="img-fluid figure-img" width="960"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="references" class="level2">




</section>

<div id="quarto-appendix" class="default"><section class="quarto-appendix-contents" role="doc-bibliography" id="quarto-bibliography"><h2 class="anchored quarto-appendix-heading">References</h2><div id="refs" class="references csl-bib-body hanging-indent" data-entry-spacing="0" data-line-spacing="2" role="list">
<div id="ref-Allendorf2022" class="csl-entry" role="listitem">
Allendorf, F. W., Funk, W. C., Aitken, S. N., Byrne, M., Luikart, G., &amp; Antunes, A. (2022). <em><span class="nocase">Conservation and the Genomics of Populations</span></em>. Oxford University Press. Retrieved from <a href="https://doi.org/10.1093/oso/9780198856566.001.0001">https://doi.org/10.1093/oso/9780198856566.001.0001</a>
</div>
<div id="ref-Charlesworth2009a" class="csl-entry" role="listitem">
Charlesworth, B. (2009). <span class="nocase">Fundamental concepts in genetics: Effective population size and patterns of molecular evolution and variation</span>. <em>Nature Reviews Genetics</em>, <em>10</em>(3), 195–205. Retrieved from <a href="http://www.nature.com/doifinder/10.1038/nrg2526">http://www.nature.com/doifinder/10.1038/nrg2526</a>
</div>
<div id="ref-Chen2018" class="csl-entry" role="listitem">
Chen, N., Juric, I., Cosgrove, E. J., Bowman, R., Fitzpatrick, J. W., Schoech, S. J., Clark, A. G., et al. (2018). Allele frequency dynamics in a pedigreed natural population. <em>Proceedings of the National Academy of Sciences</em>, <em>116</em>(6), 2158–2164. Proceedings of the National Academy of Sciences. Retrieved from <a href="https://doi.org/10.1073%2Fpnas.1813852116">https://doi.org/10.1073%2Fpnas.1813852116</a>
</div>
<div id="ref-Donald2007" class="csl-entry" role="listitem">
Donald, P. F. (2007). Adult sex ratios in wild bird populations. <em>Ibis</em>, <em>149</em>(4), 671–692. Retrieved from <a href="https://onlinelibrary.wiley.com/doi/abs/10.1111/j.1474-919X.2007.00724.x">https://onlinelibrary.wiley.com/doi/abs/10.1111/j.1474-919X.2007.00724.x</a>
</div>
<div id="ref-Frankham1995a" class="csl-entry" role="listitem">
Frankham, R. (1995). <a href="https://doi.org/10.1017/S0016672308009695"><span class="nocase">Effective population size/adult population size ratios in wildlife: a review.</span></a> <em>Genetical research</em>, <em>89</em>(5-6), 491–503.
</div>
<div id="ref-Frankham1996" class="csl-entry" role="listitem">
Frankham, R. (1996). <a href="https://doi.org/10.1046/j.1523-1739.1996.10061500.x"><span class="nocase">Relationship of genetic variation to population size in wildlife</span></a>. <em>Conserv Biol</em>, <em>10</em>(6), 1500–1508.
</div>
<div id="ref-Frankham2010" class="csl-entry" role="listitem">
Frankham, R., Ballou, J. D., &amp; Briscoe, D. A. (2010). <em>Introduction to conservation genetics</em>. Cambridge University Press. Retrieved from <a href="https://doi.org/10.1017%2Fcbo9780511809002">https://doi.org/10.1017%2Fcbo9780511809002</a>
</div>
<div id="ref-Hedrick2005" class="csl-entry" role="listitem">
Hedrick, P. W. (2005). <em><span class="nocase">Genetics of Populations</span></em> (Third., pp. 1–737). Sudbury, MA: Jones; Bartlett.
</div>
<div id="ref-Reed2003b" class="csl-entry" role="listitem">
Reed, D. H., OGrady, J. J., Ballou, J. D., &amp; Frankham, R. (2003). The frequency and severity of catastrophic die-offs in vertebrates. <em>Animal Conservation</em>, <em>6</em>(2), 109–114. Wiley. Retrieved from <a href="https://doi.org/10.1017%2Fs1367943003003147">https://doi.org/10.1017%2Fs1367943003003147</a>
</div>
<div id="ref-Rosenberg2002" class="csl-entry" role="listitem">
Rosenberg, N. A., &amp; Nordborg, M. (2002). Genealogical trees, coalescent theory and the analysis of genetic polymorphisms. <em>Nature Reviews Genetics</em>, <em>3</em>(5), 380–390. Springer Science; Business Media <span>LLC</span>. Retrieved from <a href="https://doi.org/10.1038%2Fnrg795">https://doi.org/10.1038%2Fnrg795</a>
</div>
<div id="ref-Russo2021" class="csl-entry" role="listitem">
Russo, T., Colosimo, G., Fanelli, C., Franceschini, S., &amp; Parisi, A. (2021). Un viaggio nell’ambiente r. Retrieved from <a href="https://tommaso-russo.github.io/libroR/index.html">https://tommaso-russo.github.io/libroR/index.html</a>
</div>
<div id="ref-Soule1973" class="csl-entry" role="listitem">
Soule, M. (1973). The epistasis cycle: A theory of marginal populations. <em>Annual Review of Ecology and Systematics</em>, <em>4</em>, 165–187. Annual Reviews. Retrieved April 5, 2023, from <a href="http://www.jstor.org/stable/2096809">http://www.jstor.org/stable/2096809</a>
</div>
<div id="ref-Wright1931" class="csl-entry" role="listitem">
Wright, S. (1931). <a href="https://doi.org/10.1016/S0092-8240(05)80011-4"><span class="nocase">Evolution in Mendelian populations</span></a>. <em>Bulletin of Mathematical Biology</em>, <em>52</em>(1-2), 241–295.
</div>
</div></section></div></main> <!-- /main -->

<script>
  // htmlwidgets need to know to resize themselves when slides are shown/hidden.
  // Fire the "slideenter" event (handled by htmlwidgets.js) when the current
  // slide changes (different for each slide format).
  (function () {
    // dispatch for htmlwidgets
    function fireSlideEnter() {
      const event = window.document.createEvent("Event");
      event.initEvent("slideenter", true, true);
      window.document.dispatchEvent(event);
    }

    function fireSlideChanged(previousSlide, currentSlide) {
      fireSlideEnter();

      // dispatch for shiny
      if (window.jQuery) {
        if (previousSlide) {
          window.jQuery(previousSlide).trigger("hidden");
        }
        if (currentSlide) {
          window.jQuery(currentSlide).trigger("shown");
        }
      }
    }

    // hookup for slidy
    if (window.w3c_slidy) {
      window.w3c_slidy.add_observer(function (slide_num) {
        // slide_num starts at position 1
        fireSlideChanged(null, w3c_slidy.slides[slide_num - 1]);
      });
    }

  })();
</script>

<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const disableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'prefetch';
    }
  }
  const enableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'stylesheet';
    }
  }
  const manageTransitions = (selector, allowTransitions) => {
    const els = window.document.querySelectorAll(selector);
    for (let i=0; i < els.length; i++) {
      const el = els[i];
      if (allowTransitions) {
        el.classList.remove('notransition');
      } else {
        el.classList.add('notransition');
      }
    }
  }
  const toggleGiscusIfUsed = (isAlternate, darkModeDefault) => {
    const baseTheme = document.querySelector('#giscus-base-theme')?.value ?? 'light';
    const alternateTheme = document.querySelector('#giscus-alt-theme')?.value ?? 'dark';
    let newTheme = '';
    if(darkModeDefault) {
      newTheme = isAlternate ? baseTheme : alternateTheme;
    } else {
      newTheme = isAlternate ? alternateTheme : baseTheme;
    }
    const changeGiscusTheme = () => {
      // From: https://github.com/giscus/giscus/issues/336
      const sendMessage = (message) => {
        const iframe = document.querySelector('iframe.giscus-frame');
        if (!iframe) return;
        iframe.contentWindow.postMessage({ giscus: message }, 'https://giscus.app');
      }
      sendMessage({
        setConfig: {
          theme: newTheme
        }
      });
    }
    const isGiscussLoaded = window.document.querySelector('iframe.giscus-frame') !== null;
    if (isGiscussLoaded) {
      changeGiscusTheme();
    }
  }
  const toggleColorMode = (alternate) => {
    // Switch the stylesheets
    const alternateStylesheets = window.document.querySelectorAll('link.quarto-color-scheme.quarto-color-alternate');
    manageTransitions('#quarto-margin-sidebar .nav-link', false);
    if (alternate) {
      enableStylesheet(alternateStylesheets);
      for (const sheetNode of alternateStylesheets) {
        if (sheetNode.id === "quarto-bootstrap") {
          toggleBodyColorMode(sheetNode);
        }
      }
    } else {
      disableStylesheet(alternateStylesheets);
      toggleBodyColorPrimary();
    }
    manageTransitions('#quarto-margin-sidebar .nav-link', true);
    // Switch the toggles
    const toggles = window.document.querySelectorAll('.quarto-color-scheme-toggle');
    for (let i=0; i < toggles.length; i++) {
      const toggle = toggles[i];
      if (toggle) {
        if (alternate) {
          toggle.classList.add("alternate");     
        } else {
          toggle.classList.remove("alternate");
        }
      }
    }
    // Hack to workaround the fact that safari doesn't
    // properly recolor the scrollbar when toggling (#1455)
    if (navigator.userAgent.indexOf('Safari') > 0 && navigator.userAgent.indexOf('Chrome') == -1) {
      manageTransitions("body", false);
      window.scrollTo(0, 1);
      setTimeout(() => {
        window.scrollTo(0, 0);
        manageTransitions("body", true);
      }, 40);  
    }
  }
  const isFileUrl = () => { 
    return window.location.protocol === 'file:';
  }
  const hasAlternateSentinel = () => {  
    let styleSentinel = getColorSchemeSentinel();
    if (styleSentinel !== null) {
      return styleSentinel === "alternate";
    } else {
      return false;
    }
  }
  const setStyleSentinel = (alternate) => {
    const value = alternate ? "alternate" : "default";
    if (!isFileUrl()) {
      window.localStorage.setItem("quarto-color-scheme", value);
    } else {
      localAlternateSentinel = value;
    }
  }
  const getColorSchemeSentinel = () => {
    if (!isFileUrl()) {
      const storageValue = window.localStorage.getItem("quarto-color-scheme");
      return storageValue != null ? storageValue : localAlternateSentinel;
    } else {
      return localAlternateSentinel;
    }
  }
  const darkModeDefault = false;
  let localAlternateSentinel = darkModeDefault ? 'alternate' : 'default';
  // Dark / light mode switch
  window.quartoToggleColorScheme = () => {
    // Read the current dark / light value 
    let toAlternate = !hasAlternateSentinel();
    toggleColorMode(toAlternate);
    setStyleSentinel(toAlternate);
    toggleGiscusIfUsed(toAlternate, darkModeDefault);
  };
  // Ensure there is a toggle, if there isn't float one in the top right
  if (window.document.querySelector('.quarto-color-scheme-toggle') === null) {
    const a = window.document.createElement('a');
    a.classList.add('top-right');
    a.classList.add('quarto-color-scheme-toggle');
    a.href = "";
    a.onclick = function() { try { window.quartoToggleColorScheme(); } catch {} return false; };
    const i = window.document.createElement("i");
    i.classList.add('bi');
    a.appendChild(i);
    window.document.body.appendChild(a);
  }
  // Switch to dark mode if need be
  if (hasAlternateSentinel()) {
    toggleColorMode(true);
  } else {
    toggleColorMode(false);
  }
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp("https:\/\/giulianocolosimo\.github\.com");
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-left">
<p>© Copyright 2024 Giuliano Colosimo</p>
</div>   
    <div class="nav-footer-center">
      &nbsp;
    <div class="toc-actions d-sm-block d-md-none"><ul><li><a href="https://github.com/quarto-dev/quarto-demo/issues/new" class="toc-action"><i class="bi bi-github"></i>Report an issue</a></li></ul></div></div>
    <div class="nav-footer-right">
<p>This page is built with ❤️ and <a href="https://quarto.org/">Quarto</a>. Inspiration for this site comes from <a href="https://www.mm218.dev">here</a>.</p>
</div>
  </div>
</footer>




</body></html>