<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.4.553">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="dcterms.date" content="2024-05-07">

<title>Biologia della Conservazione 23/24 - Biologia della Conservazione</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../">
<script src="../../site_libs/quarto-html/quarto.js"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-text-highlighting-styles">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-text-highlighting-styles">
<script src="../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-bootstrap" data-mode="light">
<link href="../../site_libs/bootstrap/bootstrap-dark.min.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-bootstrap" data-mode="dark">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>

  <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>

<meta property="og:title" content="Biologia della Conservazione 23/24 - Biologia della Conservazione">
<meta property="og:description" content="Class 13 - Population Structure/Hybridization">
<meta property="og:site_name" content="Biologia della Conservazione 23/24">
<meta property="og:locale" content="en_US">
</head>

<body class="nav-sidebar docked nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="../../index.html">
    <span class="navbar-title">Biologia della Conservazione 23/24</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item">
    <a class="nav-link active" href="../../index.html" aria-current="page"> 
<span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../calendar.html"> 
<span class="menu-text">Calendar</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../../contact.html"> 
<span class="menu-text">Contact</span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
          <div class="quarto-navbar-tools">
  <a href="" class="quarto-color-scheme-toggle quarto-navigation-tool  px-1" onclick="window.quartoToggleColorScheme(); return false;" title="Toggle dark mode"><i class="bi"></i></a>
  <a href="" class="quarto-reader-toggle quarto-navigation-tool px-1" onclick="window.quartoToggleReader(); return false;" title="Toggle reader mode">
  <div class="quarto-reader-toggle-btn">
  <i class="bi"></i>
  </div>
</a>
</div>
      </div> <!-- /container-fluid -->
    </nav>
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item">Biologia della Conservazione</li></ol></nav>
        <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation docked overflow-auto">
    <div class="pt-lg-2 mt-2 text-center sidebar-header">
      <a href="../../index.html" class="sidebar-logo-link">
      <img src="../.././figures/avatar.jpg" alt="" class="sidebar-logo py-0 d-lg-inline d-none">
      </a>
      <div class="sidebar-tools-main">
    <a href="mailto:giuliano.colosimo@protonmail.com" title="" class="quarto-navigation-tool px-1" aria-label=""><i class="bi bi-envelope"></i></a>
</div>
      </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true">
 <span class="menu-text">Classes</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../.././classes/1/class_1.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Class 1 - Introduction to conservation biology</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../.././classes/2/class_2.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Class 2 - Biodiversity</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../.././classes/3/class_3.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Class 3 - Biodiversity &amp; Extinctions</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../.././classes/4/class_4.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Class 4 - Populations (1)</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../.././classes/5/class_5.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Class 5 - Populations (2) - Intro to Conservation Genetics</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../.././classes/6/class_6.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Class 6 - Conservation Genetics</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../.././classes/7/class_7.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Class 7 - Genetic Drift/Effective Population Size</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../.././classes/8/class_8.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Class 8 - Manuscript discussion-Genetic Drift/Effective Population Size</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../.././classes/9/class_9.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Class 9 - Manuscript discussion-Genetic Drift/Effective Population Size</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../.././classes/10/class_10.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Class 10 - Inbreeding</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../.././classes/11/class_11.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Class 11 - Arguments review for mid term</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../.././classes/12/class_12.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Class 12 - Mid-term exam revision/Genetic load</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../.././classes/13/class_13.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Class 13 - Population Structure/Hybridization_1</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../.././classes/14/class_14.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Class 14 - Population Structure/Hybridization_2</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../.././classes/15/class_15.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Class 15 - Heritability</span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#topics" id="toc-topics" class="nav-link active" data-scroll-target="#topics">Topics</a></li>
  <li><a href="#population-structure" id="toc-population-structure" class="nav-link" data-scroll-target="#population-structure">Population structure</a></li>
  <li><a href="#population-structure-1" id="toc-population-structure-1" class="nav-link" data-scroll-target="#population-structure-1">Population structure</a></li>
  <li><a href="#population-structure-2" id="toc-population-structure-2" class="nav-link" data-scroll-target="#population-structure-2">Population structure</a></li>
  <li><a href="#population-structure-3" id="toc-population-structure-3" class="nav-link" data-scroll-target="#population-structure-3">Population structure</a></li>
  <li><a href="#population-structure-4" id="toc-population-structure-4" class="nav-link" data-scroll-target="#population-structure-4">Population structure</a></li>
  <li><a href="#population-structure-5" id="toc-population-structure-5" class="nav-link" data-scroll-target="#population-structure-5">Population structure</a></li>
  <li><a href="#population-structure-6" id="toc-population-structure-6" class="nav-link" data-scroll-target="#population-structure-6">Population structure</a></li>
  <li><a href="#population-structure-7" id="toc-population-structure-7" class="nav-link" data-scroll-target="#population-structure-7">Population structure</a></li>
  <li><a href="#population-structure-8" id="toc-population-structure-8" class="nav-link" data-scroll-target="#population-structure-8">Population structure</a></li>
  <li><a href="#f-statistics" id="toc-f-statistics" class="nav-link" data-scroll-target="#f-statistics"><span class="math inline">\(F\)</span> statistics</a></li>
  <li><a href="#f_is" id="toc-f_is" class="nav-link" data-scroll-target="#f_is"><span class="math inline">\(F_{IS}\)</span></a></li>
  <li><a href="#f_st" id="toc-f_st" class="nav-link" data-scroll-target="#f_st"><span class="math inline">\(F_{ST}\)</span></a></li>
  <li><a href="#f_it" id="toc-f_it" class="nav-link" data-scroll-target="#f_it"><span class="math inline">\(F_{IT}\)</span></a></li>
  <li><a href="#the-wahlund-effect" id="toc-the-wahlund-effect" class="nav-link" data-scroll-target="#the-wahlund-effect">The Wahlund effect</a></li>
  <li><a href="#population-structure-and-habitat-fragmentation" id="toc-population-structure-and-habitat-fragmentation" class="nav-link" data-scroll-target="#population-structure-and-habitat-fragmentation">Population structure and habitat fragmentation</a></li>
  <li><a href="#population-structure-and-habitat-fragmentation-1" id="toc-population-structure-and-habitat-fragmentation-1" class="nav-link" data-scroll-target="#population-structure-and-habitat-fragmentation-1">Population structure and habitat fragmentation</a></li>
  <li><a href="#population-structure-and-habitat-fragmentation-2" id="toc-population-structure-and-habitat-fragmentation-2" class="nav-link" data-scroll-target="#population-structure-and-habitat-fragmentation-2">Population structure and habitat fragmentation</a></li>
  <li><a href="#population-structure-and-habitat-fragmentation-3" id="toc-population-structure-and-habitat-fragmentation-3" class="nav-link" data-scroll-target="#population-structure-and-habitat-fragmentation-3">Population structure and habitat fragmentation</a></li>
  <li><a href="#population-structure-and-habitat-fragmentation-4" id="toc-population-structure-and-habitat-fragmentation-4" class="nav-link" data-scroll-target="#population-structure-and-habitat-fragmentation-4">Population structure and habitat fragmentation</a>
  <ul class="collapse">
  <li><a href="#gene-flow" id="toc-gene-flow" class="nav-link" data-scroll-target="#gene-flow">Gene flow</a></li>
  </ul></li>
  <li><a href="#the-black-footed-rock-wallaby" id="toc-the-black-footed-rock-wallaby" class="nav-link" data-scroll-target="#the-black-footed-rock-wallaby">The black footed rock wallaby</a></li>
  <li><a href="#example---1" id="toc-example---1" class="nav-link" data-scroll-target="#example---1">Example - 1</a></li>
  <li><a href="#example---2" id="toc-example---2" class="nav-link" data-scroll-target="#example---2">Example - 2</a></li>
  <li><a href="#example---3" id="toc-example---3" class="nav-link" data-scroll-target="#example---3">Example - 3</a></li>
  <li><a href="#example---4" id="toc-example---4" class="nav-link" data-scroll-target="#example---4">Example - 4</a></li>
  <li><a href="#migration-and-gene-flow" id="toc-migration-and-gene-flow" class="nav-link" data-scroll-target="#migration-and-gene-flow">Migration and gene flow</a></li>
  <li><a href="#references" id="toc-references" class="nav-link" data-scroll-target="#references">References</a></li>
  </ul>
<div class="toc-actions"><ul><li><a href="https://github.com/quarto-dev/quarto-demo/issues/new" class="toc-action"><i class="bi bi-github"></i>Report an issue</a></li></ul></div><div class="quarto-alternate-formats"><h2>Other Formats</h2><ul><li><a href="class_13.html"><i class="bi bi-file-slides"></i>RevealJS</a></li></ul></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Biologia della Conservazione</h1>
<p class="subtitle lead">Class 13 - Population Structure/Hybridization</p>
</div>



<div class="quarto-title-meta">

    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">May 7, 2024</p>
    </div>
  </div>
  
    
  </div>
  


</header>


<table class="table">
<thead>
<tr class="header">
<th>Student</th>
<th>Title</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>FALVO</td>
<td><a href="https://drive.proton.me/urls/9WVQGE6J5M#GEndO0gEvVJq">Developing gene drive technologies to eradicate invasive rodents from islands</a></td>
</tr>
<tr class="even">
<td>D’ALESSANDRO</td>
<td><a href="https://drive.proton.me/urls/HQMV65ETB0#vB21xDrPrAXN">Population genetic analysis of the recently rediscovered hula painted frog (<em>Latonia nigriventer</em>) reveals high genetic diversity and low inbreeding</a></td>
</tr>
<tr class="odd">
<td>PIROLI</td>
<td><a href="https://drive.proton.me/urls/TD1F4DDC70#hzwxC3gFnM51">First evidence of recruitment in critically endangered Galapagos pink iguanas</a></td>
</tr>
<tr class="even">
<td>TRONCONE</td>
<td><a href="https://drive.proton.me/urls/SW4YPN2ZGC#RWYgHuXfrvK4">Genetic characterization and implications for conservation of the last autochthonous Mouflon population in Europe</a></td>
</tr>
<tr class="odd">
<td>GARCIA</td>
<td><a href="https://drive.proton.me/urls/FJ5FMEQ8EC#QhbaqVfIyFkE">The critically endangered vaquita is not doomed to extinction by inbreeding depression</a></td>
</tr>
</tbody>
</table>
<hr>
<section id="topics" class="level2">
<h2 class="anchored" data-anchor-id="topics">Topics</h2>
<div class="columns">
<div class="column" style="width:50%;">
<ol type="1">
<li><span style="color: green;">Introduction to conservation biology</span></li>
<li><span style="color: green;">Biodiversity</span></li>
<li><span style="color: green;">Biodiversity and extinctions</span></li>
<li><span style="color: green;">Populations</span></li>
<li><span style="color: green;">Basics of conservation genetics</span></li>
<li><span style="color: green;">Genetic drift</span></li>
<li><span style="color: green;">Effective population size</span></li>
<li><span style="color: green;">Inbreeding depression</span></li>
<li><span style="color: orange;">Population structure</span></li>
<li><span style="color: orange;">Hybridization</span></li>
<li><span style="color: blue;">Applying genetic information to conservation</span></li>
</ol>
</div><div class="column" style="width:50%;">
<ol start="12" type="1">
<li><span style="color: blue;">Managing genetic diversity in captive populations</span></li>
<li><span style="color: blue;">Managing genetic diversity in the wild</span></li>
<li><span style="color: blue;">Population viability analysis</span></li>
<li><span style="color: blue;">Invasive species</span></li>
<li><span style="color: blue;">CITES and trade regulations of endangered species</span></li>
<li><span style="color: blue;">Conservation and climate change</span></li>
<li><span style="color: blue;">Future climate impacts</span></li>
<li><span style="color: blue;">Conservation economics</span></li>
<li><span style="color: blue;">Conservation technology</span></li>
</ol>
</div>
</div>
</section>
<section id="population-structure" class="level2">
<h2 class="anchored" data-anchor-id="population-structure">Population structure</h2>
<ul>
<li>In most species individuals are scattered in local conglomerates that are called <em>demes</em> (<em>i.e.</em>, local populations or subpopulations)</li>
</ul>
<p>. . .</p>
<div class="columns">
<div class="column" style="width:50%;">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="./class_13_1.png" class="img-fluid figure-img"></p>
<figcaption>After <span class="citation" data-cites="Akakaya2007">Akçakaya, Mills, &amp; Doncaster (<a href="#ref-Akakaya2007" role="doc-biblioref">2007</a>)</span></figcaption>
</figure>
</div>
</div><div class="column" style="width:50%;">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="./class_13_2.jpg" class="img-fluid figure-img"></p>
<figcaption>After <span class="citation" data-cites="Frankham2010">Frankham, Ballou, &amp; Briscoe (<a href="#ref-Frankham2010" role="doc-biblioref">2010</a>)</span></figcaption>
</figure>
</div>
</div>
</div>
</section>
<section id="population-structure-1" class="level2">
<h2 class="anchored" data-anchor-id="population-structure-1">Population structure</h2>
<ul>
<li><p>A metapopulation is defined as a set of discrete populations (demes) of the same species, in the same general geographical area, potentially able to exchange individuals through migration, dispersal, or human-mediated movement <span class="citation" data-cites="Akakaya2007 Hanski1997">(<a href="#ref-Akakaya2007" role="doc-biblioref">Akçakaya et al., 2007</a>; <a href="#ref-Hanski1997" role="doc-biblioref">Hanski &amp; Simberloff, 1997</a>)</span></p></li>
<li><p>This subdivision (or structure) of populations can be natural and/or human driven</p></li>
<li><p>Regardless of the cause, the existence of population structure in many species forces us to think at the genetic variation within a species at two primary levels:</p>
<ol type="1">
<li>Genetic variation within local demes.</li>
<li>Genetic variation bewteen local demes.</li>
</ol></li>
<li><p>Not all species are scattered!</p></li>
</ul>
</section>
<section id="population-structure-2" class="level2">
<h2 class="anchored" data-anchor-id="population-structure-2">Population structure</h2>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="./class_13_3.png" class="img-fluid figure-img"></p>
<figcaption>After <span class="citation" data-cites="Allendorf2022">Allendorf et al. (<a href="#ref-Allendorf2022" role="doc-biblioref">2022</a>)</span>, p.&nbsp;184</figcaption>
</figure>
</div>
</section>
<section id="population-structure-3" class="level2">
<h2 class="anchored" data-anchor-id="population-structure-3">Population structure</h2>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="./class_13_4.png" class="img-fluid figure-img"></p>
<figcaption>Boreal forest distibution in North America. Available <a href="https://canadiangeographic.ca/articles/new-framework-identifies-climate-change-refugia-in-boreal-forest/">here</a></figcaption>
</figure>
</div>
</section>
<section id="population-structure-4" class="level2">
<h2 class="anchored" data-anchor-id="population-structure-4">Population structure</h2>
<ul>
<li>Even in the absence of barriers to gene flow, genetic differences can accrue so that individuals that are very distant from each other may present significant differences at the genetic level: isolation by distance</li>
</ul>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="./class_13_5.png" class="img-fluid figure-img"></p>
<figcaption>Variability in the correlation between species genetic diversity and geographic distance. ranges. Modified from <span class="citation" data-cites="Forbes1999">Forbes &amp; Hogg (<a href="#ref-Forbes1999" role="doc-biblioref">1999</a>)</span> with additional unpublished data from M.K. Schwartz <span class="citation" data-cites="Allendorf2022">(<a href="#ref-Allendorf2022" role="doc-biblioref">Allendorf et al., 2022</a>)</span>.</figcaption>
</figure>
</div>
</section>
<section id="population-structure-5" class="level2">
<h2 class="anchored" data-anchor-id="population-structure-5">Population structure</h2>
<ul>
<li>Describing the patterns and amount of differentiation among demes is of the uttermost importance to develop effective conservation strategies <span class="citation" data-cites="Thatte2018">(see for example <a href="#ref-Thatte2018" role="doc-biblioref">Thatte, Joshi, Vaidyanathan, Landguth, &amp; Ramakrishnan, 2018</a>)</span>.</li>
</ul>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="./class_13_6.png" class="img-fluid figure-img"></p>
<figcaption>Study area in <span class="citation" data-cites="Thatte2018">Thatte et al. (<a href="#ref-Thatte2018" role="doc-biblioref">2018</a>)</span>. Genetically identified tigers as red dots</figcaption>
</figure>
</div>
</section>
<section id="population-structure-6" class="level2">
<h2 class="anchored" data-anchor-id="population-structure-6">Population structure</h2>
<ul>
<li><span class="citation" data-cites="Frankham2011">Frankham et al. (<a href="#ref-Frankham2011" role="doc-biblioref">2011</a>)</span> describe the risks of outbreeding depression when translocating individuals from genetically distinct populations to supplement imperiled ones.</li>
</ul>
<p>. . .</p>
<div class="columns">
<div class="column" style="width:50%;">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="./class_13_7.png" class="img-fluid figure-img" style="width:85.0%"></p>
<figcaption>After <span class="citation" data-cites="Welch2017">Welch et al. (<a href="#ref-Welch2017" role="doc-biblioref">2017</a>)</span>.</figcaption>
</figure>
</div>
</div><div class="column" style="width:50%;">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="./class_13_8.png" class="img-fluid figure-img" style="width:83.0%"></p>
<figcaption>After <span class="citation" data-cites="Welch2017">Welch et al. (<a href="#ref-Welch2017" role="doc-biblioref">2017</a>)</span>.</figcaption>
</figure>
</div>
</div>
</div>
</section>
<section id="population-structure-7" class="level2">
<h2 class="anchored" data-anchor-id="population-structure-7">Population structure</h2>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="./class_13_9.png" class="img-fluid figure-img"></p>
<figcaption>After <span class="citation" data-cites="Colosimo2021b">Colosimo et al. (<a href="#ref-Colosimo2021b" role="doc-biblioref">2021</a>)</span></figcaption>
</figure>
</div>
</section>
<section id="population-structure-8" class="level2">
<h2 class="anchored" data-anchor-id="population-structure-8">Population structure</h2>
<ul>
<li>Developing priorities for the conservation of a species often requires an understanding of adaptive genetic differentiation among populations <span class="citation" data-cites="Allendorf2022">(<a href="#ref-Allendorf2022" role="doc-biblioref">Allendorf et al., 2022</a>)</span>.</li>
<li>To understand how genetic variation is distributed at different loci within subdivided populations it is necessary to look at different interacting forces like gene flow, genetic drift and natural selection.</li>
<li><span class="math inline">\(F\)</span> statistics <span class="citation" data-cites="Wright1931 Wright1951 Malecot1948">(<a href="#ref-Malecot1948" role="doc-biblioref">Malécot, 1948</a>; <a href="#ref-Wright1931" role="doc-biblioref">Wright, 1931</a>, <a href="#ref-Wright1951" role="doc-biblioref">1951</a>)</span> are undoubtedly the most used indexes to describe genetic differentiation in and between demes.</li>
<li><span class="math inline">\(F\)</span> statistics are, in short, inbreeding coefficients measuring different types of inbreeding at different levels.</li>
</ul>
</section>
<section id="f-statistics" class="level2">
<h2 class="anchored" data-anchor-id="f-statistics"><span class="math inline">\(F\)</span> statistics</h2>
<ul>
<li>We recognize three primary coefficients: <span class="math inline">\(F_{IS}\)</span>, <span class="math inline">\(F_{IT}\)</span>, and <span class="math inline">\(F_{ST}\)</span>.</li>
<li><span class="math inline">\(F_{IS}\)</span> measures Hardy–Weinberg departures within local demes.</li>
<li><span class="math inline">\(F_{ST}\)</span> measures divergence in allele frequencies among demes.</li>
<li><span class="math inline">\(F_{IT}\)</span> measures Hardy–Weinberg departures in the metapopulation due to <span class="math inline">\(F_{IS}\)</span> (nonrandom mating within local demes) and <span class="math inline">\(F_{ST}\)</span> (allele frequency divergence among demes).</li>
<li>The original <span class="math inline">\(F\)</span> statistics developed by wright focused on loci with only two alleles.</li>
<li><span class="citation" data-cites="Nei1977">Nei (<a href="#ref-Nei1977" role="doc-biblioref">1977</a>)</span> extended these measures to multiple alleles and used a different notation to name what he called <em>analysis of gene diversity</em>: <span class="math inline">\(G_{IS}\)</span>, <span class="math inline">\(G_{IT}\)</span>, and <span class="math inline">\(G_{ST}\)</span>. Often the two groups of indexes are used interchangeably.</li>
</ul>
</section>
<section id="f_is" class="level2">
<h2 class="anchored" data-anchor-id="f_is"><span class="math inline">\(F_{IS}\)</span></h2>
<p>This is a measure of departure from HW proportions within local demes.</p>
<p><span class="math display">\[
\label{fis}
\tag{13.1}
F_{IS} = 1-\frac{H_O}{H_S}
\]</span></p>
<p>where <span class="math inline">\(H_O\)</span> is the observed heterozygosity averaged over all demes while <span class="math inline">\(H_S\)</span> is the expected heterozygosity averaged over all demes. <span class="math inline">\(F_{IS}\)</span> values can range from <span class="math inline">\(-1 \leq 0 \leq 1\)</span>, with positive values indicating an excess of homozygotes and negative values indicating an excess of heterozygotes <span class="citation" data-cites="Allendorf2022">(<a href="#ref-Allendorf2022" role="doc-biblioref">Allendorf et al., 2022</a>)</span>.</p>
</section>
<section id="f_st" class="level2">
<h2 class="anchored" data-anchor-id="f_st"><span class="math inline">\(F_{ST}\)</span></h2>
<p>Often called <em>fixation index</em>, it is a measure of genetic divergence among demes.</p>
<p><span class="math display">\[
\label{fst}
\tag{13.2}
F_{ST} = 1-\frac{H_S}{H_T}
\]</span></p>
<p>with <span class="math inline">\(H_T\)</span> as the expected HW heterozygosity if the metapopulation were panmictic, and <span class="math inline">\(H_S\)</span> is the HW proportion of heterozygotes in each separate deme and then averaged over all demes. This measure varies from <span class="math inline">\(0\)</span> (demes all have equal allele frequencies) to <span class="math inline">\(1\)</span> (demes are all fixed for different alleles).</p>
</section>
<section id="f_it" class="level2">
<h2 class="anchored" data-anchor-id="f_it"><span class="math inline">\(F_{IT}\)</span></h2>
<p>Measures the departure from HW proportions due to departures from HW proportions within local demes and divergence among demes <span class="citation" data-cites="Allendorf2022">(<a href="#ref-Allendorf2022" role="doc-biblioref">Allendorf et al., 2022</a>)</span>.</p>
<p><span class="math display">\[
\label{fit}
\tag{13.3}
F_{IT} = 1-\frac{H_O}{H_T}
\]</span></p>
<p>where <span class="math inline">\(H_O\)</span> is the observed heterozygosity averaged over all demes while <span class="math inline">\(H_T\)</span> is the expected HW heterozygosity if the metapopulation were panmictic.</p>
<p>. . .</p>
<ul>
<li>The fixation indexes described in the previous slides are related by the following expression:</li>
</ul>
<p><span class="math display">\[
\label{findexrelation}
\tag{13.4}
F_{IT}= F_{IS}+F_{ST}-(F_{IS}*F_{ST})
\]</span></p>
</section>
<section id="the-wahlund-effect" class="level2">
<h2 class="anchored" data-anchor-id="the-wahlund-effect">The Wahlund effect</h2>
<ul>
<li>A deficit of heterozygosity due to the presence of differential distribution of genetic diversity among demes is often called Wahlund effect, for the the Swedish geneticist who described it <span class="citation" data-cites="Wahlund1928">(<a href="#ref-Wahlund1928" role="doc-biblioref">Wahlund, 1928</a>)</span>.</li>
<li>He analyzed the excess of homozygotes and deficit of heterozygotes in terms of the variance of allele frequencies among S deme <span class="citation" data-cites="Allendorf2022">(<a href="#ref-Allendorf2022" role="doc-biblioref">Allendorf et al., 2022</a>)</span>.</li>
</ul>
<p><span class="math display">\[
\label{wahlund}
\tag{13.5}
Var(q)=\frac{1}{S}\sum(q_i-\bar{q})^2 \\ \\ F_{ST} = \frac{Var(q)}{pq}
\]</span></p>
</section>
<section id="population-structure-and-habitat-fragmentation" class="level2">
<h2 class="anchored" data-anchor-id="population-structure-and-habitat-fragmentation">Population structure and habitat fragmentation</h2>
<ul>
<li>Habitat fragmentation is a combination of two processes: reduction in total area, and the creation of separate islands.</li>
<li>Human induced fragmentation and habitat loss is recognized as the main cause of biodiversity loss <span class="citation" data-cites="deBaan2013">(<a href="#ref-deBaan2013" role="doc-biblioref">Baan, Alkemade, &amp; Koellner, 2013</a>)</span>.</li>
<li>Habitat fragmentation causes an overall reduction in population size.</li>
<li>Habitat fragmentation also causes an overall reduction in gene flow among the patches it creates.</li>
<li>Inhospitable, cleared habitat among fragments typically inhibits gene flow. The same goes for roads rivers, water bodies, mountains, or any other artificial or natural barrier <span class="citation" data-cites="Colosimo2014">Riley et al. (<a href="#ref-Riley2006" role="doc-biblioref">2006</a>)</span>.</li>
<li>With restricted gene flow fragmentation is highly deleterious in the long term.</li>
<li>Single Large or Several Small <span class="citation" data-cites="Frankham2010">(<a href="#ref-Frankham2010" role="doc-biblioref">Frankham et al., 2010</a>)</span>.</li>
</ul>
</section>
<section id="population-structure-and-habitat-fragmentation-1" class="level2">
<h2 class="anchored" data-anchor-id="population-structure-and-habitat-fragmentation-1">Population structure and habitat fragmentation</h2>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="./class_13_10.png" class="img-fluid figure-img"></p>
<figcaption>Fragmentation of Atlantic forest. After <span class="citation" data-cites="Luci2011">Nunes (<a href="#ref-Luci2011" role="doc-biblioref">2011</a>)</span></figcaption>
</figure>
</div>
</section>
<section id="population-structure-and-habitat-fragmentation-2" class="level2">
<h2 class="anchored" data-anchor-id="population-structure-and-habitat-fragmentation-2">Population structure and habitat fragmentation</h2>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="./class_13_11.jpg" class="img-fluid figure-img"></p>
<figcaption>Single large (SL) VS several small (SS). After <span class="citation" data-cites="Frankham2010">Frankham et al. (<a href="#ref-Frankham2010" role="doc-biblioref">2010</a>)</span></figcaption>
</figure>
</div>
</section>
<section id="population-structure-and-habitat-fragmentation-3" class="level2">
<h2 class="anchored" data-anchor-id="population-structure-and-habitat-fragmentation-3">Population structure and habitat fragmentation</h2>
<p>Fragmentation happens in two steps:</p>
<ol type="1">
<li>Initial genetic subdivision of the population.</li>
<li>Cumulative diversification through genetic drift, selection, inbreeding etc. etc. etc.</li>
</ol>
<p>After an initial fragmentation, different fragments will have different initial allele frequencies. This diversification can be described by the binomial sampling variance (<span class="math inline">\(\sigma_p^2 = \frac{pq}{2N}\)</span>) and the smaller the fragment, the higher the variance.</p>
<p>. . .</p>
<p>The degree of differentiation among fragments can be described by partitioning the overall inbreeding into components within and among populations (<span class="math inline">\(F\)</span> statistics).</p>
<p>. . .</p>
<p>We will look at some examples on estimating <span class="math inline">\(F\)</span> statistics from genotype data.</p>
</section>
<section id="population-structure-and-habitat-fragmentation-4" class="level2">
<h2 class="anchored" data-anchor-id="population-structure-and-habitat-fragmentation-4">Population structure and habitat fragmentation</h2>
<section id="gene-flow" class="level3">
<h3 class="anchored" data-anchor-id="gene-flow">Gene flow</h3>
<p>What is gene flow?</p>
<ul>
<li>Movement of alleles between population due to migration.</li>
</ul>
<p>The effects of fragmentation on gene flow depend on:</p>
<ul>
<li>Number of population fragments.</li>
<li>Distribution of population sizes in the fragments.</li>
<li>Distance between fragments.</li>
<li>Dispersal ability of the species.</li>
<li>Migration rates among fragments.</li>
<li>Environment of the matrix among fragments.</li>
<li>Times since fragmentation and extinction and recolonization rates across fragments.</li>
</ul>
</section>
</section>
<section id="the-black-footed-rock-wallaby" class="level2">
<h2 class="anchored" data-anchor-id="the-black-footed-rock-wallaby">The black footed rock wallaby</h2>
<div class="columns">
<div class="column" style="width:50%;">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="./class_13_12.jpg" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:20.0%"></p>
</figure>
</div>
</div><div class="column" style="width:50%;">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="./class_13_14.jpg" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:50.0%"></p>
</figure>
</div>
</div>
</div>
<table class="table">
<colgroup>
<col style="width: 24%">
<col style="width: 26%">
<col style="width: 24%">
<col style="width: 24%">
</colgroup>
<thead>
<tr class="header">
<th>Location</th>
<th>Prop. of polymorphic loci</th>
<th>Mean # of alleles</th>
<th>Average heterozygosity</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Barrow Island (1)</td>
<td>0.1</td>
<td>1.2</td>
<td>0.05</td>
</tr>
<tr class="even">
<td>Exmouth (2)</td>
<td>1.0</td>
<td>3.4</td>
<td>0.62</td>
</tr>
<tr class="odd">
<td>Wheatbelt (3)</td>
<td>1.0</td>
<td>4.4</td>
<td>0.56</td>
</tr>
</tbody>
</table>
<ul>
<li>Data from <span class="citation" data-cites="Eldridge1999">Eldridge et al. (<a href="#ref-Eldridge1999" role="doc-biblioref">1999</a>)</span>, images after <span class="citation" data-cites="Frankham2010">Frankham et al. (<a href="#ref-Frankham2010" role="doc-biblioref">2010</a>)</span>.</li>
</ul>
</section>
<section id="example---1" class="level2">
<h2 class="anchored" data-anchor-id="example---1">Example - 1</h2>
<ul>
<li>Populations with same allele frequencies. One with random mating the other one with inbreeding.</li>
</ul>
<table class="table">
<colgroup>
<col style="width: 18%">
<col style="width: 15%">
<col style="width: 15%">
<col style="width: 15%">
<col style="width: 20%">
<col style="width: 7%">
<col style="width: 7%">
</colgroup>
<thead>
<tr class="header">
<th>Population</th>
<th><span class="math inline">\(A_1A_1\)</span></th>
<th><span class="math inline">\(A_1A_2\)</span></th>
<th><span class="math inline">\(A_2A_2\)</span></th>
<th>Allele Freq</th>
<th>F</th>
<th>He</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>1</td>
<td>0.25</td>
<td>0.5</td>
<td>0.25</td>
<td><span class="math inline">\(p=0.5\)</span></td>
<td>0</td>
<td>0.5</td>
</tr>
<tr class="even">
<td></td>
<td></td>
<td></td>
<td></td>
<td><span class="math inline">\(q=0.5\)</span></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td>2</td>
<td>0.4</td>
<td>0.2</td>
<td>0.4</td>
<td><span class="math inline">\(p=0.5\)</span></td>
<td>0.6</td>
<td>0.5</td>
</tr>
<tr class="even">
<td></td>
<td></td>
<td></td>
<td></td>
<td><span class="math inline">\(q=0.5\)</span></td>
<td></td>
<td></td>
</tr>
</tbody>
</table>
<ul>
<li><span class="math inline">\(H_O\)</span> observed heterozygosity averaged over all demes.</li>
<li><span class="math inline">\(H_S\)</span> expected heterozygosity averaged over all demes.</li>
<li><span class="math inline">\(H_T\)</span> expected heterozygosity of the metapopulation.</li>
<li><span class="math inline">\(F_{IS} = 1-\frac{H_O}{H_S}\)</span>; <span class="math inline">\(F_{IT} = 1-\frac{H_O}{H_T}\)</span>; <span class="math inline">\(F_{ST} = 1-\frac{H_S}{H_T}\)</span>.</li>
</ul>
</section>
<section id="example---2" class="level2">
<h2 class="anchored" data-anchor-id="example---2">Example - 2</h2>
<ul>
<li>Random mating populations with different allele frequencies.</li>
</ul>
<table class="table">
<colgroup>
<col style="width: 18%">
<col style="width: 15%">
<col style="width: 15%">
<col style="width: 15%">
<col style="width: 19%">
<col style="width: 7%">
<col style="width: 9%">
</colgroup>
<thead>
<tr class="header">
<th>Population</th>
<th><span class="math inline">\(A_1A_1\)</span></th>
<th><span class="math inline">\(A_1A_2\)</span></th>
<th><span class="math inline">\(A_2A_2\)</span></th>
<th>Allele Freq</th>
<th>F</th>
<th>He</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>1</td>
<td>0.25</td>
<td>0.5</td>
<td>0.25</td>
<td><span class="math inline">\(p=0.5\)</span></td>
<td>0</td>
<td>0.5</td>
</tr>
<tr class="even">
<td></td>
<td></td>
<td></td>
<td></td>
<td><span class="math inline">\(q=0.5\)</span></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td>2</td>
<td>0.04</td>
<td>0.32</td>
<td>0.64</td>
<td><span class="math inline">\(p=0.2\)</span></td>
<td>0</td>
<td>0.32</td>
</tr>
<tr class="even">
<td></td>
<td></td>
<td></td>
<td></td>
<td><span class="math inline">\(q=0.8\)</span></td>
<td></td>
<td></td>
</tr>
</tbody>
</table>
<ul>
<li><span class="math inline">\(H_O\)</span> observed heterozygosity averaged over all demes.</li>
<li><span class="math inline">\(H_S\)</span> expected heterozygosity averaged over all demes.</li>
<li><span class="math inline">\(H_T\)</span> expected heterozygosity of the metapopulation.</li>
<li><span class="math inline">\(F_{IS} = 1-\frac{H_O}{H_S}\)</span>; <span class="math inline">\(F_{IT} = 1-\frac{H_O}{H_T}\)</span>; <span class="math inline">\(F_{ST} = 1-\frac{H_S}{H_T}\)</span>.</li>
</ul>
</section>
<section id="example---3" class="level2">
<h2 class="anchored" data-anchor-id="example---3">Example - 3</h2>
<ul>
<li>Populations with different allele frequencies, one random mating and one with inbreeding.</li>
</ul>
<table class="table">
<colgroup>
<col style="width: 18%">
<col style="width: 15%">
<col style="width: 15%">
<col style="width: 15%">
<col style="width: 19%">
<col style="width: 7%">
<col style="width: 9%">
</colgroup>
<thead>
<tr class="header">
<th>Population</th>
<th><span class="math inline">\(A_1A_1\)</span></th>
<th><span class="math inline">\(A_1A_2\)</span></th>
<th><span class="math inline">\(A_2A_2\)</span></th>
<th>Allele Freq</th>
<th>F</th>
<th>He</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>1</td>
<td>0.25</td>
<td>0.5</td>
<td>0.25</td>
<td><span class="math inline">\(p=0.5\)</span></td>
<td>0</td>
<td>0.5</td>
</tr>
<tr class="even">
<td></td>
<td></td>
<td></td>
<td></td>
<td><span class="math inline">\(q=0.5\)</span></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td>2</td>
<td>0.14</td>
<td>0.13</td>
<td>0.74</td>
<td><span class="math inline">\(p=0.2\)</span></td>
<td>0.6</td>
<td>0.32</td>
</tr>
<tr class="even">
<td></td>
<td></td>
<td></td>
<td></td>
<td><span class="math inline">\(q=0.8\)</span></td>
<td></td>
<td></td>
</tr>
</tbody>
</table>
<ul>
<li><span class="math inline">\(H_O\)</span> observed heterozygosity averaged over all demes.</li>
<li><span class="math inline">\(H_S\)</span> expected heterozygosity averaged over all demes.</li>
<li><span class="math inline">\(H_T\)</span> expected heterozygosity of the metapopulation.</li>
<li><span class="math inline">\(F_{IS} = 1-\frac{H_O}{H_S}\)</span>; <span class="math inline">\(F_{IT} = 1-\frac{H_O}{H_T}\)</span>; <span class="math inline">\(F_{ST} = 1-\frac{H_S}{H_T}\)</span>.</li>
</ul>
</section>
<section id="example---4" class="level2">
<h2 class="anchored" data-anchor-id="example---4">Example - 4</h2>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="./class_13_15.jpg" class="img-fluid figure-img"></p>
<figcaption><span class="math inline">\(F_{ST}\)</span> increases over generations in fragmented populations at a rate inversely dependent on population size. Values that are <span class="math inline">\(\ge 0.15\)</span> indicate significant differentiation between fragments <span class="citation" data-cites="Frankham2010">(<a href="#ref-Frankham2010" role="doc-biblioref">Frankham et al., 2010</a>)</span>.</figcaption>
</figure>
</div>
</section>
<section id="migration-and-gene-flow" class="level2">
<h2 class="anchored" data-anchor-id="migration-and-gene-flow">Migration and gene flow</h2>
<ul>
<li>Migration reduces the impact of fragmentation in a way that is dependent on the rate of gene flow.</li>
<li>When is gene flow sufficiently high to prevent the genetic impact of fragmentation?</li>
</ul>
<p>. . .</p>
<div class="columns">
<div class="column" style="width:50%;">
<ul>
<li>A single migrant per generation is considered sufficient to prevent differentiation between demes.</li>
<li>The migration rate <span class="math inline">\(m\)</span> is the proportion of genes in a population derived from migrants per generation.</li>
<li><span class="math inline">\(Nm\)</span> is the number of migrants per generation <span class="citation" data-cites="Frankham2010">(<a href="#ref-Frankham2010" role="doc-biblioref">Frankham et al., 2010</a>)</span>.</li>
</ul>
</div><div class="column" style="width:50%;">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="./class_13_16.jpg" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:65.0%"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="references" class="level2">




</section>

<div id="quarto-appendix" class="default"><section class="quarto-appendix-contents" role="doc-bibliography" id="quarto-bibliography"><h2 class="anchored quarto-appendix-heading">References</h2><div id="refs" class="references csl-bib-body hanging-indent" data-entry-spacing="0" data-line-spacing="2" role="list">
<div id="ref-Akakaya2007" class="csl-entry" role="listitem">
Akçakaya, H. R., Mills, G., &amp; Doncaster, C. P. (2007). The role of metapopulations in conservation.
</div>
<div id="ref-Allendorf2022" class="csl-entry" role="listitem">
Allendorf, F. W., Funk, W. C., Aitken, S. N., Byrne, M., Luikart, G., &amp; Antunes, A. (2022). <em><span class="nocase">Conservation and the Genomics of Populations</span></em>. Oxford University Press. Retrieved from <a href="https://doi.org/10.1093/oso/9780198856566.001.0001">https://doi.org/10.1093/oso/9780198856566.001.0001</a>
</div>
<div id="ref-deBaan2013" class="csl-entry" role="listitem">
Baan, L. de, Alkemade, R., &amp; Koellner, T. (2013). Land use impacts on biodiversity in LCA: A global approach. <em>The International Journal of Life Cycle Assessment</em>, <em>18</em>(6), 1216–1230. Retrieved from <a href="https://link.springer.com/content/pdf/10.1007/s11367-012-0412-0.pdf">https://link.springer.com/content/pdf/10.1007/s11367-012-0412-0.pdf</a>
</div>
<div id="ref-Colosimo2021b" class="csl-entry" role="listitem">
Colosimo, G., Jackson, A. C., Benton, A., Varela-Stokes, A., Iverson, J., Knapp, C. R., &amp; Welch, M. (2021). <span class="nocase">Correlated Population Genetic Structure in a Three-Tiered Host-Parasite System: The Potential for Coevolution and Adaptive Divergence</span>. <em>Journal of Heredity</em>, <em>112</em>(7), 590–601. Retrieved from <a href="https://doi.org/10.1093/jhered/esab058">https://doi.org/10.1093/jhered/esab058</a>
</div>
<div id="ref-Colosimo2014" class="csl-entry" role="listitem">
Colosimo, G., Knapp, C. R., Wallace, L. E., &amp; Welch, M. E. (2014). <span class="nocase">Inferred vs realized patterns of gene flow: an analysis of population structure in the Andros Island Rock Iguana</span>. (V. Ketmaier, Ed.)<em>PLoS ONE</em>, <em>9</em>(9), e106963. Retrieved from <a href="http://dx.plos.org/10.1371/journal.pone.0106963">http://dx.plos.org/10.1371/journal.pone.0106963</a>
</div>
<div id="ref-Eldridge1999" class="csl-entry" role="listitem">
Eldridge, M. D. B., King, J. M., Loupis, A. K., Spencer, P. B. S., Taylor, A. C., Pope, L. C., &amp; Hall, G. P. (1999). Unprecedented low levels of genetic variation and inbreeding depression in an island population of the black-footed rock-wallaby. <em>Conservation Biology</em>, <em>13</em>(3), 531–541. Retrieved from <a href="https://conbio.onlinelibrary.wiley.com/doi/abs/10.1046/j.1523-1739.1999.98115.x">https://conbio.onlinelibrary.wiley.com/doi/abs/10.1046/j.1523-1739.1999.98115.x</a>
</div>
<div id="ref-Forbes1999" class="csl-entry" role="listitem">
Forbes, S. H., &amp; Hogg, J. T. (1999). Assessing population structure at high levels of differentiation: Microsatellite comparisons of bighorn sheep and large carnivores. <em>Animal Conservation</em>, <em>2</em>(3), 223–233. Retrieved from <a href="https://zslpublications.onlinelibrary.wiley.com/doi/abs/10.1111/j.1469-1795.1999.tb00068.x">https://zslpublications.onlinelibrary.wiley.com/doi/abs/10.1111/j.1469-1795.1999.tb00068.x</a>
</div>
<div id="ref-Frankham2010" class="csl-entry" role="listitem">
Frankham, R., Ballou, J. D., &amp; Briscoe, D. A. (2010). <em>Introduction to conservation genetics</em>. Cambridge University Press. Retrieved from <a href="https://doi.org/10.1017%2Fcbo9780511809002">https://doi.org/10.1017%2Fcbo9780511809002</a>
</div>
<div id="ref-Frankham2011" class="csl-entry" role="listitem">
Frankham, R., Ballou, J. D., Eldridge, M. D. B., Lacy, R. C., Ralls, K., Dudash, M. R., &amp; Fenster, C. B. (2011). <a href="https://doi.org/10.1111/j.1523-1739.2011.01662.x"><span class="nocase">Predicting the probability of outbreeding depression</span></a>. <em>Conservation Biology</em>, <em>25</em>(3), 465–475.
</div>
<div id="ref-Hanski1997" class="csl-entry" role="listitem">
Hanski, I., &amp; Simberloff, D. (1997). The metapopulation approach, its history, conceptual domain, and application to conservation. In I. Hanski &amp; M. E. Gilpin (Eds.), <em>Metapopulation biology</em> (pp. 5–26). San Diego: Academic Press. Retrieved from <a href="https://www.sciencedirect.com/science/article/pii/B9780123234452500031">https://www.sciencedirect.com/science/article/pii/B9780123234452500031</a>
</div>
<div id="ref-Malecot1948" class="csl-entry" role="listitem">
Malécot, G. (1948). <em>Les math<span>é</span>matiques de l’h<span>é</span>r<span>é</span>dit<span>é</span></em>. Masson. Retrieved from <a href="https://books.google.fi/books?id=g7FjAQAACAAJ">https://books.google.fi/books?id=g7FjAQAACAAJ</a>
</div>
<div id="ref-Nei1977" class="csl-entry" role="listitem">
Nei, M. (1977). F-statistics and analysis of gene diversity in subdivided populations. <em>Annals of Human Genetics</em>, <em>41</em>(2), 225–233. Wiley. Retrieved from <a href="https://doi.org/10.1111%2Fj.1469-1809.1977.tb01918.x">https://doi.org/10.1111%2Fj.1469-1809.1977.tb01918.x</a>
</div>
<div id="ref-Luci2011" class="csl-entry" role="listitem">
Nunes, L. (2011). <a href="https://doi.org/10.22498/pages.19.2.60">Landslides in s<span>ã</span>o paulo, brazil: An integrated historical perspective</a>. <em>Pages News</em>, <em>19</em>, 60–63.
</div>
<div id="ref-Riley2006" class="csl-entry" role="listitem">
Riley, S. P. D., Pollinger, J. P., Sauvajot, R. M., York, E. C., BROMLEY, C., FULLER, T. K., &amp; WAYNE, R. K. (2006). A southern california freeway is a physical and social barrier to gene flow in carnivores. <em>Molecular Ecology</em>, <em>15</em>(7), 1733–1741. Wiley. Retrieved from <a href="https://doi.org/10.1111%2Fj.1365-294x.2006.02907.x">https://doi.org/10.1111%2Fj.1365-294x.2006.02907.x</a>
</div>
<div id="ref-Thatte2018" class="csl-entry" role="listitem">
Thatte, P., Joshi, A., Vaidyanathan, S., Landguth, E., &amp; Ramakrishnan, U. (2018). Maintaining tiger connectivity and minimizing extinction into the next century: Insights from landscape genetics and spatially-explicit simulations. <em>Biological Conservation</em>, <em>218</em>, 181–191. Retrieved from <a href="https://www.sciencedirect.com/science/article/pii/S0006320717307346">https://www.sciencedirect.com/science/article/pii/S0006320717307346</a>
</div>
<div id="ref-Wahlund1928" class="csl-entry" role="listitem">
Wahlund, S. (1928). ZUSAMMENSETZUNG VON POPULATIONEN UND KORRELATIONSERSCHEINUNGEN VOM STANDPUNKT DER VERERBUNGSLEHRE AUS BETRACHTET. <em>Hereditas</em>, <em>11</em>(1), 65–106. Retrieved from <a href="https://onlinelibrary.wiley.com/doi/abs/10.1111/j.1601-5223.1928.tb02483.x">https://onlinelibrary.wiley.com/doi/abs/10.1111/j.1601-5223.1928.tb02483.x</a>
</div>
<div id="ref-Welch2017" class="csl-entry" role="listitem">
Welch, M. E., Colosimo, G., Pasachnik, S. A., Malone, C. L., Hilton, J., Long, J., Getz, A. H., et al. (2017). <span class="nocase">Molecular variation and population structure in critically endangered Turks and Caicos Rock Iguanas: identifying intraspecific conservation units and revising subspecific taxonomy</span>. <em>Conservation Genetics</em>, <em>18</em>(2), 479–493. Retrieved from <a href="http://link.springer.com/10.1007/s10592-016-0922-6">http://link.springer.com/10.1007/s10592-016-0922-6</a>
</div>
<div id="ref-Wright1931" class="csl-entry" role="listitem">
Wright, S. (1931). <a href="https://doi.org/10.1016/S0092-8240(05)80011-4"><span class="nocase">Evolution in Mendelian populations</span></a>. <em>Bulletin of Mathematical Biology</em>, <em>52</em>(1-2), 241–295.
</div>
<div id="ref-Wright1951" class="csl-entry" role="listitem">
Wright, S. (1951). <a href="https://doi.org/10.2307/2407273"><span class="nocase">The genetical structure of populations</span></a>. <em>Annals of eugenics</em>, <em>15</em>, 323–354.
</div>
</div></section></div></main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const disableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'prefetch';
    }
  }
  const enableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'stylesheet';
    }
  }
  const manageTransitions = (selector, allowTransitions) => {
    const els = window.document.querySelectorAll(selector);
    for (let i=0; i < els.length; i++) {
      const el = els[i];
      if (allowTransitions) {
        el.classList.remove('notransition');
      } else {
        el.classList.add('notransition');
      }
    }
  }
  const toggleGiscusIfUsed = (isAlternate, darkModeDefault) => {
    const baseTheme = document.querySelector('#giscus-base-theme')?.value ?? 'light';
    const alternateTheme = document.querySelector('#giscus-alt-theme')?.value ?? 'dark';
    let newTheme = '';
    if(darkModeDefault) {
      newTheme = isAlternate ? baseTheme : alternateTheme;
    } else {
      newTheme = isAlternate ? alternateTheme : baseTheme;
    }
    const changeGiscusTheme = () => {
      // From: https://github.com/giscus/giscus/issues/336
      const sendMessage = (message) => {
        const iframe = document.querySelector('iframe.giscus-frame');
        if (!iframe) return;
        iframe.contentWindow.postMessage({ giscus: message }, 'https://giscus.app');
      }
      sendMessage({
        setConfig: {
          theme: newTheme
        }
      });
    }
    const isGiscussLoaded = window.document.querySelector('iframe.giscus-frame') !== null;
    if (isGiscussLoaded) {
      changeGiscusTheme();
    }
  }
  const toggleColorMode = (alternate) => {
    // Switch the stylesheets
    const alternateStylesheets = window.document.querySelectorAll('link.quarto-color-scheme.quarto-color-alternate');
    manageTransitions('#quarto-margin-sidebar .nav-link', false);
    if (alternate) {
      enableStylesheet(alternateStylesheets);
      for (const sheetNode of alternateStylesheets) {
        if (sheetNode.id === "quarto-bootstrap") {
          toggleBodyColorMode(sheetNode);
        }
      }
    } else {
      disableStylesheet(alternateStylesheets);
      toggleBodyColorPrimary();
    }
    manageTransitions('#quarto-margin-sidebar .nav-link', true);
    // Switch the toggles
    const toggles = window.document.querySelectorAll('.quarto-color-scheme-toggle');
    for (let i=0; i < toggles.length; i++) {
      const toggle = toggles[i];
      if (toggle) {
        if (alternate) {
          toggle.classList.add("alternate");     
        } else {
          toggle.classList.remove("alternate");
        }
      }
    }
    // Hack to workaround the fact that safari doesn't
    // properly recolor the scrollbar when toggling (#1455)
    if (navigator.userAgent.indexOf('Safari') > 0 && navigator.userAgent.indexOf('Chrome') == -1) {
      manageTransitions("body", false);
      window.scrollTo(0, 1);
      setTimeout(() => {
        window.scrollTo(0, 0);
        manageTransitions("body", true);
      }, 40);  
    }
  }
  const isFileUrl = () => { 
    return window.location.protocol === 'file:';
  }
  const hasAlternateSentinel = () => {  
    let styleSentinel = getColorSchemeSentinel();
    if (styleSentinel !== null) {
      return styleSentinel === "alternate";
    } else {
      return false;
    }
  }
  const setStyleSentinel = (alternate) => {
    const value = alternate ? "alternate" : "default";
    if (!isFileUrl()) {
      window.localStorage.setItem("quarto-color-scheme", value);
    } else {
      localAlternateSentinel = value;
    }
  }
  const getColorSchemeSentinel = () => {
    if (!isFileUrl()) {
      const storageValue = window.localStorage.getItem("quarto-color-scheme");
      return storageValue != null ? storageValue : localAlternateSentinel;
    } else {
      return localAlternateSentinel;
    }
  }
  const darkModeDefault = false;
  let localAlternateSentinel = darkModeDefault ? 'alternate' : 'default';
  // Dark / light mode switch
  window.quartoToggleColorScheme = () => {
    // Read the current dark / light value 
    let toAlternate = !hasAlternateSentinel();
    toggleColorMode(toAlternate);
    setStyleSentinel(toAlternate);
    toggleGiscusIfUsed(toAlternate, darkModeDefault);
  };
  // Ensure there is a toggle, if there isn't float one in the top right
  if (window.document.querySelector('.quarto-color-scheme-toggle') === null) {
    const a = window.document.createElement('a');
    a.classList.add('top-right');
    a.classList.add('quarto-color-scheme-toggle');
    a.href = "";
    a.onclick = function() { try { window.quartoToggleColorScheme(); } catch {} return false; };
    const i = window.document.createElement("i");
    i.classList.add('bi');
    a.appendChild(i);
    window.document.body.appendChild(a);
  }
  // Switch to dark mode if need be
  if (hasAlternateSentinel()) {
    toggleColorMode(true);
  } else {
    toggleColorMode(false);
  }
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp("https:\/\/giulianocolosimo\.github\.com");
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-left">
<p>© Copyright 2024 Giuliano Colosimo</p>
</div>   
    <div class="nav-footer-center">
      &nbsp;
    <div class="toc-actions d-sm-block d-md-none"><ul><li><a href="https://github.com/quarto-dev/quarto-demo/issues/new" class="toc-action"><i class="bi bi-github"></i>Report an issue</a></li></ul></div></div>
    <div class="nav-footer-right">
<p>This page is built with ❤️ and <a href="https://quarto.org/">Quarto</a>. Inspiration for this site comes from <a href="https://www.mm218.dev">here</a>.</p>
</div>
  </div>
</footer>




</body></html>